// @coinbase/cdp-react/dist/components/forms/OTP/index.js
import { jsxs as f, jsx as t } from "react/jsx-runtime";
import { OneTimePasswordField as T, OneTimePasswordFieldInput as y, OneTimePasswordFieldHiddenInput as I } from "@radix-ui/react-one-time-password-field";
import { forwardRef as N, useId as s, useMemo as v, useRef as w, useEffect as $ } from "react";
import { Input as j } from "../Input/index.js";
import { Label as R } from "../Label/index.js";
import { Error as F } from "../../ui/Error/index.js";
import { SuccessMessage as x } from "../../ui/SuccessMessage/index.js";
import '../../../assets/OTP.css';const A = "OTP-module__otp___UjnM8", B = "OTP-module__input___Vbd5m", E = "OTP-module__label___cK41n", o = {
  otp: A,
  "group-container": "OTP-module__group-container___hNqnB",
  "input-container": "OTP-module__input-container___snmj7",
  input: B,
  label: E
}, q = N(
  ({ error: n, successMessage: i, passwordLength: d = 6, className: b = "", label: l, ...m }, r) => {
    const c = s(), p = s(), _ = s(), h = v(() => {
      if (n)
        return c;
      if (i)
        return p;
    }, [n, c, i, p]), u = w(/* @__PURE__ */ new Map()), O = (e, a) => {
      e ? u.current.set(a, e) : u.current.delete(a);
    };
    return $(() => {
      const e = Array.from(u.current.values());
      typeof r == "function" ? r(e) : r && (r.current = e);
    }, [d, r]), /* @__PURE__ */ f("div", { className: `${o.otp} ${b}`, children: [
      l && /* @__PURE__ */ t(R, { as: "p", id: _, className: o.label, "data-part": "label", children: l }),
      /* @__PURE__ */ f(
        T,
        {
          ...m,
          className: o["group-container"],
          "data-part": "input-group",
          "aria-invalid": !!n,
          "aria-describedby": h,
          "aria-labelledby": l ? _ : void 0,
          style: {
            "--cdp-web-otp-input-width": `${Math.floor(100 / d)}%`
          },
          children: [
            Array.from({ length: d }, (e, a) => /* @__PURE__ */ t("div", { className: o["input-container"], "data-part": "input-container", children: /* @__PURE__ */ t(y, { asChild: !0, children: /* @__PURE__ */ t(
              j,
              {
                name: `${m.name || "otp"}-${a}`,
                ref: (P) => O(P, a),
                className: o.input,
                "aria-invalid": !!n,
                "data-part": "input",
                "data-success": !!i
              }
            ) }) }, a)),
            /* @__PURE__ */ t(I, {})
          ]
        }
      ),
      !!n && /* @__PURE__ */ t(F, { id: c, "data-part": "error", children: n }),
      !!i && /* @__PURE__ */ t(x, { id: p, "data-part": "success", children: i })
    ] });
  }
);
q.displayName = "OTP";
export {
  q as OTP
};
