// @coinbase/cdp-react/dist/components/Fund/index.js
import { jsx as n, jsxs as i, Fragment as f } from "react/jsx-runtime";
import { u as p } from "../../chunks/useSendComponentCallOnce.BHZMuo6E.js";
import { useId as c } from "react";
import { useAppConfig as F } from "../CDPReactProvider/index.js";
import { CoinbaseFooter as a } from "../ui/CoinbaseFooter/index.js";
import { FundForm as l } from "./FundForm.js";
import { FundProvider as C, useFundContext as _ } from "./FundProvider.js";
import { FundTitle as b } from "./FundTitle.js";
import { useSetupOnrampEventListeners as x } from "./hooks/useSetupOnrampEventListeners.js";
import '../../assets/Fund.css';const g = "Fund-module__fund___6j-Og", h = {
  fund: g
}, N = ({
  className: o = "",
  children: r,
  openIn: e,
  style: s,
  submitLabel: u,
  title: t,
  ...m
}) => (p("fund"), /* @__PURE__ */ n(C, { ...m, children: /* @__PURE__ */ n("div", { className: `${h.fund} ${o}`, style: s, children: /* @__PURE__ */ n(j, { openIn: e, submitLabel: u, title: t, children: r }) }) })), j = ({
  children: o,
  openIn: r,
  submitLabel: e,
  title: s
}) => {
  const { state: u } = _(), t = c(), { showCoinbaseFooter: m } = F();
  return x(), o ? typeof o == "function" ? o(u) : o : /* @__PURE__ */ n(l, { "aria-labelledby": t, openIn: r, submitLabel: e, children: ({ Content: d }) => /* @__PURE__ */ i(f, { children: [
    /* @__PURE__ */ n(b, { id: t, children: s }),
    d,
    m && /* @__PURE__ */ n(v, {})
  ] }) });
}, v = (o) => /* @__PURE__ */ n(a, { ...o });
export {
  N as Fund,
  v as FundFooter,
  l as FundForm,
  b as FundTitle,
  _ as useFundContext
};
