// @coinbase/cdp-react/dist/components/Fund/useFundReducer.js
import { useReducer as t } from "react";
const n = (a, e) => {
  if (e.statusName === "init" || e.statusName === "exit" || e.statusName === "transactionPending")
    return e;
  if (e.statusName === "error" || e.statusName === "transactionSuccess") {
    if (a.statusName === "error")
      return e;
    if (a.statusName === "transactionSuccess" && e.statusName === "transactionSuccess" && a.statusData)
      return {
        ...e,
        statusData: {
          ...a.statusData,
          ...e.statusData
        }
      };
  }
  return e;
};
function s(a, e) {
  const { type: r } = e;
  switch (r) {
    case "SET_AMOUNTS":
      return {
        ...a,
        cryptoAmount: e.payload.cryptoAmount,
        fiatAmount: e.payload.fiatAmount
      };
    case "SET_FIELD":
      return e.payload ? { ...a, [e.payload.field]: e.payload.value } : a;
    case "FETCH_EXCHANGE_RATE":
      return {
        ...a,
        isExchangeRatePending: !0,
        exchangeRateError: null
      };
    case "SET_EXCHANGE_RATE_SUCCESS":
      return {
        ...a,
        isExchangeRatePending: !1,
        exchangeRate: e.payload.exchangeRate,
        exchangeRateError: null
      };
    case "SET_EXCHANGE_RATE_ERROR":
      return {
        ...a,
        exchangeRate: void 0,
        isExchangeRatePending: !1,
        exchangeRateError: {
          code: e.payload.error.code || "UNKNOWN_EXCHANGE_RATE_ERROR",
          message: e.payload.error.message || "Something went wrong fetching exchange rate"
        }
      };
    case "FETCH_PAYMENT_METHODS":
      return {
        ...a,
        isPaymentMethodsPending: !0,
        paymentMethodsError: null
      };
    case "SET_PAYMENT_METHODS_SUCCESS":
      return {
        ...a,
        isPaymentMethodsPending: !1,
        paymentMethods: e.payload.paymentMethods,
        selectedPaymentMethod: e.payload.paymentMethods[0],
        paymentMethodsError: null
      };
    case "SET_PAYMENT_METHODS_ERROR":
      return {
        ...a,
        paymentMethods: void 0,
        isPaymentMethodsPending: !1,
        paymentMethodsError: {
          code: e.payload.error.code || "UNKNOWN_PAYMENT_METHODS_ERROR",
          message: e.payload.error.message || "Something went wrong fetching payment methods"
        }
      };
    case "SET_TRANSACTION_STATUS":
      return {
        ...a,
        transactionStatus: n(
          a.transactionStatus,
          e.payload.transactionStatus
        )
      };
    case "SYNC_WITH_PROPS":
      return {
        ...a,
        ...e.payload
      };
    default:
      throw new Error("Unknown action type");
  }
}
const d = (a) => t(s, a), E = (a, e) => ({
  type: "SET_FIELD",
  payload: {
    field: a,
    value: e
  }
});
export {
  E as createSetFieldAction,
  d as useFundReducer
};
