// @coinbase/cdp-react/dist/components/ui/Banner/index.js
import { jsxs as e, jsx as t } from "react/jsx-runtime";
import { c as l } from "../../../chunks/lite.1fxw3LjI.js";
import "react";
import { IconCheckCircle as _ } from "../../../icons/IconCheckCircle.js";
import { IconExclamationCircle as x } from "../../../icons/IconExclamationCircle.js";
import { IconExclamationTriangle as d } from "../../../icons/IconExclamationTriangle.js";
import '../../../assets/Banner.css';const p = "Banner-module__banner___vODjX", f = "Banner-module__icon___wzB6C", u = "Banner-module__text___fTYC-", c = {
  banner: p,
  icon: f,
  text: u
}, B = {
  error: x,
  success: _,
  warning: d
}, N = ({ className: a = "", children: s, icon: n, variant: o, ...i }) => {
  const r = o ? B[o] : void 0, m = n || r;
  return /* @__PURE__ */ e("div", { "data-variant": o, className: l(c.banner, a), ...i, children: [
    m && /* @__PURE__ */ e("span", { className: c.icon, children: [
      n,
      !n && r && /* @__PURE__ */ t(r, {})
    ] }),
    /* @__PURE__ */ t("div", { className: c.text, children: s })
  ] });
};
export {
  N as Banner
};
