// @coinbase/cdp-core/dist/native/index.native113.js
import { handleChainId as t } from "./index.native139.js";
import { handleDisconnect as e } from "./index.native140.js";
import { handleSwitchEthereumChain as a } from "./index.native141.js";
import { handleAccounts as n } from "./index.native142.js";
import { handleEthSign as o } from "./index.native143.js";
import { handleGetCallsStatus as r } from "./index.native144.js";
import { handleGetCapabilities as i } from "./index.native145.js";
import { handlePersonalSign as l } from "./index.native146.js";
import { handleRequestAccounts as h } from "./index.native147.js";
import { handleSendCalls as m } from "./index.native148.js";
import { handleSendTransaction as s } from "./index.native149.js";
import { handleSignTypedData as d } from "./index.native150.js";
function E() {
  return {
    eth_accounts: n,
    eth_requestAccounts: h,
    personal_sign: l,
    eth_sendTransaction: s,
    eth_signTypedData_v4: d,
    wallet_disconnect: e,
    wallet_switchEthereumChain: a,
    eth_sign: o,
    eth_chainId: t,
    wallet_sendCalls: m,
    wallet_getCallsStatus: r,
    wallet_getCapabilities: i
  };
}
export {
  E as createSmartAccountHandlers
};
