// @coinbase/cdp-react/dist/components/SignIn/hooks/useEmailForm.js
import "@coinbase/cdp-hooks";
import { useCallback as p, useMemo as m } from "react";
import { isApiError as y } from "../../../utils/isApiError.js";
import { useSignInContext as d } from "../SignInProvider.js";
const f = ({
  submit: t
}) => {
  const { state: n, dispatch: o } = d(), a = "An error occurred while signing in. Please try again.", s = p(
    (e) => {
      o({ type: "SET_EMAIL", payload: { email: e } });
    },
    [o]
  ), i = p(
    async ({
      email: e,
      onSuccess: l,
      onError: E
    }) => {
      if (!n.isPending) {
        o({ type: "SUBMIT_EMAIL", payload: { email: e } });
        try {
          const { flowId: r } = await t(e);
          o({ type: "SUBMIT_EMAIL_SUCCESS", payload: { flowId: r } }), l?.();
        } catch (r) {
          const c = y(r) ? r : r instanceof Error && r.message || a;
          o({ type: "SUBMIT_EMAIL_FAILURE", payload: { error: c } }), console.error(r), E?.(c);
        }
      }
    },
    [o, n.isPending, t]
  );
  return m(
    () => ({
      setEmail: s,
      submitEmail: i
    }),
    [s, i]
  );
};
export {
  f as useEmailForm
};
