// @coinbase/cdp-react/dist/components/SignIn/hooks/useOTPForm.js
import "@coinbase/cdp-hooks";
import { useCallback as i, useEffect as l, useMemo as u } from "react";
import { useTimer as E } from "../../../hooks/useTimer.js";
import { isApiError as S } from "../../../utils/isApiError.js";
import { useSignInContext as _ } from "../SignInProvider.js";
const A = ({
  passwordLength: a,
  submit: T
}) => {
  const { state: n, dispatch: e } = _(), {
    timeRemaining: o,
    start: c,
    reset: r
  } = E(), p = "An error occurred while signing in. Please try again.", m = i(
    (t) => {
      e({ type: "SET_OTP", payload: { otp: t } });
    },
    [e]
  ), y = i(
    async ({
      otp: t,
      onSuccess: d,
      onError: f
    }) => {
      if (!n.isPending) {
        if (t.length !== a) {
          e({ type: "SET_OTP", payload: { otp: t } });
          return;
        }
        e({ type: "SUBMIT_OTP", payload: { otp: t } });
        try {
          await T(t), e({ type: "SUBMIT_OTP_SUCCESS", payload: { otp: t } }), d?.(), r();
        } catch (s) {
          const O = S(s) ? s : s instanceof Error && s.message || p;
          e({ type: "SUBMIT_OTP_FAILURE", payload: { error: O } }), console.error(s), f?.(O);
        }
      }
    },
    [e, n.isPending, a, T, r]
  ), P = i(() => {
    e({ type: "RESET_OTP" }), r();
  }, [e, r]);
  return l(() => {
    o === 0 && n.step === "verification" && !n.canResetOTP && (e({ type: "ALLOW_RESET_OTP" }), r());
  }, [o, r, n.step, n.canResetOTP, e]), u(
    () => ({
      resendCountdown: o,
      resetOTP: P,
      resetResendTimer: r,
      setOTP: m,
      startResendTimer: c,
      submitOtp: y
    }),
    [o, P, r, m, c, y]
  );
};
export {
  A as useOTPForm
};
