// @coinbase/cdp-react/dist/components/SignIn/hooks/usePhoneNumberForm.js
import "@coinbase/cdp-hooks";
import { useCallback as p, useMemo as l } from "react";
import { isApiError as y } from "../../../utils/isApiError.js";
import { useSignInContext as d } from "../SignInProvider.js";
const U = ({
  submit: t
}) => {
  const { state: n, dispatch: o } = d(), s = "An error occurred while signing in. Please try again.", a = p(
    (e) => {
      o({ type: "SET_PHONE_NUMBER", payload: { phoneNumber: e } });
    },
    [o]
  ), i = p(
    async ({
      phoneNumber: e,
      onSuccess: E,
      onError: m
    }) => {
      if (!n.isPending) {
        o({ type: "SUBMIT_PHONE_NUMBER", payload: { phoneNumber: e } });
        try {
          const { flowId: r } = await t(e);
          o({ type: "SUBMIT_PHONE_NUMBER_SUCCESS", payload: { flowId: r } }), E?.();
        } catch (r) {
          const c = y(r) ? r : r instanceof Error && r.message || s;
          o({ type: "SUBMIT_PHONE_NUMBER_FAILURE", payload: { error: c } }), console.error(r), m?.(c);
        }
      }
    },
    [o, n.isPending, t]
  );
  return l(
    () => ({
      setPhoneNumber: a,
      submitPhoneNumber: i
    }),
    [a, i]
  );
};
export {
  U as usePhoneNumberForm
};
