// @coinbase/cdp-react/dist/components/CopySolanaKeyButton/index.js
import { jsx as B } from "react/jsx-runtime";
import { useConfig as _ } from "@coinbase/cdp-hooks";
import { u as j } from "../../chunks/useSendComponentCallOnce.BHZMuo6E.js";
import { useRef as K, useState as i, useCallback as T } from "react";
import "../ui/Button/index.js";
import { IframeButton as U } from "../ui/IframeButton/index.js";
import { useKeyExportPostMessage as v } from "../../hooks/useKeyExportPostMessage.js";
import "@coinbase/cdp-core";
const F = ({
  address: m,
  className: p = "",
  copiedLabel: c = "Copied",
  icon: e = !0,
  fullWidth: u,
  label: t = "Copy key",
  onReady: o,
  onSessionExpired: r,
  onSuccess: s,
  onError: a,
  size: f,
  theme: l,
  variant: d
}) => {
  j("copy_solana_key_button");
  const {
    config: { projectId: y, secureIframeBasePath: b }
  } = _(), n = K(null), [h, k] = i(void 0), [C, g] = i(!1), x = T(
    (S, w) => {
      switch (S) {
        case "ready":
          o?.();
          break;
        case "success":
          s?.();
          break;
        case "error":
          a?.(w);
          break;
        case "expired":
          g(!0), r?.();
          break;
      }
    },
    [o, s, a, r]
  ), { iframeUrl: I, isPending: P } = v({
    address: m,
    type: "solana",
    projectId: y,
    label: t,
    basePath: b,
    copiedLabel: c,
    icon: e,
    iframeRef: n,
    onStatusUpdate: x,
    theme: h
  });
  return C ? null : /* @__PURE__ */ B(
    U,
    {
      "data-testid": "copy-solana-key-button",
      ref: n,
      src: I,
      allow: "clipboard-read; clipboard-write",
      isPending: P,
      label: t,
      icon: e,
      fullWidth: u,
      onThemeChange: k,
      size: f,
      theme: l,
      variant: d,
      className: p
    }
  );
};
export {
  F as CopySolanaKeyButton
};
