// @coinbase/cdp-react/dist/components/ui/Button/index.js
import { jsx as t, jsxs as y, Fragment as c } from "react/jsx-runtime";
import { c as p } from "../../../chunks/lite.1fxw3LjI.js";
import { forwardRef as u } from "react";
import { ButtonBase as k } from "../ButtonBase/index.js";
import { LoadingSpinner as B } from "../LoadingSpinner/index.js";
import { VisuallyHidden as f } from "../VisuallyHidden/index.js";
import "../../../theme/theme.js";
import '../../../assets/Button.css';const b = "Button-module__btn___Wj6Dr", h = "Button-module__primary___WGT--", S = "Button-module__secondary___Fx0LJ", r = {
  btn: b,
  "link-primary": "Button-module__link-primary___6Zgkh",
  "link-secondary": "Button-module__link-secondary___S0Qdo",
  primary: h,
  secondary: S,
  "transparent-primary": "Button-module__transparent-primary___iwWv5",
  "transparent-secondary": "Button-module__transparent-secondary___GfRXu",
  "control-btn": "Button-module__control-btn___9Gkoz"
}, e = {
  primary: r.primary,
  secondary: r.secondary,
  linkPrimary: r["link-primary"],
  linkSecondary: r["link-secondary"],
  transparentPrimary: r["transparent-primary"],
  transparentSecondary: r["transparent-secondary"],
  control: r["control-btn"]
}, x = u(
  ({
    children: i,
    className: m = "",
    isPending: o,
    pendingLabel: s = "Loading...",
    size: _ = "md",
    variant: n = "primary",
    ...a
  }, d) => {
    const l = n === "linkPrimary" || n === "linkSecondary" ? "none" : _;
    return /* @__PURE__ */ t(
      k,
      {
        className: a.asChild ? "" : p(r.btn, e[n] ?? e.primary, m),
        ref: d,
        isPending: o,
        size: l,
        "data-variant": n,
        ...a,
        children: o ? /* @__PURE__ */ y(c, { children: [
          /* @__PURE__ */ t(B, { "aria-hidden": "true" }),
          /* @__PURE__ */ t(f, { children: s })
        ] }) : i
      }
    );
  }
);
x.displayName = "Button";
export {
  x as Button
};
