// @coinbase/cdp-react/dist/components/ui/LoadingSpinner/index.js
import { jsx as n, jsxs as s } from "react/jsx-runtime";
import "react";
import '../../../assets/LoadingSpinner.css';const p = "LoadingSpinner-module__spinner___-zo6Y", i = {
  spinner: p,
  "spinner-circle": "LoadingSpinner-module__spinner-circle___9Jfm7"
}, g = ({
  className: o,
  staticStroke: c,
  staticStrokeColor: r = "currentColor",
  strokeWidth: e = "3",
  style: l
}) => /* @__PURE__ */ n(
  "svg",
  {
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 24 24",
    role: "img",
    "aria-label": "loading",
    width: "24",
    height: "24",
    stroke: "currentColor",
    className: o,
    style: l,
    children: /* @__PURE__ */ s("g", { className: i.spinner, children: [
      c && /* @__PURE__ */ n(
        "circle",
        {
          cx: "12",
          cy: "12",
          r: "9.5",
          fill: "none",
          strokeWidth: e,
          stroke: r,
          opacity: r === "currentColor" ? "0.2" : void 0
        }
      ),
      /* @__PURE__ */ n(
        "circle",
        {
          className: i["spinner-circle"],
          cx: "12",
          cy: "12",
          r: "9.5",
          fill: "none",
          strokeWidth: e
        }
      )
    ] })
  }
);
export {
  g as LoadingSpinner
};
