// @coinbase/cdp-core/dist/esm/index.web60.js
import { Analytics as E } from "./index.web2.js";
import "./index.web3.js";
import { VERSION as r } from "./index.web8.js";
import "./index.web39.js";
import { APIError as c, ErrorType as P, HttpErrorType as U, OAuth2ProviderType as F, SendEvmTransactionWithEndUserAccountBodyNetwork as I, SendSolanaTransactionWithEndUserAccountBodyNetwork as d } from "@coinbase/cdp-api-client";
import "@solana/web3.js";
import "bs58";
import "jose";
import "viem";
import "./index.web48.js";
import "@solana/kit";
import "zustand";
import { DEFAULT_SECURE_IFRAME_BASE_PATH as y, DEFAULT_SECURE_IFRAME_THEME as M, SECURE_IFRAME_EVENT_TYPE as N, SECURE_IFRAME_EVENT_TYPE_PREFIX as C, SECURE_IFRAME_KEY_EXPORT_EVENT_TYPE as V } from "./index.web36.js";
E.registerPackageVersion("core", r);
export {
  c as APIError,
  y as DEFAULT_SECURE_IFRAME_BASE_PATH,
  M as DEFAULT_SECURE_IFRAME_THEME,
  P as ErrorType,
  U as HttpErrorType,
  F as OAuth2ProviderType,
  N as SECURE_IFRAME_EVENT_TYPE,
  C as SECURE_IFRAME_EVENT_TYPE_PREFIX,
  V as SECURE_IFRAME_KEY_EXPORT_EVENT_TYPE,
  I as SendEvmTransactionWithEndUserAccountBodyNetwork,
  d as SendSolanaTransactionWithEndUserAccountBodyNetwork
};
