// @coinbase/cdp-core/dist/esm/index.native.js
import { Analytics as r } from "./index.native2.js";
import "./index.native3.js";
import { nativeCrypto as o } from "./index.native4.js";
import { nativeOAuth as t } from "./index.native5.js";
import { nativeSecureStorage as e } from "./index.native6.js";
import { nativeStorage as n } from "./index.native7.js";
import { setPlatformServices as i } from "./index.native8.js";
import { VERSION as a } from "./index.native9.js";
import { authenticateWithJWT as A, exportEvmAccount as h, exportSolanaAccount as v, getAccessToken as T, getAccessTokenExpiration as O, getCurrentUser as y, getUserOperation as P, initialize as k, isSignedIn as M, onAuthStateChange as C, onOAuthStateChange as I, sendEvmTransaction as W, sendSolanaTransaction as V, sendUserOperation as R, signEvmHash as U, signEvmMessage as D, signEvmTransaction as N, signEvmTypedData as b, signInWithEmail as w, signInWithOAuth as B, signInWithSms as H, signOut as _, signSolanaMessage as z, signSolanaTransaction as G, verifyEmailOTP as J, verifyOAuth as X, verifySmsOTP as j } from "./index.native10.js";
import { createSpendPermission as F } from "./index.native11.js";
import { listSpendPermissions as L } from "./index.native12.js";
import { APIError as Y, ErrorType as Z, HttpErrorType as $, OAuth2ProviderType as rr, SendEvmTransactionWithEndUserAccountBodyNetwork as or, SendSolanaTransactionWithEndUserAccountBodyNetwork as tr } from "@coinbase/cdp-api-client";
import { revokeSpendPermission as nr } from "./index.native13.js";
import { getCurrentUserSync as ar } from "./index.native14.js";
import { fetchWithX402 as sr } from "./index.native15.js";
import { linkGoogle as fr } from "./index.native16.js";
import { linkSms as lr } from "./index.native17.js";
import { linkEmail as gr } from "./index.native18.js";
import { linkOAuth as xr } from "./index.native19.js";
import { linkApple as dr } from "./index.native20.js";
import { createEvmEoaAccount as hr } from "./index.native21.js";
import { createEvmSmartAccount as Tr } from "./index.native22.js";
import { createSolanaAccount as yr } from "./index.native23.js";
import { getMfaConfig as kr } from "./index.native24.js";
import { recordMfaEnrollmentPrompted as Cr } from "./index.native25.js";
import { initiateMfaEnrollment as Wr } from "./index.native26.js";
import { submitMfaEnrollment as Rr } from "./index.native27.js";
import { initiateMfaVerification as Dr } from "./index.native28.js";
import { submitMfaVerification as br } from "./index.native29.js";
import { AuthManager as Br } from "./index.native30.js";
import { CustomAuthError as _r } from "./index.native31.js";
import { toViemAccount as Gr } from "./index.native32.js";
import { InputValidationError as Xr, InternalError as jr, ModuleResolutionError as qr, ValidationError as Fr } from "./index.native33.js";
import { createCDPEmbeddedWallet as Lr } from "./index.native34.js";
import { EIP1193ProviderError as Yr, STANDARD_ERROR_CODES as Zr } from "./index.native35.js";
import "viem";
import "zustand";
r.registerPackageVersion("core", a);
i({
  crypto: o,
  storage: n,
  secureStorage: e,
  oauth: t
});
export {
  Y as APIError,
  Br as AuthManager,
  _r as CustomAuthError,
  Yr as EIP1193ProviderError,
  Z as ErrorType,
  $ as HttpErrorType,
  Xr as InputValidationError,
  jr as InternalError,
  qr as ModuleResolutionError,
  rr as OAuth2ProviderType,
  Zr as STANDARD_ERROR_CODES,
  or as SendEvmTransactionWithEndUserAccountBodyNetwork,
  tr as SendSolanaTransactionWithEndUserAccountBodyNetwork,
  Fr as ValidationError,
  A as authenticateWithJWT,
  Lr as createCDPEmbeddedWallet,
  hr as createEvmEoaAccount,
  Tr as createEvmSmartAccount,
  yr as createSolanaAccount,
  F as createSpendPermission,
  h as exportEvmAccount,
  v as exportSolanaAccount,
  sr as fetchWithX402,
  T as getAccessToken,
  O as getAccessTokenExpiration,
  y as getCurrentUser,
  ar as getCurrentUserSync,
  kr as getMfaConfig,
  P as getUserOperation,
  k as initialize,
  Wr as initiateMfaEnrollment,
  Dr as initiateMfaVerification,
  M as isSignedIn,
  dr as linkApple,
  gr as linkEmail,
  fr as linkGoogle,
  xr as linkOAuth,
  lr as linkSms,
  L as listSpendPermissions,
  C as onAuthStateChange,
  I as onOAuthStateChange,
  Cr as recordMfaEnrollmentPrompted,
  nr as revokeSpendPermission,
  W as sendEvmTransaction,
  V as sendSolanaTransaction,
  R as sendUserOperation,
  U as signEvmHash,
  D as signEvmMessage,
  N as signEvmTransaction,
  b as signEvmTypedData,
  w as signInWithEmail,
  B as signInWithOAuth,
  H as signInWithSms,
  _ as signOut,
  z as signSolanaMessage,
  G as signSolanaTransaction,
  Rr as submitMfaEnrollment,
  br as submitMfaVerification,
  Gr as toViemAccount,
  J as verifyEmailOTP,
  X as verifyOAuth,
  j as verifySmsOTP
};
