// @coinbase/cdp-core/dist/esm/index.native148.js
import { sendUserOperation as d } from "./index.native10.js";
import { isChainIdSupportedForCDPUserOps as m, userOperationChainIdToNameMapping as l } from "./index.native47.js";
import { validateUserHasEvmSmartAccount as f, RPCRequestError as o, STANDARD_ERROR_CODES as n } from "./index.native35.js";
async function I(r, t) {
  const [{ calls: i, capabilities: e }] = r, c = (await f()).evmSmartAccounts[0], s = t.getState().chainId;
  if (!m(s))
    throw new o(
      n.rpc.invalidParams,
      "Chain ID is not supported for user operations"
    );
  const u = e?.dataSuffix?.value, p = e?.paymasterService?.url;
  try {
    return (await d({
      evmSmartAccount: c,
      calls: v(i),
      network: l[s],
      paymasterUrl: p,
      dataSuffix: u
    })).userOperationHash;
  } catch (a) {
    throw new o(
      n.rpc.transactionRejected,
      a instanceof Error ? a.message : "User operation failed"
    );
  }
}
function v(r) {
  return r.map((t) => ({
    to: t.to,
    data: t.data,
    value: t.value ? BigInt(t.value) : void 0
  }));
}
export {
  I as handleSendCalls
};
