// @coinbase/cdp-core/dist/esm/index.web35.js
import { InputValidationError as o } from "./index.web32.js";
import { DEFAULT_SECURE_IFRAME_BASE_PATH as l, DEFAULT_SECURE_IFRAME_THEME as i, SECURE_IFRAME_KEY_EXPORT_EVENT_TYPE as f, SECURE_IFRAME_EVENT_TYPE_PREFIX as u } from "./index.web36.js";
function s() {
  if (!document)
    throw new Error("Document not found. This function must be called in a browser environment.");
}
function p(e) {
  const t = e.basePath || l, n = ["fullWidth"];
  return e.projectId && n.push(`projectId=${e.projectId.trim()}`), e.label && n.push(`label=${encodeURIComponent(e.label.trim())}`), e.copiedLabel && n.push(`copiedLabel=${encodeURIComponent(e.copiedLabel.trim())}`), e.icon !== !1 && n.push("icon"), `${t}?${n.join("&")}`;
}
function S(e, t) {
  if (!e?.contentWindow || !e?.src) return;
  const n = E(e.src);
  n && e.contentWindow.postMessage(t, n);
}
function E(e) {
  try {
    return new URL(e).origin;
  } catch (t) {
    return console.error("Error parsing iframe origin", t), "";
  }
}
function h(e) {
  if (e instanceof HTMLIFrameElement)
    throw new o(
      "target cannot be an iframe element. Pass a container element or CSS selector instead."
    );
  s();
  let t;
  if (typeof e == "string") {
    const r = document.querySelector(e);
    if (!r)
      throw new o(`No element found for selector "${e}".`);
    if (r instanceof HTMLIFrameElement)
      throw new o(
        `Selector "${e}" resolved to an iframe element. Pass a container element or CSS selector instead.`
      );
    t = r;
  } else
    t = e;
  const n = document.createElement("iframe");
  return t.appendChild(n), n;
}
const m = {
  lg: 1,
  // 1em top + 1em bottom
  md: 0.75,
  // 0.75em top + 0.75em bottom
  sm: 0.5,
  // 0.5em top + 0.5em bottom
  xs: 0.25
  // 0.25em top + 0.25em bottom
};
function _(e) {
  const t = e?.buttonSize || i.buttonSize, n = e?.buttonFontSize || i.buttonFontSize, r = 1.5, c = 4, a = m[t] * 2 + r;
  return Math.ceil(a * n + c);
}
const b = (e) => typeof e == "object" && e !== null && "type" in e && typeof e.type == "string" && !!f[e.type.replace(
  u,
  ""
)];
export {
  p as buildKeyExportIframeUrl,
  _ as calculateSecureIframeHeight,
  h as createIframe,
  E as getSecureIframeOrigin,
  b as isSecureIframeKeyExportMessage,
  S as sendSecureIframeMessage,
  s as validateSecureIframeEnvironment
};
