// @coinbase/cdp-core/dist/esm/index.web116.js
import { handleChainId as t } from "./index.web148.js";
import { handleDisconnect as e } from "./index.web149.js";
import { handleSwitchEthereumChain as a } from "./index.web150.js";
import { handleAccounts as n } from "./index.web151.js";
import { handleEthSign as o } from "./index.web152.js";
import { handleGetCallsStatus as r } from "./index.web153.js";
import { handleGetCapabilities as i } from "./index.web154.js";
import { handlePersonalSign as l } from "./index.web155.js";
import { handleRequestAccounts as h } from "./index.web156.js";
import { handleSendCalls as m } from "./index.web157.js";
import { handleSendTransaction as s } from "./index.web158.js";
import { handleSignTypedData as d } from "./index.web159.js";
function E() {
  return {
    eth_accounts: n,
    eth_requestAccounts: h,
    personal_sign: l,
    eth_sendTransaction: s,
    eth_signTypedData_v4: d,
    wallet_disconnect: e,
    wallet_switchEthereumChain: a,
    eth_sign: o,
    eth_chainId: t,
    wallet_sendCalls: m,
    wallet_getCallsStatus: r,
    wallet_getCapabilities: i
  };
}
export {
  E as createSmartAccountHandlers
};
