// @coinbase/cdp-core/dist/esm/index.web.js
import { Analytics as r } from "./index.web2.js";
import "./index.web3.js";
import { setPlatformServices as e } from "./index.web4.js";
import { webCrypto as o } from "./index.web5.js";
import { webOAuth as t } from "./index.web6.js";
import { webStorage as n } from "./index.web7.js";
import { VERSION as i } from "./index.web8.js";
import { authenticateWithJWT as g, exportEvmAccount as A, exportSolanaAccount as d, getAccessToken as T, getAccessTokenExpiration as I, getCurrentUser as h, getUserOperation as _, initialize as v, isSignedIn as P, onAuthStateChange as M, onOAuthStateChange as R, sendEvmTransaction as y, sendSolanaTransaction as O, sendUserOperation as C, signEvmHash as U, signEvmMessage as k, signEvmTransaction as V, signEvmTypedData as W, signInWithEmail as F, signInWithOAuth as b, signInWithSms as D, signOut as N, signSolanaMessage as w, signSolanaTransaction as H, verifyEmailOTP as K, verifyOAuth as Y, verifySmsOTP as B } from "./index.web9.js";
import { createSpendPermission as L } from "./index.web10.js";
import { listSpendPermissions as G } from "./index.web11.js";
import { APIError as j, ErrorType as q, HttpErrorType as Q, OAuth2ProviderType as Z, SendEvmTransactionWithEndUserAccountBodyNetwork as $, SendSolanaTransactionWithEndUserAccountBodyNetwork as rr } from "@coinbase/cdp-api-client";
import { revokeSpendPermission as or } from "./index.web12.js";
import { getCurrentUserSync as nr } from "./index.web13.js";
import { fetchWithX402 as ar } from "./index.web14.js";
import { linkGoogle as Er } from "./index.web15.js";
import { linkSms as sr } from "./index.web16.js";
import { linkEmail as cr } from "./index.web17.js";
import { linkOAuth as xr } from "./index.web18.js";
import { linkApple as ur } from "./index.web19.js";
import { createEvmEoaAccount as Ar } from "./index.web20.js";
import { createEvmSmartAccount as Tr } from "./index.web21.js";
import { createSolanaAccount as hr } from "./index.web22.js";
import { getMfaConfig as vr } from "./index.web23.js";
import { recordMfaEnrollmentPrompted as Mr } from "./index.web24.js";
import { initiateMfaEnrollment as yr } from "./index.web25.js";
import { submitMfaEnrollment as Cr } from "./index.web26.js";
import { initiateMfaVerification as kr } from "./index.web27.js";
import { submitMfaVerification as Wr } from "./index.web28.js";
import { AuthManager as br } from "./index.web29.js";
import { CustomAuthError as Nr } from "./index.web30.js";
import { toViemAccount as Hr } from "./index.web31.js";
import { InputValidationError as Yr, InternalError as Br, ModuleResolutionError as Xr, ValidationError as Lr } from "./index.web32.js";
import { createCDPEmbeddedWallet as Gr } from "./index.web33.js";
import { EIP1193ProviderError as jr, STANDARD_ERROR_CODES as qr } from "./index.web34.js";
import "viem";
import "zustand";
import { buildKeyExportIframeUrl as Zr, calculateSecureIframeHeight as $r, createIframe as re, getSecureIframeOrigin as ee, isSecureIframeKeyExportMessage as oe, sendSecureIframeMessage as te, validateSecureIframeEnvironment as ne } from "./index.web35.js";
import { DEFAULT_SECURE_IFRAME_BASE_PATH as ae, DEFAULT_SECURE_IFRAME_THEME as me, SECURE_IFRAME_EVENT_TYPE as Ee, SECURE_IFRAME_EVENT_TYPE_PREFIX as pe, SECURE_IFRAME_KEY_EXPORT_EVENT_TYPE as se } from "./index.web36.js";
import { createEvmKeyExportIframe as ce, createSolanaKeyExportIframe as le } from "./index.web37.js";
r.registerPackageVersion("core", i);
e({
  crypto: o,
  storage: n,
  oauth: t
});
export {
  j as APIError,
  br as AuthManager,
  Nr as CustomAuthError,
  ae as DEFAULT_SECURE_IFRAME_BASE_PATH,
  me as DEFAULT_SECURE_IFRAME_THEME,
  jr as EIP1193ProviderError,
  q as ErrorType,
  Q as HttpErrorType,
  Yr as InputValidationError,
  Br as InternalError,
  Xr as ModuleResolutionError,
  Z as OAuth2ProviderType,
  Ee as SECURE_IFRAME_EVENT_TYPE,
  pe as SECURE_IFRAME_EVENT_TYPE_PREFIX,
  se as SECURE_IFRAME_KEY_EXPORT_EVENT_TYPE,
  qr as STANDARD_ERROR_CODES,
  $ as SendEvmTransactionWithEndUserAccountBodyNetwork,
  rr as SendSolanaTransactionWithEndUserAccountBodyNetwork,
  Lr as ValidationError,
  g as authenticateWithJWT,
  Zr as buildKeyExportIframeUrl,
  $r as calculateSecureIframeHeight,
  Gr as createCDPEmbeddedWallet,
  Ar as createEvmEoaAccount,
  ce as createEvmKeyExportIframe,
  Tr as createEvmSmartAccount,
  re as createIframe,
  hr as createSolanaAccount,
  le as createSolanaKeyExportIframe,
  L as createSpendPermission,
  A as exportEvmAccount,
  d as exportSolanaAccount,
  ar as fetchWithX402,
  T as getAccessToken,
  I as getAccessTokenExpiration,
  h as getCurrentUser,
  nr as getCurrentUserSync,
  vr as getMfaConfig,
  ee as getSecureIframeOrigin,
  _ as getUserOperation,
  v as initialize,
  yr as initiateMfaEnrollment,
  kr as initiateMfaVerification,
  oe as isSecureIframeKeyExportMessage,
  P as isSignedIn,
  ur as linkApple,
  cr as linkEmail,
  Er as linkGoogle,
  xr as linkOAuth,
  sr as linkSms,
  G as listSpendPermissions,
  M as onAuthStateChange,
  R as onOAuthStateChange,
  Mr as recordMfaEnrollmentPrompted,
  or as revokeSpendPermission,
  y as sendEvmTransaction,
  te as sendSecureIframeMessage,
  O as sendSolanaTransaction,
  C as sendUserOperation,
  U as signEvmHash,
  k as signEvmMessage,
  V as signEvmTransaction,
  W as signEvmTypedData,
  F as signInWithEmail,
  b as signInWithOAuth,
  D as signInWithSms,
  N as signOut,
  w as signSolanaMessage,
  H as signSolanaTransaction,
  Cr as submitMfaEnrollment,
  Wr as submitMfaVerification,
  Hr as toViemAccount,
  ne as validateSecureIframeEnvironment,
  K as verifyEmailOTP,
  Y as verifyOAuth,
  B as verifySmsOTP
};
