// @coinbase/cdp-react/dist/components/SendSolanaTransactionButton/index.js
import { jsx as _ } from "react/jsx-runtime";
import { useSendSolanaTransaction as b, APIError as T } from "@coinbase/cdp-hooks";
import { u as w } from "../../chunks/useSendComponentCallOnce.BHZMuo6E.js";
import { useState as x, useCallback as y } from "react";
import { c as B } from "../../chunks/lite.1fxw3LjI.js";
import { Button as C } from "../ui/Button/index.js";
import { getMessageFromUnknownError as E } from "../../utils/getMessageFromUnknownError.js";
import '../../assets/SendSolanaTransactionButton.css';const P = {
  "send-solana-transaction-button": "SendSolanaTransactionButton-module__send-solana-transaction-button___OxwvB"
}, D = ({
  account: t,
  children: u,
  className: f = "",
  network: s,
  onClick: e,
  onError: r,
  onSuccess: i,
  pendingLabel: p,
  transaction: o,
  ...l
}) => {
  w("send_solana_transaction_button");
  const [S, c] = x(!1), { sendSolanaTransaction: m } = b(), g = y(
    async (d) => {
      d.preventDefault(), e?.(d);
      try {
        if (!t || !s || !o) {
          const n = [];
          throw t || n.push("account"), s || n.push("network"), o || n.push("transaction"), new Error(`Missing required parameters: ${n.join(", ")}`);
        }
        c(!0);
        const { transactionSignature: a } = await m({
          transaction: o,
          solanaAccount: t,
          network: s
        });
        i?.(a);
      } catch (a) {
        let n;
        if (a instanceof T || a instanceof Error)
          n = a;
        else {
          const h = E(a);
          n = new Error(h);
        }
        r?.(n);
      } finally {
        c(!1);
      }
    },
    [t, s, e, r, i, m, o]
  );
  return /* @__PURE__ */ _(
    C,
    {
      className: l.asChild ? "" : B(P["send-solana-transaction-button"], f),
      isPending: S,
      pendingLabel: p || "Sending transaction...",
      onClick: g,
      ...l,
      children: u || "Send transaction"
    }
  );
};
export {
  D as SendSolanaTransactionButton
};
