// @coinbase/cdp-react/dist/components/ui/SwitchTransition/index.js
import { jsx as b, Fragment as H } from "react/jsx-runtime";
import { useRef as R, useId as J, useState as _, useImperativeHandle as P, useLayoutEffect as j, useEffect as Q } from "react";
import { useTransitionMap as U } from "../../../hooks/useTransitionMap.js";
import '../../../assets/SwitchTransition.css';const V = {
  "transition-wrapper": "SwitchTransition-module__transition-wrapper___jrI35"
}, W = (v) => {
  const {
    children: n,
    index: r,
    itemKey: i,
    initialEntered: u,
    setItem: f,
    deleteItem: p,
    stateMap: y,
    transitionName: I,
    parentKey: m
  } = v;
  Q(() => (f(i, { initialEntered: u }), () => {
    p(i);
  }), [i, u, f, p]);
  const { isEnter: h, isMounted: w, isResolved: x, status: g } = y.get(i) || {};
  return w ? n({
    "data-index": r,
    inert: !h,
    itemKey: i,
    "data-transition": I,
    "data-status": g,
    "data-is-enter": h,
    "data-is-resolved": x,
    "data-parent": m
  }) : null;
}, k = ({
  allowMultiple: v,
  animateHeight: n,
  autoFocus: r,
  initialEntered: i,
  as: u = "div",
  children: f,
  className: p = "",
  enter: y,
  exit: I,
  items: m,
  mountOnEnter: h,
  onStateChange: w,
  preEnter: x,
  preExit: g,
  style: q,
  timeout: e = 200,
  transitionName: $,
  transitionRef: z,
  unmountOnExit: L,
  ...E
}) => {
  const N = R(m[0]), O = J(), o = E.id || O, [S, A] = _(null), [C, K] = _(null), T = R(null), a = U({
    allowMultiple: v,
    initialEntered: !1,
    mountOnEnter: h,
    unmountOnExit: L,
    preEnter: x,
    preExit: g,
    enter: y,
    exit: I,
    timeout: e,
    onStateChange: w
  });
  P(z, () => ({
    transition: n || r ? {
      ...a,
      toggle: (t) => {
        a.toggle(t), A(t);
      }
    } : a
  })), j(() => {
    if (!n)
      return;
    let t = null;
    const s = T.current;
    if (!s)
      return;
    const d = setTimeout(() => {
      const c = s.querySelector(
        `[data-is-enter="true"][data-parent="${o}"]`
      );
      if (!c)
        return;
      const l = c?.scrollHeight ?? null;
      l !== null && K(l), t = new ResizeObserver((B) => {
        for (const D of B) {
          const G = D.contentRect.height;
          K(G);
        }
      }), t.observe(c);
    }, 0);
    return () => {
      d && clearTimeout(d), t?.disconnect();
    };
  }, [n, S, o]), j(() => {
    if (!r)
      return;
    const t = typeof e == "number" ? e : e.enter;
    if (t === void 0)
      return;
    const s = setTimeout(() => {
      const d = T.current?.querySelector(
        `[data-is-enter="true"][data-parent="${o}"]`
      );
      if (!d)
        return;
      const c = typeof r == "string" ? r : 'input, textarea, select, button, [tabindex]:not([tabindex="-1"])', l = d.querySelector(c);
      l && l.focus();
    }, t);
    return () => {
      s && clearTimeout(s);
    };
  }, [r, S, o, e]);
  const M = /* @__PURE__ */ b(H, { children: m.map((t, s) => /* @__PURE__ */ b(
    W,
    {
      index: s,
      itemKey: t,
      setItem: a.setItem,
      deleteItem: a.deleteItem,
      stateMap: a.stateMap,
      initialEntered: i ? N.current === t : void 0,
      transitionName: $,
      parentKey: o,
      children: f
    },
    t
  )) });
  return /* @__PURE__ */ b(
    u,
    {
      ...E,
      id: o,
      ref: T,
      "data-transition": $ || void 0,
      "data-animate-height": n,
      className: `${V["transition-wrapper"]} ${p}`,
      style: {
        ...q,
        "--cdp-web-transition-enter-timeout": typeof e == "object" ? `${e.enter ?? 200}ms` : void 0,
        "--cdp-web-transition-exit-timeout": typeof e == "object" ? `${e.exit ?? 200}ms` : void 0,
        "--cdp-web-transition-timeout": typeof e == "number" ? `${e}ms` : void 0,
        "--cdp-web-transition-height": n && C ? `${C}px` : void 0
      },
      children: n ? /* @__PURE__ */ b("div", { children: M }) : M
    }
  );
};
export {
  k as SwitchTransition
};
