// @coinbase/cdp-react/dist/components/forms/PhoneNumberForm/index.js
import { jsxs as u, jsx as e } from "react/jsx-runtime";
import "@coinbase/cdp-hooks";
import { Form as I, FormSubmit as S } from "@radix-ui/react-form";
import { useMemo as C, Fragment as E } from "react";
import { Field as x } from "../Field/index.js";
import { PhoneNumberInput as y } from "../PhoneNumberInput/index.js";
import { Button as M } from "../../ui/Button/index.js";
import { ServerError as U } from "../../ui/ServerError/index.js";
import { VisuallyHidden as w } from "../../ui/VisuallyHidden/index.js";
import { usePhoneNumberValidators as T } from "../../../hooks/usePhoneNumberValidators.js";
import '../../../assets/PhoneNumberForm.css';const j = "PhoneNumberForm-module__form___ExO8G", m = {
  form: j,
  "server-error": "PhoneNumberForm-module__server-error___AQcMg",
  "phone-number-fieldset": "PhoneNumberForm-module__phone-number-fieldset___0TsSH"
}, q = ({ children: n }) => /* @__PURE__ */ u("fieldset", { className: m["phone-number-fieldset"], children: [
  /* @__PURE__ */ e("legend", { children: /* @__PURE__ */ e(w, { children: "International phone number" }) }),
  n
] }), D = ({
  className: n = "",
  countryCode: t = "US",
  countryOptions: o = !0,
  error: r,
  isPending: h,
  onCountryCodeChange: a,
  onPhoneNumberChange: d,
  phoneNumber: p,
  submitLabel: b = "Continue",
  ...v
}) => {
  const {
    isInvalidCountry: f,
    isInvalidLength: c,
    isInvalidNumber: _,
    isInvalidTooShort: P,
    isInvalidTooLong: N,
    isInvalidPhoneNumberError: s
  } = T({ countryCode: t }), i = C(() => r && s(r) ? "Use a valid phone number" : "", [r, s]), g = !!r && !i, F = (typeof o == "boolean" ? o : o.length > 0) ? q : E;
  return /* @__PURE__ */ u(I, { className: `${m.form} ${n}`, ...v, children: [
    /* @__PURE__ */ e(F, { children: /* @__PURE__ */ e(
      x,
      {
        label: "Phone number",
        name: "phone",
        inlineError: i,
        validators: [
          ["valueMissing", "Phone number is required"],
          ["typeMismatch", "Use a valid phone number"],
          [_, "Use a valid phone number"],
          [f, "Country is not supported"],
          [P, "Phone number is too short"],
          [N, "Phone number is too long"],
          [c, "Phone number is the wrong length"]
        ],
        children: (l) => /* @__PURE__ */ e(
          y,
          {
            "aria-invalid": !(l === void 0 || l.valid) || !!i,
            autoComplete: "tel-national",
            countryOptions: o,
            phoneNumber: p,
            onPhoneNumberChange: d,
            countryCode: t,
            onCountryCodeChange: a,
            required: !0
          }
        )
      }
    ) }),
    g && /* @__PURE__ */ e(U, { error: r, className: m["server-error"] }),
    /* @__PURE__ */ e(S, { asChild: !0, children: /* @__PURE__ */ e(M, { type: "submit", isPending: h, children: b }) })
  ] });
};
export {
  D as PhoneNumberForm
};
