// @coinbase/cdp-react/dist/components/Fund/hooks/useBuyUrl.js
import { useCallback as y, useMemo as R } from "react";
const _ = (o) => {
  try {
    return new URL(o), !0;
  } catch {
    return !1;
  }
}, h = ({
  country: o,
  cryptoCurrency: a,
  destinationAddress: c,
  fetchBuyQuote: u,
  fiatCurrency: p,
  network: d,
  onError: n,
  redirectUrl: s,
  subdivision: i
}) => {
  const U = y(
    async ({ paymentMethod: l, paymentAmount: m }, f) => {
      if (isNaN(m) || !l)
        return "";
      try {
        const e = (await u({
          purchaseCurrency: a.toUpperCase(),
          purchaseNetwork: d.toLowerCase(),
          paymentCurrency: p.toUpperCase(),
          paymentAmount: m.toFixed(),
          paymentMethod: l.replace("_GUEST", ""),
          // remove "_GUEST" from the payment method since it is not valid in the Buy Quote API
          country: o.toUpperCase(),
          subdivision: i?.toUpperCase(),
          destinationAddress: c
        })).onrampUrl;
        if (!e) {
          const r = {
            errorType: "handled_error",
            code: "NO_ONRAMP_URL",
            debugMessage: "No payment methods found for the selected country and currency. See docs for more information: https://docs.cdp.coinbase.com/onramp/docs/api-configurations"
          };
          return f?.(r), n?.(r), "";
        }
        if (s && _(s)) {
          const r = new URL(e);
          return r.searchParams.set("redirectUrl", s), r.toString();
        }
        return e;
      } catch (t) {
        let e;
        return t instanceof Error ? e = {
          errorType: "handled_error",
          code: "BUY_URL_ERROR",
          debugMessage: t.message
        } : e = {
          errorType: "unknown_error",
          code: "UNKNOWN_BUY_URL_ERROR",
          debugMessage: JSON.stringify(t)
        }, f?.(e), n?.(e), "";
      }
    },
    [
      o,
      a,
      c,
      u,
      p,
      d,
      n,
      s,
      i
    ]
  );
  return R(() => ({ fetchBuyUrl: U }), [U]);
};
export {
  h as useBuyUrl
};
