// @coinbase/cdp-react/dist/components/Fund/hooks/useExchangeRate.js
import { Big as _ } from "big.js";
import { useCallback as m, useMemo as i } from "react";
const N = ({
  country: t,
  cryptoCurrency: a,
  fetchBuyQuote: s,
  fiatCurrency: n,
  network: p,
  onError: c,
  dispatch: o,
  subdivision: u
}) => {
  const E = m(async () => {
    const C = a.toUpperCase(), R = n.toUpperCase(), d = p.toLowerCase(), g = t.toUpperCase(), y = u?.toUpperCase();
    o({ type: "FETCH_EXCHANGE_RATE" });
    try {
      const r = await s({
        purchaseCurrency: C,
        paymentCurrency: R,
        paymentAmount: "100",
        paymentMethod: "CARD",
        country: g,
        subdivision: y,
        purchaseNetwork: d,
        destinationAddress: ""
      }), e = _(r.purchaseAmount.value).div(r.paymentSubtotal.value).toNumber();
      o({ type: "SET_EXCHANGE_RATE_SUCCESS", payload: { exchangeRate: e } });
    } catch (r) {
      let e;
      r instanceof Error ? e = {
        errorType: "handled_error",
        code: "EXCHANGE_RATE_ERROR",
        debugMessage: r.message
      } : e = {
        errorType: "unknown_error",
        code: "UNKNOWN_EXCHANGE_RATE_ERROR",
        debugMessage: JSON.stringify(r)
      }, c?.(e), o({
        type: "SET_EXCHANGE_RATE_ERROR",
        payload: {
          error: {
            code: e?.code,
            message: e.errorType === "handled_error" ? e?.debugMessage : "Something went wrong fetching the exchange rate"
          }
        }
      });
    }
  }, [
    t,
    a,
    o,
    s,
    n,
    p,
    c,
    u
  ]);
  return i(() => ({ fetchExchangeRate: E }), [E]);
};
export {
  N as useExchangeRate
};
