// @coinbase/cdp-react/dist/components/Fund/hooks/useSetupOnrampEventListeners.js
import { useCallback as n, useEffect as o } from "react";
import { useFundContext as i } from "../FundProvider.js";
import { setupOnrampEventListeners as c } from "../utils/setupOnrampEventListeners.js";
const s = (t) => ({
  type: "SET_TRANSACTION_STATUS",
  payload: {
    transactionStatus: t
  }
}), S = () => {
  const { dispatch: t } = i(), a = n(
    (e) => {
      e.eventName === "transition_view" ? e.pageRoute.endsWith("/order-submitted") ? t(
        s({
          statusName: "transactionSubmitted",
          statusData: null
        })
      ) : t(
        s({
          statusName: "transactionPending",
          statusData: null
        })
      ) : e.eventName === "error" && t(
        s({
          statusName: "error",
          statusData: e.error
        })
      );
    },
    [t]
  ), r = n(
    (e) => {
      t(
        s({
          statusName: "transactionSuccess",
          statusData: e || null
        })
      );
    },
    [t]
  ), u = n(() => {
    t(
      s({
        statusName: "exit",
        statusData: null
      })
    );
  }, [t]);
  o(() => {
    const e = c({
      onEvent: a,
      onExit: u,
      onSuccess: r
    });
    return () => {
      e();
    };
  }, []);
};
export {
  S as useSetupOnrampEventListeners
};
