// @coinbase/cdp-react/dist/components/forms/Select/index.js
import { jsx as e, jsxs as i } from "react/jsx-runtime";
import { SelectTrigger as u, SelectValue as p, SelectIcon as h, SelectPortal as f, SelectContent as g, SelectScrollUpButton as C, SelectViewport as I, SelectScrollDownButton as N, SelectItem as $, SelectItemText as y, SelectItemIndicator as T, Select as w } from "@radix-ui/react-select";
import { forwardRef as a } from "react";
import { useTheme as x } from "../../ThemeProvider/index.js";
import { ButtonBase as B } from "../../ui/ButtonBase/index.js";
import { IconCheck as V } from "../../../icons/IconCheck.js";
import { IconChevronDown as m } from "../../../icons/IconChevronDown.js";
import "../../../theme/theme.js";
import '../../../assets/Select.css';const o = {
  "select-trigger": "Select-module__select-trigger___yqioM",
  "select-icon": "Select-module__select-icon___hE7Wy",
  "select-content": "Select-module__select-content___6R8J9",
  "select-item": "Select-module__select-item___12l2C",
  "rotate-180": "Select-module__rotate-180___KuvGV"
}, K = ({ children: l, ...t }) => /* @__PURE__ */ e(w, { ...t, children: l }), b = a(
  ({ children: l, className: t = "", placeholder: c = "Select an option", size: r = "md", ...s }, n) => /* @__PURE__ */ e(u, { ...s, asChild: !0, children: l || /* @__PURE__ */ i(
    B,
    {
      className: `${o["select-trigger"]} ${t}`,
      fullWidth: !0,
      ref: n,
      size: r,
      children: [
        /* @__PURE__ */ e(p, { placeholder: c }),
        /* @__PURE__ */ e(h, { asChild: !0, children: /* @__PURE__ */ e(m, { className: o["select-icon"] }) })
      ]
    }
  ) })
);
b.displayName = "SelectTrigger";
const j = a(
  ({
    children: l,
    className: t = "",
    style: c,
    position: r = "popper",
    side: s = "bottom",
    sideOffset: n = 8,
    ..._
  }, d) => {
    const { cssVariables: S } = x();
    return /* @__PURE__ */ e(f, { children: /* @__PURE__ */ i(
      g,
      {
        position: r,
        side: s,
        sideOffset: n,
        className: `${o["select-content"]} ${t}`,
        ref: d,
        style: c ? { ...S, ...c } : S,
        ..._,
        children: [
          /* @__PURE__ */ e(C, { children: /* @__PURE__ */ e(m, { className: o["rotate-180"] }) }),
          /* @__PURE__ */ e(I, { children: l }),
          /* @__PURE__ */ e(N, { children: /* @__PURE__ */ e(m, {}) })
        ]
      }
    ) });
  }
);
j.displayName = "SelectContent";
const v = a(
  ({ children: l, indicator: t, ...c }, r) => /* @__PURE__ */ i($, { ...c, ref: r, className: o["select-item"], children: [
    /* @__PURE__ */ e(y, { children: l }),
    t && /* @__PURE__ */ e(T, { asChild: !0, children: typeof t == "boolean" ? /* @__PURE__ */ e(V, {}) : t })
  ] })
);
v.displayName = "SelectItem";
export {
  K as Select,
  j as SelectContent,
  v as SelectItem,
  b as SelectTrigger
};
