// @coinbase/cdp-react/dist/components/FundModal/index.js
import { jsx as o, jsxs as i, Fragment as p } from "react/jsx-runtime";
import { u as w } from "../../chunks/useSendComponentCallOnce.BHZMuo6E.js";
import { createContext as j, useState as A, useCallback as F, useMemo as S, useId as $, useContext as z } from "react";
import { c as x } from "../../chunks/lite.1fxw3LjI.js";
import { useAppConfig as V } from "../CDPReactProvider/index.js";
import { Fund as k, FundFooter as D } from "../Fund/index.js";
import { Button as v } from "../ui/Button/index.js";
import { Modal as H, ModalContent as L, ModalTitle as R, ModalClose as U, ModalTrigger as W } from "../ui/Modal/index.js";
import { VisuallyHidden as C } from "../ui/VisuallyHidden/index.js";
import { IconXMark as X } from "../../icons/IconXMark.js";
import { childrenHasComponent as g } from "../../utils/childrenHasComponent.js";
import { FundTitle as _ } from "../Fund/FundTitle.js";
import { FundForm as q } from "../Fund/FundForm.js";
import '../../assets/FundModal.css';const B = "FundModal-module__trigger___2IuXj", E = "FundModal-module__fund___1AmQa", K = "FundModal-module__title___lNsoT", P = "FundModal-module__content___9f3ze", t = {
  trigger: B,
  fund: E,
  "no-footer": "FundModal-module__no-footer___WKV78",
  "title-buttons": "FundModal-module__title-buttons___Lctf7",
  "close-button": "FundModal-module__close-button___jzdFM",
  "close-icon": "FundModal-module__close-icon___WoA4a",
  "fund-inner": "FundModal-module__fund-inner___UuOq3",
  "tx-status": "FundModal-module__tx-status___fc-3f",
  title: K,
  content: P,
  "error-view": "FundModal-module__error-view___hR1A0"
}, N = j(null), y = () => {
  const n = z(N);
  if (!n)
    throw new Error("useFundModalContext must be used within a FundModal");
  return n;
}, b = ({ children: n, className: l = "", label: e, ...s }) => {
  const { cryptoCurrency: r } = y();
  return /* @__PURE__ */ o(W, { asChild: !0, children: n || /* @__PURE__ */ o(v, { variant: "primary", className: x(t.trigger, l), ...s, children: e || `Deposit ${r.toUpperCase()}` }) });
}, f = (n) => {
  const { showCoinbaseFooter: l } = V(), e = $(), { unmount: s, ...r } = y(), { openIn: m, submitLabel: u, title: a } = r;
  return /* @__PURE__ */ o(L, { "aria-describedby": void 0, ...n, children: /* @__PURE__ */ o(
    k,
    {
      className: x(t.fund, l ? "" : t["no-footer"]),
      ...r,
      children: /* @__PURE__ */ i(p, { children: [
        /* @__PURE__ */ o(C, { children: /* @__PURE__ */ o(R, { asChild: !0, children: /* @__PURE__ */ o(_, { as: "span", children: a }) }) }),
        /* @__PURE__ */ o(
          q,
          {
            "aria-labelledby": e,
            openIn: m,
            unmountOnTransactionSuccess: !0,
            unmount: s,
            submitLabel: u,
            children: ({ view: d, Content: c }) => /* @__PURE__ */ i(
              "div",
              {
                className: `${t["fund-inner"]} ${d === "transaction-status" ? t["tx-status"] : ""} ${d === "error" ? t["error-view"] : ""}`,
                children: [
                  /* @__PURE__ */ i("div", { className: t["title-buttons"], children: [
                    d === "error" ? /* @__PURE__ */ o(C, { children: /* @__PURE__ */ o(_, { id: e, children: a }) }) : /* @__PURE__ */ o(_, { className: t.title, id: e, children: a }),
                    /* @__PURE__ */ o(U, { asChild: !0, children: /* @__PURE__ */ o(
                      v,
                      {
                        className: t["close-button"],
                        "aria-label": "Close",
                        size: "md",
                        variant: "transparentSecondary",
                        children: /* @__PURE__ */ o(X, { className: t["close-icon"] })
                      }
                    ) })
                  ] }),
                  /* @__PURE__ */ o("div", { className: t.content, children: c })
                ]
              }
            )
          }
        ),
        l && /* @__PURE__ */ o(D, {})
      ] })
    }
  ) });
}, io = ({ children: n, open: l, setIsOpen: e, ...s }) => {
  w("fund_modal");
  const r = l !== void 0, [m, u] = A(!1), a = r ? l : m, d = F(
    (h) => r ? e?.(h) : u(h),
    [r, e, u]
  ), c = F(() => {
    d(!1);
  }, [d]), O = S(
    () => ({
      unmount: c,
      ...s
    }),
    [c, s]
  ), T = n ? g(n, b) : !1, M = n ? g(n, f) : !1, I = !T && !M;
  return /* @__PURE__ */ o(N.Provider, { value: O, children: /* @__PURE__ */ o(H, { open: a, onOpenChange: d, children: I ? /* @__PURE__ */ i(p, { children: [
    /* @__PURE__ */ o(b, { children: n }),
    /* @__PURE__ */ o(f, {})
  ] }) : /* @__PURE__ */ i(p, { children: [
    n,
    !M && /* @__PURE__ */ o(f, {})
  ] }) }) });
};
export {
  io as FundModal,
  f as FundModalContent,
  b as FundModalTrigger
};
