// @coinbase/cdp-react/dist/components/CopyEvmKeyButton/index.js
import { jsx as B } from "react/jsx-runtime";
import { useConfig as S } from "@coinbase/cdp-hooks";
import { u as _ } from "../../chunks/useSendComponentCallOnce.BHZMuo6E.js";
import { useRef as j, useState as i, useCallback as E } from "react";
import "../ui/Button/index.js";
import { IframeButton as K } from "../ui/IframeButton/index.js";
import { useKeyExportPostMessage as T } from "../../hooks/useKeyExportPostMessage.js";
import "@coinbase/cdp-core";
const F = ({
  address: p,
  className: n = "",
  copiedLabel: c = "Copied",
  icon: e = !0,
  fullWidth: u,
  label: t = "Copy key",
  onReady: r,
  onSessionExpired: o,
  onSuccess: s,
  onError: a,
  size: f,
  theme: d,
  variant: l
}) => {
  _("copy_evm_key_button");
  const {
    config: { projectId: y, secureIframeBasePath: b }
  } = S(), m = j(null), [h, k] = i(void 0), [C, g] = i(!1), v = E(
    (P, w) => {
      switch (P) {
        case "ready":
          r?.();
          break;
        case "success":
          s?.();
          break;
        case "error":
          a?.(w);
          break;
        case "expired":
          g(!0), o?.();
          break;
      }
    },
    [r, s, a, o]
  ), { iframeUrl: x, isPending: I } = T({
    address: p,
    type: "evm",
    projectId: y,
    label: t,
    basePath: b,
    copiedLabel: c,
    icon: e,
    iframeRef: m,
    onStatusUpdate: v,
    theme: h
  });
  return C ? null : /* @__PURE__ */ B(
    K,
    {
      "data-testid": "copy-evm-key-button",
      ref: m,
      src: x,
      allow: "clipboard-read; clipboard-write",
      isPending: I,
      label: t,
      icon: e,
      fullWidth: u,
      onThemeChange: k,
      size: f,
      theme: d,
      variant: l,
      className: n
    }
  );
};
export {
  F as CopyEvmKeyButton
};
