// @coinbase/cdp-react/dist/components/LinkAuthModal/index.js
import { jsx as o, jsxs as n, Fragment as d } from "react/jsx-runtime";
import { u as p } from "../../chunks/useSendComponentCallOnce.BHZMuo6E.js";
import { createContext as f, useMemo as L, useContext as g } from "react";
import { c as A } from "../../chunks/lite.1fxw3LjI.js";
import "../CDPReactProvider/index.js";
import { LinkAuth as C, LinkAuthError as x } from "../LinkAuth/index.js";
import { Button as m } from "../ui/Button/index.js";
import { Modal as b, ModalContent as v, ModalTitle as w, ModalClose as N, ModalTrigger as T } from "../ui/Modal/index.js";
import { VisuallyHidden as y } from "../ui/VisuallyHidden/index.js";
import { IconXMark as F } from "../../icons/IconXMark.js";
import { childrenHasComponent as c } from "../../utils/childrenHasComponent.js";
import { LinkAuthTitle as H } from "../LinkAuth/LinkAuthTitle.js";
import { LinkAuthFlowBackButton as j } from "../LinkAuth/LinkAuthFlowBackButton.js";
import { a as B } from "../../chunks/LinkAuthFlow.3HN888DV.js";
import '../../assets/LinkAuthModal.css';const S = "LinkAuthModal-module__header___qoco6", D = "LinkAuthModal-module__error___Srl7H", E = "LinkAuthModal-module__trigger___aHDfG", r = {
  header: S,
  error: D,
  "close-icon": "LinkAuthModal-module__close-icon___Fe02C",
  "close-button": "LinkAuthModal-module__close-button___x6j3c",
  trigger: E
}, h = f(null), O = () => {
  const e = g(h);
  if (!e)
    throw new Error("useLinkAuthModalContext must be used within a LinkAuthModal");
  return e;
}, u = ({
  children: e,
  className: t = "",
  label: i,
  ...a
}) => /* @__PURE__ */ o(T, { asChild: !0, children: e || /* @__PURE__ */ o(m, { className: A(r.trigger, t), ...a, children: i || "Link accounts" }) }), l = (e) => {
  const { onLinkSuccess: t } = O();
  return /* @__PURE__ */ n(v, { "aria-describedby": void 0, ...e, children: [
    /* @__PURE__ */ o(y, { children: /* @__PURE__ */ o(w, { children: "Link a profile" }) }),
    /* @__PURE__ */ n(C, { onLinkSuccess: t, children: [
      /* @__PURE__ */ n("div", { className: r.header, children: [
        /* @__PURE__ */ o(H, {}),
        /* @__PURE__ */ o(j, {}),
        /* @__PURE__ */ o(N, { asChild: !0, children: /* @__PURE__ */ o(
          m,
          {
            className: r["close-button"],
            "aria-label": "Close",
            size: "md",
            variant: "transparentSecondary",
            children: /* @__PURE__ */ o(F, { className: r["close-icon"] })
          }
        ) })
      ] }),
      /* @__PURE__ */ o("div", { className: r.error, children: /* @__PURE__ */ o(x, {}) }),
      /* @__PURE__ */ o(B, {})
    ] })
  ] });
}, Z = ({ children: e, open: t, setIsOpen: i, onLinkSuccess: a }) => {
  p("manage_auth_modal");
  const _ = e ? c(e, u) : !1, s = e ? c(e, l) : !1, k = !_ && !s, M = L(() => ({ onLinkSuccess: a }), [a]);
  return /* @__PURE__ */ o(h.Provider, { value: M, children: /* @__PURE__ */ o(b, { open: t, onOpenChange: i, children: k ? /* @__PURE__ */ n(d, { children: [
    /* @__PURE__ */ o(u, { children: e }),
    /* @__PURE__ */ o(l, {})
  ] }) : /* @__PURE__ */ n(d, { children: [
    e,
    !s && /* @__PURE__ */ o(l, {})
  ] }) }) });
};
export {
  Z as LinkAuthModal,
  l as LinkAuthModalContent,
  u as LinkAuthModalTrigger
};
