// @coinbase/cdp-react/dist/components/Fund/components/FundErrorScreen.js
import { jsxs as d, jsx as o, Fragment as y } from "react/jsx-runtime";
import { useMemo as i, Fragment as E } from "react";
import { IconExclamationTriangle as m } from "../../../icons/IconExclamationTriangle.js";
import { useFundContext as h } from "../FundProvider.js";
import '../../../assets/FundErrorScreen.css';const f = "FundErrorScreen-module__icon___Mf6ck", O = "FundErrorScreen-module__title___8Qv6o", g = "FundErrorScreen-module__body___-p-r9", U = "FundErrorScreen-module__keywords___u85Do", c = {
  icon: f,
  title: O,
  body: g,
  "docs-link": "FundErrorScreen-module__docs-link___mzeH8",
  keywords: U
}, C = (s) => s.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"), b = ({
  message: s,
  keywords: e,
  className: t = ""
}) => {
  if (!s || !e?.length)
    return /* @__PURE__ */ o(y, { children: s });
  const a = new RegExp(`(${e.map(C).join("|")})\\b`, "g"), r = s.split(a);
  return /* @__PURE__ */ o(y, { children: r.map((n, u) => n ? e.includes(n) ? /* @__PURE__ */ o("span", { className: t, "data-part": "keyword", children: n }, u) : /* @__PURE__ */ o(E, { children: n }, u) : null) });
}, T = ({ error: s }) => {
  const { state: e } = h(), { code: t, message: a } = s || {}, r = process.env.NODE_ENV === "development", n = e.network.charAt(0).toUpperCase() + e.network.slice(1), u = i(() => {
    switch (t) {
      case "NO_ONRAMP_URL":
      case "BUY_URL_ERROR":
      case "UNKNOWN_BUY_URL_ERROR":
        return "https://docs.cdp.coinbase.com/onramp-&-offramp/onramp-apis/generating-quotes#example-request%2Fresponse-with-one-click-buy-url";
      case "EXCHANGE_RATE_ERROR":
      case "UNKNOWN_EXCHANGE_RATE_ERROR":
        return "https://docs.cdp.coinbase.com/onramp-&-offramp/onramp-apis/generating-quotes#onramp-quote";
      default:
        return "https://docs.cdp.coinbase.com/onramp-&-offramp/onramp-apis/countries-&-currencies#fiat-currencies-and-crypto-assets-supported";
    }
  }, [t]), l = i(() => {
    if (!r || !t)
      return "";
    switch (t) {
      case "NO_ONRAMP_URL":
      case "BUY_URL_ERROR":
      case "UNKNOWN_BUY_URL_ERROR":
        return "Use the Onramp Quote API with a destination address to get a one-click-buy URL.";
      case "EXCHANGE_RATE_ERROR":
      case "UNKNOWN_EXCHANGE_RATE_ERROR":
        return "Use the Onramp Quote API to calculate the exchange rate.";
      default:
        return "Use the Buy Options API to see the full list of crypto assets that can be bought for each region your app supports.";
    }
  }, [t, r]), R = i(() => {
    switch (t) {
      case "INVALID_PURCHASE_CURRENCY":
        return r ? "Invalid parameters" : `${e.cryptoCurrency} is not available`;
      case "INVALID_NETWORK":
        return r ? "Invalid parameters" : `${n} network is not available`;
      case "NO_ONRAMP_URL":
      case "BUY_URL_ERROR":
      case "UNKNOWN_BUY_URL_ERROR":
        return r ? "Onramp URL error" : "Something went wrong";
      case "EXCHANGE_RATE_ERROR":
      case "UNKNOWN_EXCHANGE_RATE_ERROR":
        return "No exchange rate";
      default:
        return "No payment methods";
    }
  }, [t, e.cryptoCurrency, n, r]), _ = i(() => {
    switch (t) {
      case "PAYMENT_METHODS_ERROR":
      case "UNKNOWN_PAYMENT_METHODS_ERROR":
        return r ? a || "There was an error fetching the payment methods." : `Unfortunately there's no available payment method to purchase ${e.cryptoCurrency} with ${e.fiatCurrency} on ${n} in your region. Please select a different asset to buy.`;
      case "INVALID_PURCHASE_CURRENCY":
        return r ? `${e.cryptoCurrency} crypto asset is not available.` : "Please select a different asset to buy.";
      case "INVALID_NETWORK":
        return r ? `${e.network} is not available for buying ${e.cryptoCurrency} in your region.` : `Coinbase doesn’t support buying ${e.cryptoCurrency} on ${n} in your region. Please select a different network to buy ${e.cryptoCurrency} on.`;
      case "NO_ONRAMP_URL":
        return r ? "An onramp url was unable to be created" : "We’re running into some issues. Please try again later.";
      case "BUY_URL_ERROR":
      case "UNKNOWN_BUY_URL_ERROR":
        return r ? a || "There was an error creating the onramp url." : "We’re running into some issues. Please try again later.";
      case "EXCHANGE_RATE_ERROR":
      case "UNKNOWN_EXCHANGE_RATE_ERROR":
        return r ? a || "There was an error fetching the exchange rate." : `No exchange rate found for ${e.fiatCurrency}<>${e.cryptoCurrency} on ${n} in your region.`;
      default:
        return r ? `There's no available payment method to purchase ${e.cryptoCurrency} with ${e.fiatCurrency} on ${e.network} in your region.` : `Unfortunately there's no available payment method to purchase ${e.cryptoCurrency} with ${e.fiatCurrency} on ${n} in your region. Please select a different asset to buy.`;
    }
  }, [
    t,
    a,
    e.cryptoCurrency,
    e.network,
    n,
    r,
    e.fiatCurrency
  ]), N = i(() => {
    const p = [e.cryptoCurrency, e.network, e.country, e.fiatCurrency];
    return e.subdivision && p.push(e.subdivision), p;
  }, [e.cryptoCurrency, e.network, e.country, e.fiatCurrency, e.subdivision]);
  return !R || !_ ? null : /* @__PURE__ */ d("div", { children: [
    /* @__PURE__ */ o(m, { className: c.icon }),
    /* @__PURE__ */ o("p", { className: `${c.title}`, children: R }),
    /* @__PURE__ */ o("p", { className: `${c.body} ${r ? c.keywords : ""}`, children: /* @__PURE__ */ o(b, { message: _, keywords: N }) }),
    l && /* @__PURE__ */ d("p", { className: `${c.body}`, children: [
      " ",
      l,
      " ",
      /* @__PURE__ */ o(
        "a",
        {
          className: c["docs-link"],
          href: u,
          target: "_blank",
          rel: "noopener noreferrer",
          children: "View docs"
        }
      )
    ] })
  ] });
};
export {
  T as FundErrorScreen
};
