// @coinbase/cdp-react/dist/components/Fund/components/FundPaymentMethods.js
import { jsxs as l, jsx as n } from "react/jsx-runtime";
import { useCallback as r } from "react";
import { Select as _, SelectTrigger as h, SelectContent as p, SelectItem as u } from "../../forms/Select/index.js";
import { LoadingSkeleton as y } from "../../ui/LoadingSkeleton/index.js";
import { useFundContext as g } from "../FundProvider.js";
import { createSetFieldAction as P } from "../useFundReducer.js";
import '../../../assets/FundPaymentMethods.css';const s = {
  "select-loading-skeleton": "FundPaymentMethods-module__select-loading-skeleton___ZuD3n",
  "select-item": "FundPaymentMethods-module__select-item___9VyxH",
  "select-item-icon": "FundPaymentMethods-module__select-item-icon___28unQ",
  "select-item-name": "FundPaymentMethods-module__select-item-name___5vLc0",
  "select-item-description": "FundPaymentMethods-module__select-item-description___C5mTB",
  "select-trigger": "FundPaymentMethods-module__select-trigger___ypebl"
}, k = ({
  as: a = "div",
  ...c
}) => {
  const { state: e, dispatch: i } = g(), m = r(
    (t) => {
      const d = e.paymentMethods?.find((o) => o.id === t);
      d && i(P("selectedPaymentMethod", d));
    },
    [i, e.paymentMethods]
  );
  return /* @__PURE__ */ l(a, { ...c, children: [
    e.isPaymentMethodsPending && /* @__PURE__ */ n(y, { className: s["select-loading-skeleton"] }),
    !e.isPaymentMethodsPending && /* @__PURE__ */ l(
      _,
      {
        defaultValue: e.selectedPaymentMethod?.id || e.paymentMethods?.[0]?.id,
        onValueChange: m,
        children: [
          /* @__PURE__ */ n(
            h,
            {
              "aria-label": "Select a payment method",
              placeholder: "Select a payment method",
              className: s["select-trigger"]
            }
          ),
          /* @__PURE__ */ n(p, { children: e.paymentMethods?.map((t) => /* @__PURE__ */ n(u, { value: t.id, children: /* @__PURE__ */ l("span", { className: s["select-item"], children: [
            /* @__PURE__ */ n("span", { className: s["select-item-icon"], children: t.icon }),
            /* @__PURE__ */ l("span", { children: [
              /* @__PURE__ */ n("span", { className: s["select-item-name"], children: t.name }),
              /* @__PURE__ */ n("span", { className: s["select-item-description"], children: t.description })
            ] })
          ] }) }, t.id)) })
        ]
      }
    )
  ] });
};
export {
  k as FundPaymentMethods
};
