// @coinbase/cdp-react/dist/components/Fund/components/FundPresetAmountInputs.js
import { jsx as s } from "react/jsx-runtime";
import { Big as c } from "big.js";
import { useCallback as l } from "react";
import { Button as d } from "../../ui/Button/index.js";
import { formatFiat as f } from "../../../utils/formatFiat.js";
import { useFundContext as y } from "../FundProvider.js";
import '../../../assets/FundPresetAmountInputs.css';const a = {
  "preset-amount-inputs": "FundPresetAmountInputs-module__preset-amount-inputs___Skjnm",
  "preset-amount-input": "FundPresetAmountInputs-module__preset-amount-input___eyNqU"
}, P = ({
  as: r = "div",
  className: m = "",
  ...u
}) => {
  const { state: t, dispatch: o } = y(), i = l(
    (e) => {
      const n = c(e).times(t.exchangeRate ?? 0), p = n.eq(0) ? 0 : +n.toFixed(t.cryptoDecimalPlaces);
      o({
        type: "SET_AMOUNTS",
        payload: { cryptoAmount: p, fiatAmount: e }
      });
    },
    [o, t.exchangeRate, t.cryptoDecimalPlaces]
  );
  return t.presetAmountInputs?.length ? /* @__PURE__ */ s(r, { className: `${a["preset-amount-inputs"]} ${m}`, ...u, children: t.presetAmountInputs.map((e) => /* @__PURE__ */ s(
    d,
    {
      size: "xs",
      variant: "control",
      "aria-label": `Deposit ${e}`,
      className: a["preset-amount-input"],
      onClick: () => i(e),
      type: "button",
      children: f(e, t.fiatCurrency, t.fiatDecimalPlaces, t.locale)
    },
    e
  )) }) : null;
};
export {
  P as FundPresetAmountInputs
};
