// @coinbase/cdp-react/dist/components/Fund/components/FundTransactionStatus.js
import { jsx as t, jsxs as c, Fragment as e } from "react/jsx-runtime";
import { useRef as d, useCallback as h, useEffect as S, useMemo as N } from "react";
import { Button as o } from "../../ui/Button/index.js";
import { LoadingSpinner as T } from "../../ui/LoadingSpinner/index.js";
import { SwitchFadeTransition as $ } from "../../ui/SwitchFadeTransition/index.js";
import { IconCheck as g } from "../../../icons/IconCheck.js";
import { IconCurrencies as f } from "../../../icons/IconCurrencies.js";
import { IconXMark as b } from "../../../icons/IconXMark.js";
import { useFundContext as m } from "../FundProvider.js";
import '../../../assets/FundTransactionStatus.css';const w = "FundTransactionStatus-module__graphic___cam38", y = "FundTransactionStatus-module__status___2zKJu", F = "FundTransactionStatus-module__pad___-9rGO", v = "FundTransactionStatus-module__spinner___94rMk", C = "FundTransactionStatus-module__icon___0hBQO", k = "FundTransactionStatus-module__subhead___0sHIM", M = "FundTransactionStatus-module__description___W7kK2", W = "FundTransactionStatus-module__ring___1vnCu", P = "FundTransactionStatus-module__invisible___m72ZQ", s = {
  graphic: w,
  status: y,
  pad: F,
  spinner: v,
  icon: C,
  "icon-success": "FundTransactionStatus-module__icon-success___893VG",
  "icon-error": "FundTransactionStatus-module__icon-error___-aQvD",
  subhead: k,
  description: M,
  ring: W,
  "ring-success": "FundTransactionStatus-module__ring-success___Bxudq",
  "ring-error": "FundTransactionStatus-module__ring-error___WmTaE",
  invisible: P
}, x = ["waiting", "transactionSubmitted", "transactionSuccess", "error"], J = (a) => {
  const { state: n } = m(), i = d(null), l = d(null), u = h((r) => {
    l.current !== r && (l.current = r, i.current?.transition.toggle(r));
  }, []);
  return S(() => {
    (n.transactionStatus.statusName === "transactionSubmitted" || n.transactionStatus.statusName === "transactionSuccess" || n.transactionStatus.statusName === "error") && u(n.transactionStatus.statusName);
  }, [n.transactionStatus.statusName, u]), /* @__PURE__ */ t(
    $,
    {
      animateHeight: !1,
      timeout: 250,
      items: x,
      initialEntered: !0,
      transitionRef: i,
      children: ({ itemKey: r, ...p }) => {
        const _ = K[r];
        return /* @__PURE__ */ t("div", { ...p, className: s.status, children: /* @__PURE__ */ t(_, { ...a }) });
      }
    }
  );
}, G = () => {
  const { state: a } = m(), n = N(
    () => a.paymentMethods?.find((i) => i.id === a.selectedPaymentMethod?.id),
    [a.paymentMethods, a.selectedPaymentMethod?.id]
  );
  return /* @__PURE__ */ c(e, { children: [
    /* @__PURE__ */ c("div", { className: s.graphic, children: [
      /* @__PURE__ */ t(
        T,
        {
          strokeWidth: "5%",
          staticStroke: !0,
          staticStrokeColor: "var(--cdp-web-colors-line-default)",
          className: s.spinner
        }
      ),
      n && /* @__PURE__ */ t("span", { className: s.icon, children: n.icon })
    ] }),
    /* @__PURE__ */ t("p", { className: s.subhead, children: "Waiting" }),
    /* @__PURE__ */ c("p", { className: s.description, children: [
      "Purchase ",
      a.cryptoCurrency.toUpperCase(),
      " with ",
      a.fiatCurrency.toUpperCase(),
      n ? ` via ${n.name}` : ""
    ] })
  ] });
}, I = ({
  reset: a,
  timeRemaining: n,
  unmountOnSuccess: i
}) => /* @__PURE__ */ c(e, { children: [
  /* @__PURE__ */ c("div", { className: `${s.graphic} ${s.pad}`, children: [
    /* @__PURE__ */ t("span", { className: `${s.ring} ${s["ring-success"]}` }),
    /* @__PURE__ */ t("span", { className: `${s.icon} ${s["icon-success"]}`, children: /* @__PURE__ */ t(f, {}) })
  ] }),
  /* @__PURE__ */ t("p", { className: s.subhead, children: "Your funds are on the way" }),
  i && /* @__PURE__ */ c("p", { className: `${s.description} ${n === null ? s.invisible : ""}`, children: [
    "This window will automatically close",
    " ",
    n === null ? "soon" : `in ${n}`
  ] }),
  !i && /* @__PURE__ */ t("p", { className: s.description, children: "Your transaction was submitted." }),
  !i && a !== void 0 && /* @__PURE__ */ t(o, { fullWidth: !0, onClick: a, children: "Go back" })
] }), B = ({
  reset: a,
  timeRemaining: n,
  unmountOnSuccess: i
}) => /* @__PURE__ */ c(e, { children: [
  /* @__PURE__ */ c("div", { className: `${s.graphic} ${s.pad}`, children: [
    /* @__PURE__ */ t("span", { className: `${s.ring} ${s["ring-success"]}` }),
    /* @__PURE__ */ t("span", { className: `${s.icon} ${s["icon-success"]}`, children: /* @__PURE__ */ t(g, {}) })
  ] }),
  /* @__PURE__ */ t("p", { className: s.subhead, children: "Success" }),
  i && /* @__PURE__ */ c("p", { className: `${s.description} ${n === null ? s.invisible : ""}`, children: [
    "This window will automatically close",
    " ",
    n === null ? "soon" : `in ${n}`
  ] }),
  !i && /* @__PURE__ */ t("p", { className: s.description, children: "Your transaction was successful." }),
  !i && a !== void 0 && /* @__PURE__ */ t(o, { fullWidth: !0, onClick: a, children: "Go back" })
] }), E = ({ reset: a, timeRemaining: n, unmountOnError: i }) => /* @__PURE__ */ c(e, { children: [
  /* @__PURE__ */ c("div", { className: `${s.graphic} ${s.pad}`, children: [
    /* @__PURE__ */ t("span", { className: `${s.ring} ${s["ring-error"]}` }),
    /* @__PURE__ */ t("span", { className: `${s.icon} ${s["icon-error"]}`, children: /* @__PURE__ */ t(b, {}) })
  ] }),
  /* @__PURE__ */ t("p", { className: s.subhead, children: "Payment failed" }),
  i && /* @__PURE__ */ c("p", { className: `${s.description} ${n === null ? s.invisible : ""}`, children: [
    "This window will automatically close",
    " ",
    n === null ? "soon" : `in ${n}`
  ] }),
  !i && /* @__PURE__ */ t("p", { className: s.description, children: "Something went wrong. Please try again." }),
  !i && a !== void 0 && /* @__PURE__ */ t(o, { fullWidth: !0, onClick: a, children: "Try again" })
] }), K = {
  waiting: G,
  transactionSubmitted: I,
  transactionSuccess: B,
  error: E
};
export {
  J as FundTransactionStatus
};
