// @coinbase/cdp-react/dist/components/Fund/components/FundAmountInput.js
import { jsxs as I, Fragment as _, jsx as o } from "react/jsx-runtime";
import { forwardRef as x, useId as g, useCallback as i } from "react";
import { ExchangeAmountInput as F } from "../../forms/ExchangeAmountInput/index.js";
import { Error as b } from "../../ui/Error/index.js";
import { useFundContext as v } from "../FundProvider.js";
import { createSetFieldAction as a } from "../useFundReducer.js";
import '../../../assets/FundAmountInput.css';const C = "FundAmountInput-module__error___lMqzh", s = {
  "input-wrapper": "FundAmountInput-module__input-wrapper___6r-sX",
  error: C
}, P = x(
  ({ "aria-describedby": u, "aria-invalid": p, error: r, ...m }, d) => {
    const { state: t, dispatch: e } = v(), c = g(), l = i(
      (n) => {
        e(a("cryptoAmount", n));
      },
      [e]
    ), y = i(
      (n) => {
        e(a("fiatAmount", n));
      },
      [e]
    ), f = i(
      (n) => {
        e(a("selectedInputType", n));
      },
      [e]
    ), A = [r ? c : void 0, u].filter(Boolean).join(" "), h = !!(r || p);
    return /* @__PURE__ */ I(_, { children: [
      /* @__PURE__ */ o("div", { className: s["input-wrapper"], children: /* @__PURE__ */ o(
        F,
        {
          ...m,
          ref: d,
          "aria-describedby": A || void 0,
          "aria-invalid": h || void 0,
          cryptoAmount: t.cryptoAmount,
          cryptoCurrency: t.cryptoCurrency,
          cryptoDecimalPlaces: t.cryptoDecimalPlaces,
          exchangeRate: t.exchangeRate,
          fiatAmount: t.fiatAmount,
          fiatCurrency: t.fiatCurrency,
          fiatDecimalPlaces: t.fiatDecimalPlaces,
          inputType: t.selectedInputType,
          isExchangeRatePending: t.isExchangeRatePending,
          min: 0,
          setCryptoAmount: l,
          setFiatAmount: y,
          setInputType: f
        }
      ) }),
      r && /* @__PURE__ */ o(b, { id: c, className: s.error, children: r })
    ] });
  }
);
P.displayName = "FundAmountInput";
export {
  P as FundAmountInput
};
