// @coinbase/cdp-react/dist/components/AuthButton/index.js
import { jsx as u, jsxs as j, Fragment as k } from "react/jsx-runtime";
import { useIsInitialized as y, useIsSignedIn as z } from "@coinbase/cdp-hooks";
import { u as D } from "../../chunks/useSendComponentCallOnce.BHZMuo6E.js";
import { useState as m, useRef as M, useCallback as x, useEffect as f } from "react";
import { SignInModal as N } from "../SignInModal/index.js";
import { SignOutButton as $ } from "../SignOutButton/index.js";
import { LoadingSkeleton as q } from "../ui/LoadingSkeleton/index.js";
import '../../assets/AuthButton.css';const F = "AuthButton-module__loading___JpDqi", A = {
  loading: F
}, K = ({
  children: d,
  closeOnSuccessDelay: S = 800,
  onSignInSuccess: g,
  onSignInSuccessTransitionEnd: p,
  onSignOutSuccess: h,
  placeholder: I,
  signInModal: O,
  signOutButton: _,
  ...b
}) => {
  D("auth_button");
  const { isInitialized: n } = y(), { isSignedIn: r } = z(), [l, s] = m(r), [e, c] = m(!1), [i, a] = m(void 0), o = M(!1), C = x(() => {
    g?.(), a(S);
  }, [S, g]), T = x(() => {
    h?.(), s(!1);
  }, [h]);
  f(() => {
    if (i === void 0)
      return;
    if (i === null) {
      o.current = !0;
      return;
    }
    const t = setTimeout(() => {
      o.current = !0, c(!1), a(void 0);
    }, i);
    return () => {
      t && clearTimeout(t);
    };
  }, [i]), f(() => {
    if (e)
      return;
    let t = null;
    return o.current && (t = setTimeout(() => {
      p?.(), o.current = !1, s(!0);
    }, 200)), () => {
      t && clearTimeout(t);
    };
  }, [e, p]), f(() => {
    if (r) {
      if (o.current)
        return;
      e ? a((t) => t === void 0 ? 0 : t) : s(!0);
    } else
      s(!1);
  }, [r, e]);
  const v = I ? I({ className: A.loading }) : /* @__PURE__ */ u(q, { className: A.loading }), B = _ ? _({ onSuccess: T }) : /* @__PURE__ */ u($, { onSuccess: T }), w = O ? O({ open: e, setIsOpen: c, onSuccess: C }) : /* @__PURE__ */ u(N, { open: e, setIsOpen: c, onSuccess: C });
  return /* @__PURE__ */ u("div", { ...b, children: d ? d({
    isInitialized: n,
    isSignedIn: l,
    Placeholder: v,
    SignOutButton: B,
    SignInModal: w
  }) : /* @__PURE__ */ j(k, { children: [
    !n && v,
    n && l && B,
    n && !l && w
  ] }) });
};
export {
  K as AuthButton
};
