// @coinbase/cdp-react/dist/components/forms/Input/index.js
import { jsx as m } from "react/jsx-runtime";
import "@radix-ui/react-form";
import { forwardRef as p, useMemo as l } from "react";
import '../../../assets/Input.css';const d = "Input-module__input___-mK8z", t = {
  input: d,
  "size-lg": "Input-module__size-lg___IYoRG",
  "size-md": "Input-module__size-md___8hkn5",
  "size-sm": "Input-module__size-sm___cnynF",
  "size-xs": "Input-module__size-xs___aK-NH"
}, o = p(
  ({ className: e = "", size: s = "md", ...i }, n) => {
    const _ = l(() => {
      const u = [t.input], a = {
        lg: t["size-lg"],
        md: t["size-md"],
        sm: t["size-sm"],
        xs: t["size-xs"]
      };
      return s && u.push(a[s]), u.join(" ");
    }, [s]);
    return /* @__PURE__ */ m("input", { ref: n, className: `${_} ${e}`, ...i });
  }
);
o.displayName = "Input";
const I = (e) => (i) => {
  const n = i === void 0 || i.valid ? void 0 : !0;
  return /* @__PURE__ */ m(o, { ...e, "aria-invalid": n });
};
export {
  o as Input,
  I as withValidityState
};
