// @coinbase/cdp-react/dist/theme/tokens.js
const e = {
  // Brand colors
  blue500: "#0052ff",
  blue550: "#014cec",
  black: "#0a0b0d",
  white: "#ffffff",
  // grayscale
  gray50: "#f9fafb",
  gray100: "#eef0f3",
  gray200: "#dcdfe4",
  gray500: "#9397a0",
  gray700: "#5b616e",
  gray900: "#1a1d21",
  // contextual
  red500: "#cf202f",
  green500: "#098551",
  amber500: "#ed702f"
}, a = {
  bg: {
    default: { value: e.white },
    // background
    alternate: { value: e.gray100 },
    // alternate background
    contrast: { value: "{colors.fg.default}" },
    // contrast background, used to mix with other backgrounds to create relative colors (i.e. for hover / pressed states)
    overlay: { value: "{colors.bg.alternate}", modify: { type: "color-alpha", value: 0.33 } },
    // overlay background
    skeleton: { value: "{colors.fg.default}", modify: { type: "color-alpha", value: 0.1 } },
    // skeleton background
    primary: { value: e.blue500 },
    // primary color background
    secondary: { value: e.gray100 },
    // secondary color background
    positiveWash: {
      value: "{colors.fg.positive}",
      modify: { type: "color-mix", value: [["{colors.bg.default}", "92%"]] }
    },
    negativeWash: {
      value: "{colors.fg.negative}",
      modify: { type: "color-mix", value: [["{colors.bg.default}", "92%"]] }
    },
    warningWash: {
      value: "{colors.fg.warning}",
      modify: { type: "color-mix", value: [["{colors.bg.default}", "92%"]] }
    }
  },
  fg: {
    default: { value: e.black },
    // text
    muted: { value: e.gray700 },
    // secondary (muted) text
    primary: { value: e.blue500 },
    // primary color text
    onPrimary: { value: e.white },
    // text on primary color
    onSecondary: { value: e.black },
    // text on secondary color
    positive: { value: e.green500 },
    // positive color text
    negative: { value: e.red500 },
    // negative color text
    warning: { value: e.amber500 },
    // warning color text
    onPositiveWash: { value: "{colors.fg.default}" },
    // text on positive wash color
    onNegativeWash: { value: "{colors.fg.default}" },
    // text on negative wash color
    onWarningWash: { value: "{colors.fg.default}" }
    // text on warning wash color
  },
  line: {
    default: { value: e.gray200 },
    // line color
    heavy: { value: e.gray500 },
    // higher contrast line color
    primary: { value: "{colors.fg.primary}" },
    // primary line color
    positive: { value: "{colors.fg.positive}" },
    // positive line color
    negative: { value: "{colors.fg.negative}" }
    // negative line color
  }
}, r = {
  page: {
    bg: {
      default: { value: "{colors.bg.default}" }
    },
    border: {
      default: { value: "{colors.line.default}" }
    },
    text: {
      default: { value: "{colors.fg.default}" },
      muted: { value: "{colors.fg.muted}" }
    }
  },
  cta: {
    primary: {
      bg: {
        default: { value: "{colors.bg.primary}" },
        hover: {
          value: "{colors.bg.primary}",
          modify: { type: "color-mix", value: [["{colors.bg.contrast}", "15%"]] }
        },
        pressed: {
          value: "{colors.bg.primary}",
          modify: { type: "color-mix", value: [["{colors.bg.contrast}", "30%"]] }
        }
      },
      border: {
        focus: { value: "{colors.line.primary}" }
      },
      text: {
        default: { value: "{colors.fg.onPrimary}" },
        hover: { value: "{colors.fg.onPrimary}" }
      }
    },
    secondary: {
      bg: {
        default: { value: "{colors.bg.secondary}" },
        hover: {
          value: "{colors.bg.secondary}",
          modify: { type: "color-mix", value: [["{colors.bg.contrast}", "10%"]] }
        },
        pressed: {
          value: "{colors.bg.secondary}",
          modify: { type: "color-mix", value: [["{colors.bg.contrast}", "20%"]] }
        }
      },
      border: {
        focus: { value: "{colors.line.primary}" }
      },
      text: {
        default: { value: "{colors.fg.onSecondary}" },
        hover: { value: "{colors.fg.onSecondary}" }
      }
    }
  },
  link: {
    primary: {
      text: {
        default: { value: "{colors.fg.primary}" },
        hover: {
          value: "{colors.fg.primary}",
          modify: { type: "color-mix", value: [["{colors.bg.contrast}", "15%"]] }
        },
        pressed: {
          value: "{colors.fg.primary}",
          modify: { type: "color-mix", value: [["{colors.bg.contrast}", "30%"]] }
        }
      }
    },
    secondary: {
      text: {
        default: { value: "{colors.fg.default}" },
        hover: {
          value: "{colors.fg.default}",
          modify: { type: "color-mix", value: [["{colors.bg.contrast}", "10%"]] }
        },
        pressed: {
          value: "{colors.fg.default}",
          modify: { type: "color-mix", value: [["{colors.bg.contrast}", "20%"]] }
        }
      }
    }
  },
  input: {
    bg: {
      default: { value: "{colors.bg.default}" },
      readonly: { value: "{colors.bg.alternate}" }
    },
    border: {
      default: { value: "{colors.line.heavy}" },
      focus: { value: "{colors.line.primary}" },
      error: { value: "{colors.line.negative}" },
      success: { value: "{colors.line.positive}" }
    },
    label: {
      default: { value: "{colors.fg.default}" }
    },
    placeholder: {
      default: { value: "{colors.fg.muted}" }
    },
    text: {
      default: { value: "{colors.fg.default}" },
      readonly: { value: "{colors.fg.muted}" }
    },
    errorText: {
      default: { value: "{colors.fg.negative}" }
    },
    successText: {
      default: { value: "{colors.fg.positive}" }
    }
  },
  select: {
    label: {
      default: { value: "{colors.fg.default}" }
    },
    trigger: {
      bg: {
        default: { value: "{colors.bg.default}" },
        hover: {
          value: "{colors.bg.default}",
          modify: { type: "color-mix", value: [["{colors.bg.contrast}", "5%"]] }
        },
        pressed: {
          value: "{colors.bg.default}",
          modify: { type: "color-mix", value: [["{colors.bg.contrast}", "7%"]] }
        }
      },
      border: {
        default: { value: "{colors.line.default}" },
        focus: { value: "{colors.line.primary}" },
        error: { value: "{colors.line.negative}" },
        success: { value: "{colors.line.positive}" }
      },
      placeholder: {
        default: { value: "{colors.fg.muted}" }
      },
      text: {
        default: { value: "{colors.fg.default}" }
      },
      errorText: {
        default: { value: "{colors.fg.negative}" }
      },
      successText: {
        default: { value: "{colors.fg.positive}" }
      }
    },
    list: {
      bg: {
        default: { value: "{colors.bg.default}" }
      },
      border: {
        default: { value: "{colors.line.default}" },
        focus: { value: "{colors.line.primary}" },
        error: { value: "{colors.line.negative}" },
        success: { value: "{colors.line.positive}" }
      },
      item: {
        bg: {
          default: { value: "{colors.bg.default}" },
          highlight: {
            value: "{colors.bg.default}",
            modify: { type: "color-mix", value: [["{colors.bg.contrast}", "10%"]] }
          }
        },
        text: {
          default: { value: "{colors.fg.default}" },
          muted: { value: "{colors.fg.muted}" },
          onHighlight: { value: "{colors.fg.default}" },
          mutedOnHighlight: { value: "{colors.fg.muted}" }
        }
      }
    }
  },
  code: {
    bg: {
      default: { value: "{colors.bg.alternate}" }
    },
    border: {
      default: { value: "{colors.line.heavy}" }
    },
    text: {
      default: { value: "{colors.fg.default}" }
    }
  }
}, t = {
  ...a,
  ...r
}, l = {
  family: {
    mono: {
      value: '"DM Mono", monospace'
    },
    sans: {
      value: '"Rethink Sans", -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"'
    },
    body: {
      value: "{font.family.sans}"
    },
    interactive: {
      value: "{font.family.sans}"
    }
  },
  size: {
    base: { value: 16 }
  }
}, o = {
  family: {
    page: {
      value: "{font.family.body}"
    },
    cta: {
      value: "{font.family.interactive}"
    },
    link: {
      value: "{font.family.interactive}"
    },
    input: {
      value: "{font.family.interactive}"
    },
    select: {
      value: "{font.family.interactive}"
    },
    code: {
      value: "{font.family.mono}"
    },
    iframe: {
      value: "{font.family.interactive}"
    }
  },
  url: {
    iframe: {
      value: ""
    }
  }
}, u = {
  family: {
    ...l.family,
    ...o.family
  },
  url: {
    ...o.url
  },
  size: {
    ...l.size
  }
}, s = {
  none: {
    value: 0
  },
  xs: {
    value: "{font.size.base}",
    modify: { type: "multiply", value: 0.25 }
  },
  sm: {
    value: "{font.size.base}",
    modify: { type: "multiply", value: 0.5 }
  },
  md: {
    value: "{font.size.base}",
    modify: { type: "multiply", value: 0.75 }
  },
  lg: {
    value: "{font.size.base}",
    modify: { type: "multiply", value: 1 }
  },
  xl: {
    value: "{font.size.base}",
    modify: { type: "multiply", value: 1.5 }
  },
  full: {
    value: 99999
  }
}, i = {
  banner: {
    value: "{borderRadius.lg}"
  },
  cta: {
    value: "{borderRadius.full}"
  },
  input: {
    value: "{borderRadius.sm}"
  },
  link: {
    value: "{borderRadius.full}"
  },
  modal: {
    value: "{borderRadius.sm}"
  },
  select: {
    trigger: {
      value: "{borderRadius.sm}"
    },
    list: {
      value: "{borderRadius.sm}"
    }
  }
}, f = {
  ...s,
  ...i
}, c = {
  borderRadius: f,
  colors: t,
  font: u
};
export {
  f as borderRadius,
  i as borderRadiusComponents,
  s as borderRadiusSemantic,
  t as colors,
  e as colorsBase,
  r as colorsComponents,
  a as colorsSemantic,
  u as font,
  o as fontComponents,
  l as fontSemantic,
  c as tokens
};
