// @coinbase/cdp-react/dist/components/ExportWallet/index.js
import { u as S, A as K } from "../../chunks/useSendComponentCallOnce.BHZMuo6E.js";
import { createContext as N, useMemo as x, useState as V, useCallback as Y, useContext as A } from "react";
import { VERSION as z } from "../../version.js";
import "../AuthButton/index.js";
import { useAppConfig as g } from "../CDPReactProvider/index.js";
import { CopyAddress as D } from "../CopyAddress/index.js";
import { CopyEvmKeyButton as U } from "../CopyEvmKeyButton/index.js";
import { CopySolanaKeyButton as X } from "../CopySolanaKeyButton/index.js";
import { jsx as o, jsxs as d, Fragment as E } from "react/jsx-runtime";
import { useCurrentUser as Z } from "@coinbase/cdp-hooks";
import { c as h } from "../../chunks/lite.1fxw3LjI.js";
import { Banner as _ } from "../ui/Banner/index.js";
import { CoinbaseFooter as $ } from "../ui/CoinbaseFooter/index.js";
import { Button as T } from "../ui/Button/index.js";
import { Modal as q, ModalClose as H, ModalContent as L, ModalTitle as G, ModalTrigger as J } from "../ui/Modal/index.js";
import { IconXMark as Q } from "../../icons/IconXMark.js";
import { childrenHasComponent as M } from "../../utils/childrenHasComponent.js";
import "../Fund/index.js";
import "../FundModal/index.js";
import "../LinkAuth/index.js";
import "../LinkAuthModal/index.js";
import "../SendEvmTransactionButton/index.js";
import "../SendSolanaTransactionButton/index.js";
import "../SignIn/index.js";
import "../SignIn/useSignInReducer.js";
import "../SignInModal/index.js";
import "../SignOutButton/index.js";
import "../ThemeProvider/index.js";
import "../../theme/cssVariables.js";
import "../../theme/theme.js";
import "../../theme/tokens.js";
import "../../theme/utils.js";
import "../../icons/IconAppleLogo.js";
import "../../icons/IconArrowLeft.js";
import "../../icons/IconArrowsUpDown.js";
import "../../icons/IconCheck.js";
import "../../icons/IconCheckCircle.js";
import "../../icons/IconChevronDown.js";
import "../../icons/IconCopy.js";
import "../../icons/IconEnvelope.js";
import "../../icons/IconExclamationCircle.js";
import "../../icons/IconExclamationTriangle.js";
import "../../icons/IconGoogleLogo.js";
import "../../icons/IconLock.js";
import "../../icons/IconPhone.js";
import "../../icons/IconMinus.js";
import "../../icons/IconPlus.js";
import "../../utils/clamp.js";
import "../../utils/getMessageFromUnknownError.js";
import "../../utils/isApiError.js";
import "../../utils/isEmailInvalid.js";
import "../../utils/parseValuesFromPhoneNumber.js";
import "../../utils/sendIframeMessage.js";
import "@coinbase/cdp-core";
import "../Fund/FundForm.js";
import "../Fund/FundTitle.js";
import "../Fund/FundProvider.js";
import "../../chunks/LinkAuthFlow.3HN888DV.js";
import "../LinkAuth/LinkAuthFlowBackButton.js";
import "../LinkAuth/LinkAuthItem.js";
import "../LinkAuth/LinkAuthItems.js";
import "../LinkAuth/LinkAuthTitle.js";
import "../LinkAuth/LinkAuthProvider.js";
import "../SignIn/SignInAuthMethodButtons.js";
import "../SignIn/SignInBackButton.js";
import "../SignIn/SignInDescription.js";
import "../SignIn/SignInFooter.js";
import "../SignIn/SignInForm.js";
import "../SignIn/SignInImage.js";
import "../SignIn/SignInTitle.js";
import "../SignIn/SignInProvider.js";
import '../../assets/ExportWallet.css';const tt = "ExportWallet-module__footer___NvrFY", f = {
  "export-wallet": "ExportWallet-module__export-wallet___EyZys",
  "no-footer": "ExportWallet-module__no-footer___fzl3R",
  "export-wallet-title": "ExportWallet-module__export-wallet-title___MuI09",
  footer: tt
}, k = N({
  address: "",
  isSessionExpired: !1,
  type: "evm-eoa"
}), C = () => {
  const t = A(k);
  if (!t)
    throw new Error("useExportWalletContext must be used within a ExportWalletProvider");
  return t;
}, ot = ({
  address: t,
  children: e,
  onIframeError: r,
  onIframeReady: a,
  onIframeSessionExpired: l,
  onCopySuccess: s
}) => {
  const { currentUser: n } = Z(), i = x(
    () => !!n?.solanaAccountObjects?.find((w) => w.address === t),
    [n?.solanaAccountObjects, t]
  ), m = x(
    () => !!n?.evmSmartAccountObjects?.find((w) => w.address === t),
    [n?.evmSmartAccountObjects, t]
  ), c = i ? "solana" : m ? "evm-smart" : "evm-eoa", [u, y] = V(!1), v = Y(() => {
    y(!0), l?.();
  }, [l]), B = x(
    () => ({
      address: t,
      isSessionExpired: u,
      type: c,
      onIframeError: r,
      onIframeReady: a,
      onIframeSessionExpired: v,
      onCopySuccess: s
    }),
    [
      t,
      u,
      c,
      v,
      r,
      a,
      s
    ]
  );
  return /* @__PURE__ */ o(k.Provider, { value: B, children: e });
}, P = ({
  children: t,
  className: e = "",
  as: r = "h2",
  ...a
}) => /* @__PURE__ */ o(
  r,
  {
    "data-part": "title",
    className: h(f["export-wallet-title"], e),
    ...a,
    children: t || "Export your wallet"
  }
), F = (t) => {
  const { isSessionExpired: e, type: r } = C();
  return e ? /* @__PURE__ */ o(_, { "data-part": "warning", variant: "warning", ...t, children: "Your export session has expired." }) : r === "evm-smart" ? /* @__PURE__ */ o(_, { "data-part": "warning", variant: "warning", ...t, children: "Cannot export a smart account's private key; only the owner's private key can be exported." }) : /* @__PURE__ */ o(_, { "data-part": "warning", variant: "warning", ...t, children: "Do not share your private key with anyone" });
}, I = ({
  label: t = "Your wallet address",
  ...e
}) => {
  const { address: r } = C();
  return /* @__PURE__ */ o(D, { "data-part": "copy-address", address: r, label: t, ...e });
}, O = ({
  fullWidth: t = !0,
  ...e
}) => {
  const { address: r, type: a, onIframeError: l, onIframeReady: s, onIframeSessionExpired: n, onCopySuccess: i } = C();
  return a === "solana" ? /* @__PURE__ */ o(
    X,
    {
      fullWidth: t,
      address: r,
      onError: l,
      onReady: s,
      onSessionExpired: n,
      onSuccess: i,
      ...e
    }
  ) : a === "evm-eoa" ? /* @__PURE__ */ o(
    U,
    {
      fullWidth: t,
      address: r,
      onError: l,
      onReady: s,
      onSessionExpired: n,
      onSuccess: i,
      ...e
    }
  ) : null;
}, R = (t) => /* @__PURE__ */ o($, { "data-part": "coinbase-footer", ...t }), et = ({ children: t }) => {
  const { showCoinbaseFooter: e } = g(), { type: r, isSessionExpired: a } = C();
  return t && typeof t == "function" ? t({ type: r, isSessionExpired: a }) : t || /* @__PURE__ */ d(E, { children: [
    /* @__PURE__ */ o(P, {}),
    /* @__PURE__ */ o(F, {}),
    /* @__PURE__ */ o(I, {}),
    /* @__PURE__ */ o(O, {}),
    e && /* @__PURE__ */ o(R, { className: f.footer })
  ] });
}, rt = ({
  address: t,
  children: e,
  className: r = "",
  onIframeError: a,
  onIframeReady: l,
  onIframeSessionExpired: s,
  onCopySuccess: n,
  ...i
}) => {
  S("export_wallet");
  const { showCoinbaseFooter: m } = g();
  return /* @__PURE__ */ o(
    "div",
    {
      className: h(
        f["export-wallet"],
        !m && f["no-footer"],
        r
      ),
      ...i,
      children: /* @__PURE__ */ o(
        ot,
        {
          address: t,
          onIframeReady: l,
          onCopySuccess: n,
          onIframeError: a,
          onIframeSessionExpired: s,
          children: /* @__PURE__ */ o(et, { children: e })
        }
      )
    }
  );
}, at = "ExportWalletModal-module__trigger___zMk8N", nt = "ExportWalletModal-module__content___KzNol", lt = "ExportWalletModal-module__footer___byg1T", st = "ExportWalletModal-module__modal___pX1AP", p = {
  "export-wallet": "ExportWalletModal-module__export-wallet___--L56",
  trigger: at,
  "title-bar": "ExportWalletModal-module__title-bar___RTPZq",
  content: nt,
  "close-button": "ExportWalletModal-module__close-button___ImsDR",
  "close-icon": "ExportWalletModal-module__close-icon___iAfyT",
  footer: lt,
  modal: st
}, j = N(null), it = () => {
  const t = A(j);
  if (!t)
    throw new Error("useExportWalletModalContext must be used within a ExportWalletModal");
  return t;
}, b = ({
  children: t,
  className: e = "",
  label: r,
  ...a
}) => /* @__PURE__ */ o(J, { asChild: !0, children: t || /* @__PURE__ */ o(T, { className: h(p.trigger, e), ...a, children: r || "Export wallet" }) }), pt = ({
  children: t = "Copy private key to export wallet",
  ...e
}) => /* @__PURE__ */ o(G, { asChild: !0, children: /* @__PURE__ */ o(P, { ...e, children: t }) }), mt = ({ children: t }) => /* @__PURE__ */ o(H, { asChild: !0, children: t || /* @__PURE__ */ o(
  T,
  {
    "aria-label": "Close",
    className: p["close-button"],
    variant: "transparentSecondary",
    children: /* @__PURE__ */ o(Q, { className: p["close-icon"] })
  }
) }), W = ({
  children: t,
  className: e = "",
  ...r
}) => {
  const { showCoinbaseFooter: a } = g(), { address: l, onIframeReady: s, onCopySuccess: n, onIframeError: i, onIframeSessionExpired: m } = it();
  return /* @__PURE__ */ o(L, { "aria-describedby": void 0, className: h(p.modal, e), ...r, children: /* @__PURE__ */ o(
    rt,
    {
      address: l,
      onIframeReady: s,
      onCopySuccess: n,
      onIframeError: i,
      onIframeSessionExpired: m,
      className: p["export-wallet"],
      children: t || (({ isSessionExpired: c }) => /* @__PURE__ */ d(E, { children: [
        /* @__PURE__ */ d("div", { className: p["title-bar"], children: [
          /* @__PURE__ */ o(pt, {}),
          /* @__PURE__ */ o(mt, {})
        ] }),
        /* @__PURE__ */ d("div", { className: p.content, children: [
          c ? /* @__PURE__ */ o(_, { role: "alert", "data-part": "warning", variant: "warning", children: "Your export session has expired. Please close and reopen the modal to export again." }) : /* @__PURE__ */ o(F, {}),
          /* @__PURE__ */ o(I, {}),
          /* @__PURE__ */ o(O, {}),
          a && /* @__PURE__ */ o(R, { className: p.footer })
        ] })
      ] }))
    }
  ) });
}, Ao = ({
  address: t,
  children: e,
  onIframeReady: r,
  onCopySuccess: a,
  onIframeError: l,
  onIframeSessionExpired: s,
  open: n,
  setIsOpen: i
}) => {
  S("export_wallet_modal");
  const m = x(
    () => ({ address: t, onIframeReady: r, onCopySuccess: a, onIframeError: l, onIframeSessionExpired: s }),
    [t, r, a, l, s]
  ), c = e ? M(e, b) : !1, u = e ? M(e, W) : !1, y = !c && !u;
  return /* @__PURE__ */ o(j.Provider, { value: m, children: /* @__PURE__ */ o(q, { open: n, onOpenChange: i, children: y ? /* @__PURE__ */ d(E, { children: [
    /* @__PURE__ */ o(b, { children: e }),
    /* @__PURE__ */ o(W, {})
  ] }) : /* @__PURE__ */ d(E, { children: [
    e,
    !u && /* @__PURE__ */ o(W, {})
  ] }) }) });
};
K.registerPackageVersion("react", z);
export {
  Ao as E,
  rt as ExportWallet,
  I as ExportWalletCopyAddress,
  O as ExportWalletCopyKeyButton,
  R as ExportWalletFooter,
  P as ExportWalletTitle,
  F as ExportWalletWarning,
  mt as a,
  W as b,
  pt as c,
  b as d
};
