// @coinbase/cdp-react/dist/components/forms/CountrySelect/index.js
import { jsxs as l, jsx as e } from "react/jsx-runtime";
import { SelectValue as m, SelectIcon as u } from "@radix-ui/react-select";
import { Select as _, SelectTrigger as y, SelectContent as g, SelectItem as h } from "../Select/index.js";
import { ButtonBase as p } from "../../ui/ButtonBase/index.js";
import { VisuallyHidden as C } from "../../ui/VisuallyHidden/index.js";
import { countries as s } from "../../../data/countries.js";
import { IconChevronDown as S } from "../../../icons/IconChevronDown.js";
import "../../../theme/theme.js";
import '../../../assets/CountrySelect.css';const f = "CountrySelect-module__flag___bNBsn", c = {
  "country-select-trigger": "CountrySelect-module__country-select-trigger___xGSAp",
  "select-icon": "CountrySelect-module__select-icon___wEQVx",
  "country-select-content": "CountrySelect-module__country-select-content___WSd9-",
  "country-select-item": "CountrySelect-module__country-select-item___Mr6JZ",
  flag: f,
  "country-calling-code": "CountrySelect-module__country-calling-code___d61ih"
}, N = s.sort((r, o) => r.name.localeCompare(o.name)), v = ({
  className: r = "",
  size: o,
  defaultValue: a,
  value: i,
  onValueChange: d
}) => {
  const n = s.find((t) => t.code === (i ?? a));
  return /* @__PURE__ */ l(
    _,
    {
      name: "country-code",
      defaultValue: a,
      value: i,
      onValueChange: d,
      children: [
        /* @__PURE__ */ e(y, { "aria-label": "Country / region", placeholder: "+1", children: /* @__PURE__ */ l(p, { className: `${c["country-select-trigger"]} ${r}`, size: o, children: [
          /* @__PURE__ */ e(m, { "aria-label": n?.name ?? "Select a country", children: n && /* @__PURE__ */ l("span", { className: c["country-select-item"], children: [
            /* @__PURE__ */ e("span", { "aria-hidden": "true", className: c.flag, children: n.flag }),
            /* @__PURE__ */ e(C, { children: n.name }),
            /* @__PURE__ */ l("span", { children: [
              "+",
              n.callingCode
            ] })
          ] }) }),
          /* @__PURE__ */ e(u, { asChild: !0, children: /* @__PURE__ */ e(S, { className: c["select-icon"] }) })
        ] }) }),
        /* @__PURE__ */ e(g, { className: c["country-select-content"], children: N.map((t) => /* @__PURE__ */ e(h, { indicator: !0, value: t.code, children: /* @__PURE__ */ l("span", { className: c["country-select-item"], children: [
          /* @__PURE__ */ e("span", { "aria-hidden": "true", className: c.flag, children: t.flag }),
          /* @__PURE__ */ e("span", { children: t.name }),
          /* @__PURE__ */ l("span", { className: c["country-calling-code"], children: [
            "+",
            t.callingCode
          ] })
        ] }) }, t.code)) })
      ]
    }
  );
};
export {
  v as CountrySelect
};
