// @coinbase/cdp-react/dist/components/ui/IframeButton/index.js
import { jsxs as H, jsx as B } from "react/jsx-runtime";
import { c as F } from "../../../chunks/lite.1fxw3LjI.js";
import { forwardRef as R, useRef as y, useMemo as C, useState as U, useEffect as T } from "react";
import { useTheme as L } from "../../ThemeProvider/index.js";
import { LoadingSkeleton as W } from "../LoadingSkeleton/index.js";
import "../../../theme/theme.js";
import { colorsBase as i } from "../../../theme/tokens.js";
import '../../../assets/IframeButton.css';const N = "IframeButton-module__label___Vvl8W", P = "IframeButton-module__iframe___0RBLL", j = "IframeButton-module__hidden___LcF1X", s = {
  "iframe-button": "IframeButton-module__iframe-button___XeAhH",
  "full-width": "IframeButton-module__full-width___lQWpM",
  "size-lg": "IframeButton-module__size-lg___HCOiF",
  "size-md": "IframeButton-module__size-md___p44a4",
  "size-sm": "IframeButton-module__size-sm___U8ILi",
  "size-xs": "IframeButton-module__size-xs___S2jGa",
  "loading-skeleton": "IframeButton-module__loading-skeleton___fzBol",
  label: N,
  "with-icon": "IframeButton-module__with-icon___KBiq1",
  iframe: P,
  hidden: j,
  "computed-style": "IframeButton-module__computed-style___ciB13"
}, k = {
  pageBg: i.white,
  buttonBg: i.blue500,
  buttonBgHover: "#0748d6",
  buttonBgPressed: "#0b3eae",
  buttonBgFocus: i.blue550,
  buttonBorder: i.blue550,
  buttonBorderHover: "#0748d6",
  buttonBorderPressed: "#0b3eae",
  buttonBorderFocus: i.blue550,
  buttonBorderFocusInset: i.white,
  buttonText: i.white,
  buttonTextHover: i.white,
  buttonTextPressed: i.white,
  buttonTextFocus: i.white,
  buttonBorderRadius: 99999,
  buttonFontSize: 16,
  buttonFontWeight: 500,
  buttonSize: "md",
  fontUrl: "",
  fontFamily: ""
}, $ = Object.keys(k), q = (o) => $.every(
  (e) => e === "fontUrl" || e === "buttonSize" ? !0 : o[e] !== void 0
), v = (o, e) => o === e ? !0 : !o || !e ? !1 : $.every((t) => o[t] === e[t]), f = (o, e, t) => (e.style.backgroundColor = `var(--cdp-web-${o})`, A(getComputedStyle(e).backgroundColor, t)), A = (o, e) => {
  if (!o || typeof o != "string")
    return console.warn("Invalid color input:", o), "";
  const t = o.trim();
  if (t.startsWith("#"))
    return /^#[0-9A-Fa-f]{6}([0-9A-Fa-f]{2})?$/.test(t) ? t.toLowerCase() : (console.warn("Invalid hex color:", t), "");
  if (t === "transparent") return "transparent";
  if (t === "none") return "none";
  const r = e.getContext("2d", { willReadFrequently: !0 });
  if (!r)
    return console.warn("Could not create canvas context"), "";
  if (r.fillStyle = t, r.fillStyle.startsWith("#"))
    return r.fillStyle.toLowerCase();
  r.fillRect(0, 0, 1, 1);
  const a = r.getImageData(0, 0, 1, 1), [m, n, c, l] = a.data, d = m.toString(16).padStart(2, "0"), u = n.toString(16).padStart(2, "0"), _ = c.toString(16).padStart(2, "0");
  if (l < 255) {
    const b = l.toString(16).padStart(2, "0");
    return `#${d}${u}${_}${b}`;
  }
  return `#${d}${u}${_}`;
}, M = (o, e, t, r) => {
  const a = getComputedStyle(t), m = f("colors-page-bg-default", t, r), n = f(`colors-cta-${e}-bg-default`, t, r), c = f(`colors-cta-${e}-bg-hover`, t, r), l = f(`colors-cta-${e}-bg-pressed`, t, r), d = f(`colors-cta-${e}-text-default`, t, r);
  return {
    pageBg: m,
    buttonBg: n,
    buttonBgHover: c,
    buttonBgPressed: l,
    buttonBgFocus: n,
    buttonBorder: n,
    buttonBorderHover: c,
    buttonBorderPressed: l,
    buttonBorderFocus: f(`colors-cta-${e}-border-focus`, t, r),
    buttonBorderFocusInset: m,
    buttonBorderRadius: parseInt(a.borderRadius),
    buttonFontSize: parseInt(a.fontSize),
    buttonFontWeight: parseInt(a.fontWeight),
    buttonText: d,
    buttonTextHover: f(`colors-cta-${e}-text-hover`, t, r),
    buttonTextPressed: d,
    buttonTextFocus: d,
    buttonSize: o ?? "md",
    fontFamily: a.fontFamily,
    fontUrl: ""
  };
}, V = {
  xs: s["size-xs"],
  sm: s["size-sm"],
  md: s["size-md"],
  lg: s["size-lg"]
}, D = R(
  ({
    className: o = "",
    isPending: e,
    label: t,
    icon: r,
    fullWidth: a,
    size: m,
    theme: n,
    onThemeChange: c,
    ...l
  }, d) => {
    const u = m === "none" ? "md" : m, { theme: _ } = L(), b = _["font-url-iframe"], p = y(null), x = y(null), h = C(() => !!n && q(n), [n]), [S, z] = U(
      h ? {
        ...n,
        fontUrl: n?.fontUrl ?? b,
        buttonSize: u ?? "md"
      } : void 0
    );
    return T(() => {
      if (h) {
        const g = {
          ...n,
          fontUrl: n?.fontUrl ?? b,
          buttonSize: u ?? "md"
        };
        z((w) => v(g, w) ? w : g);
        return;
      }
      if (!p.current || !x.current) return;
      const I = {
        ...M(
          u ?? "md",
          l.variant ?? "primary",
          p.current,
          x.current
        ),
        fontUrl: b,
        ...n
      };
      z((g) => v(I, g) ? g : I);
    }, [l.variant, u, h, n, b, _]), T(() => {
      S && c?.(S);
    }, [S, c]), /* @__PURE__ */ H(
      "div",
      {
        className: F(
          s["iframe-button"],
          r && s["with-icon"],
          a && s["full-width"],
          u && V[u]
        ),
        "data-part": "iframe-button",
        children: [
          /* @__PURE__ */ B("span", { className: s.label, children: t }),
          e && /* @__PURE__ */ B(W, { className: F(s["loading-skeleton"]) }),
          /* @__PURE__ */ B(
            "iframe",
            {
              ref: d,
              className: F(s.iframe, e ? s.hidden : void 0, o),
              ...l
            }
          ),
          !h && /* @__PURE__ */ B("div", { ref: p, className: s["computed-style"], children: /* @__PURE__ */ B("canvas", { ref: x }) })
        ]
      }
    );
  }
);
D.displayName = "IframeButton";
export {
  D as IframeButton
};
