// @coinbase/cdp-react/dist/components/forms/OTPForm/index.js
import { jsxs as r, jsx as e } from "react/jsx-runtime";
import "@coinbase/cdp-hooks";
import { Form as u, FormSubmit as h } from "@radix-ui/react-form";
import "react";
import { OTP as v } from "../OTP/index.js";
import { Button as F } from "../../ui/Button/index.js";
import { LoadingSpinner as N } from "../../ui/LoadingSpinner/index.js";
import { VisuallyHidden as g } from "../../ui/VisuallyHidden/index.js";
import { isApiError as b } from "../../../utils/isApiError.js";
import "libphonenumber-js";
import '../../../assets/OTPForm.css';const o = {
  "otp-form": "OTPForm-module__otp-form___G1-63",
  "form-footer": "OTPForm-module__form-footer___ybmdd",
  "server-state-wrapper": "OTPForm-module__server-state-wrapper___R5j5k",
  "loading-spinner": "OTPForm-module__loading-spinner___8N9XU",
  "resend-wrapper": "OTPForm-module__resend-wrapper___aZUAz",
  "reset-timer": "OTPForm-module__reset-timer___7ElWj"
}, A = ({
  canResetOTP: m,
  error: t,
  isPending: a,
  onOTPChange: s,
  onResendOTP: l,
  otp: p,
  otpInputsRef: d,
  passwordLength: n,
  resendCountdown: i,
  successMessage: _,
  ...c
}) => /* @__PURE__ */ r(u, { ...c, children: [
  /* @__PURE__ */ e(
    v,
    {
      ref: d,
      name: "otp",
      "aria-label": "Enter code",
      autoSubmit: !0,
      passwordLength: n || 6,
      value: p,
      error: b(t) ? t.message : t,
      onChange: (f) => f.preventDefault(),
      onValueChange: s,
      className: o["otp-form"],
      successMessage: _
    }
  ),
  /* @__PURE__ */ r("div", { className: o["form-footer"], children: [
    a && !m && /* @__PURE__ */ e("div", { className: o["server-state-wrapper"], children: a && !m && /* @__PURE__ */ e(N, { className: o["loading-spinner"] }) }),
    /* @__PURE__ */ r("div", { className: o["resend-wrapper"], children: [
      m && /* @__PURE__ */ e(h, { asChild: !0, children: /* @__PURE__ */ e(
        F,
        {
          type: "button",
          onClick: l,
          isPending: a,
          variant: "linkPrimary",
          children: "Resend code"
        }
      ) }),
      !m && i !== null && /* @__PURE__ */ e("div", { className: o["reset-timer"], children: /* @__PURE__ */ r("p", { children: [
        "Resend code in ",
        /* @__PURE__ */ r("span", { "aria-hidden": "true", children: [
          i,
          "s"
        ] }),
        /* @__PURE__ */ r(g, { children: [
          i,
          " ",
          i === 1 ? "second" : "seconds"
        ] })
      ] }) })
    ] })
  ] })
] });
export {
  A as OTPForm
};
