// @coinbase/cdp-react/dist/hooks/useKeyExportPostMessage.js
import { buildKeyExportIframeUrl as K, getAccessToken as P, getAccessTokenExpiration as L, SECURE_IFRAME_KEY_EXPORT_EVENT_TYPE as n, isSecureIframeKeyExportMessage as C } from "@coinbase/cdp-core";
import { useCurrentUser as N } from "@coinbase/cdp-hooks";
import { useState as w, useRef as Y, useMemo as F, useCallback as A, useEffect as _ } from "react";
import "libphonenumber-js";
import { sendIframeMessage as x } from "../utils/sendIframeMessage.js";
const k = "https://secure-wallet.cdp.coinbase.com", G = (o, s) => {
  x(o, s);
}, O = ({
  address: o,
  basePath: s = k,
  copiedLabel: E,
  icon: d,
  iframeRef: t,
  label: p,
  onStatusUpdate: g,
  projectId: y,
  theme: a,
  type: T
}) => {
  const { currentUser: c } = N(), [I, M] = w(!0), [i, R] = w(!1), l = Y(!1), S = F(() => K({
    projectId: y,
    label: p,
    copiedLabel: E,
    icon: d,
    basePath: s
  }), [y, p, E, d, s]), f = A(async () => {
    if (!c || !t.current) return;
    const r = await P({ forceRefresh: !0 }), e = await L();
    G(t.current, {
      type: n.GET_PRIVATE_KEY,
      payload: {
        address: o,
        type: T,
        authState: {
          accessToken: r || "",
          expiresAt: e || 0,
          user: c
        },
        theme: a
      }
    });
  }, [c, o, T, a, t]), m = A(
    (r, e) => {
      g?.(r, e), M(!1);
    },
    [g]
  );
  return _(() => {
    const r = l.current;
    l.current = i, r && i && a && x(t.current, {
      type: n.THEME,
      payload: {
        theme: a
      }
    });
  }, [a, i, t]), _(() => {
    const r = (e) => {
      let u = "";
      try {
        u = new URL(s || k).origin;
      } catch (U) {
        console.error("Error parsing iframe origin", U);
      }
      if (!(!u || e.origin !== u) && C(e.data))
        switch (e.data?.type) {
          case n.LISTENING:
            R(!0), f();
            break;
          case n.STATUS:
            m(e.data.payload.status, e.data.payload.message);
            break;
        }
    };
    return window.addEventListener("message", r), () => {
      window.removeEventListener("message", r);
    };
  }, [f, m, s]), { iframeUrl: S, isPending: I };
};
export {
  G as sendKeyExportMessage,
  O as useKeyExportPostMessage
};
