// @coinbase/cdp-react/dist/components/forms/PhoneNumberInput/index.js
import { jsxs as h, jsx as s } from "react/jsx-runtime";
import "libphonenumber-js";
import { forwardRef as R, useRef as S, useCallback as d, useMemo as w, useEffect as z } from "react";
import { CountrySelect as B } from "../CountrySelect/index.js";
import { Input as H } from "../Input/index.js";
import { VisuallyHidden as M } from "../../ui/VisuallyHidden/index.js";
import { countries as W } from "../../../data/countries.js";
import { usePhoneNumberFormatter as k } from "./usePhoneNumberFormatter.js";
import '../../../assets/PhoneNumberInput.css';const A = "PhoneNumberInput-module__invalid___zDA2p", E = "PhoneNumberInput-module__success___uPjMx", G = "PhoneNumberInput-module__input___H-XG-", l = {
  "phone-number-input": "PhoneNumberInput-module__phone-number-input___mYty2",
  "inline-country-calling-code": "PhoneNumberInput-module__inline-country-calling-code___FFgtI",
  invalid: A,
  success: E,
  input: G,
  "country-calling-code": "PhoneNumberInput-module__country-calling-code___N6zDW"
}, T = R(
  ({
    autoComplete: a = "tel-national",
    className: _ = "",
    countryCode: n = "US",
    countryOptions: p = !0,
    onCountryCodeChange: g,
    onPhoneNumberChange: c,
    phoneNumber: f,
    ...u
  }, o) => {
    const b = S(null), C = d(
      (e) => {
        b.current = e, o && (typeof o == "function" ? o(e) : o.current = e);
      },
      [o]
    ), {
      formatBeforePhoneNumberChange: v,
      formatBeforeCountryCodeChange: y,
      placeholder: P,
      formattedDisplayNumber: V
    } = k(f.value, n), j = d(
      (e) => {
        let t = e.target.value.replace(/\D/g, "");
        t === f.value && (t = t.slice(0, -1));
        const F = v(t);
        c?.(F);
      },
      [v, c, f]
    ), r = d(
      (e) => {
        const t = y(e);
        g?.(e), c?.(t);
      },
      [y, g, c]
    ), x = d(
      (e) => {
        r(e), setTimeout(() => {
          b.current?.focus();
        }, 0);
      },
      [r]
    ), i = typeof p == "boolean" ? p : p.length > 0, m = !i && n ? W.find((e) => e.code === n) : void 0, N = u["aria-invalid"] && u["aria-invalid"] !== "false", I = u["data-success"] && u["data-success"] !== "false", D = w(() => {
      const e = [l["phone-number-input"], _];
      return N && e.push(l.invalid), I && e.push(l.success), !i && m && e.push(l["inline-country-calling-code"]), e.join(" ");
    }, [N, I, _, m, i]);
    return z(() => {
      r(n);
    }, [n, r]), /* @__PURE__ */ h("div", { className: D, children: [
      !i && m && /* @__PURE__ */ s(U, { country: m }),
      i && /* @__PURE__ */ s(
        B,
        {
          "aria-label": "Country code",
          defaultValue: n,
          onValueChange: x
        }
      ),
      /* @__PURE__ */ s(
        H,
        {
          ...u,
          "aria-label": "Phone number",
          ref: C,
          className: l.input,
          autoComplete: a,
          onChange: j,
          placeholder: P,
          type: "tel",
          value: V
        }
      )
    ] });
  }
);
T.displayName = "PhoneNumberInput";
const U = ({ country: a }) => /* @__PURE__ */ h("span", { className: l["country-calling-code"], children: [
  /* @__PURE__ */ s("span", { "aria-hidden": "true", children: a.flag }),
  /* @__PURE__ */ s(M, { children: a.name }),
  /* @__PURE__ */ h("span", { children: [
    "+",
    a.callingCode
  ] })
] });
export {
  T as PhoneNumberInput
};
