// @coinbase/cdp-react/dist/components/forms/ExchangeAmountInput/index.js
import { jsxs as u, Fragment as $, jsx as n } from "react/jsx-runtime";
import { forwardRef as v, useCallback as C } from "react";
import { AmountInput as L } from "../AmountInput/index.js";
import { Button as z } from "../../ui/Button/index.js";
import { LoadingSkeleton as j } from "../../ui/LoadingSkeleton/index.js";
import { DEFAULT_LOCALE as D } from "../../../data/locale.js";
import { IconArrowsUpDown as F } from "../../../icons/IconArrowsUpDown.js";
import { useExchangeAmountInput as S } from "./useExchangeAmountInput.js";
import '../../../assets/ExchangeAmountInput.css';const V = "ExchangeAmountInput-module__icon___b5JpF", t = {
  "flex-row": "ExchangeAmountInput-module__flex-row___ZKB-Q",
  "gap-1": "ExchangeAmountInput-module__gap-1___vuNQW",
  "gap-2": "ExchangeAmountInput-module__gap-2___4-s-b",
  "swap-button": "ExchangeAmountInput-module__swap-button___i9lSz",
  icon: V,
  "output-loading-skeleton": "ExchangeAmountInput-module__output-loading-skeleton___gKMEZ",
  "output-loading-skeleton-placeholder": "ExchangeAmountInput-module__output-loading-skeleton-placeholder___1-jT-",
  "output-currency": "ExchangeAmountInput-module__output-currency___vlpC1"
}, B = v(
  ({
    cryptoAmount: i,
    cryptoCurrency: l,
    cryptoDecimalPlaces: _,
    fiatAmount: d,
    fiatCurrency: p,
    fiatDecimalPlaces: g,
    exchangeRate: a,
    inputType: o,
    isExchangeRatePending: h,
    locale: c = D,
    setCryptoAmount: x,
    setFiatAmount: f,
    setInputType: r,
    ...A
  }, E) => {
    const w = o === "fiat" ? `switch input to ${l}` : `switch input to ${p}`, {
      inputValue: I,
      inputCurrency: N,
      inputDecimalPlaces: b,
      formatOutputValue: s,
      outputValue: e,
      outputCurrency: m,
      handleChange: k
    } = S({
      cryptoAmount: i,
      cryptoCurrency: l,
      cryptoDecimalPlaces: _,
      exchangeRate: a,
      fiatAmount: d,
      fiatCurrency: p,
      fiatDecimalPlaces: g,
      inputType: o,
      locale: c,
      setCryptoAmount: x,
      setFiatAmount: f
    }), y = C(() => {
      r(o === "fiat" ? "crypto" : "fiat");
    }, [o, r]);
    return /* @__PURE__ */ u($, { children: [
      /* @__PURE__ */ n(
        L,
        {
          ref: E,
          amount: I,
          unit: N,
          decimalPlaces: b,
          locale: c,
          setAmount: k,
          ...A
        }
      ),
      !a && h && /* @__PURE__ */ n(j, { className: t["output-loading-skeleton"], children: /* @__PURE__ */ u("span", { className: t["output-loading-skeleton-placeholder"], children: [
        s(e ?? 0),
        " ",
        m
      ] }) }),
      a && /* @__PURE__ */ u("span", { className: `${t["flex-row"]} ${t["gap-1"]}`, children: [
        /* @__PURE__ */ n(
          z,
          {
            "aria-label": w,
            className: t["swap-button"],
            onClick: y,
            type: "button",
            size: "xs",
            variant: "transparentSecondary",
            children: /* @__PURE__ */ n(F, { className: t.icon })
          }
        ),
        /* @__PURE__ */ u("span", { className: `${t["flex-row"]} ${t["gap-2"]}`, children: [
          /* @__PURE__ */ n(
            "span",
            {
              "data-part": "amount-output-unsized-value",
              "data-testid": "exchange-amount-output-value",
              children: s(isNaN(e ?? 0) ? 0 : e ?? 0)
            }
          ),
          /* @__PURE__ */ n(
            "span",
            {
              "data-part": "amount-output-unsized-currency",
              "data-testid": "exchange-amount-output-currency",
              className: t["output-currency"],
              children: m
            }
          )
        ] })
      ] })
    ] });
  }
);
B.displayName = "ExchangeAmountInput";
export {
  B as ExchangeAmountInput
};
