// @coinbase/cdp-react/dist/components/forms/ExchangeAmountInput/useExchangeAmountInput.js
import { Big as B } from "big.js";
import { useCallback as u, useMemo as _ } from "react";
import { useNumberParseAndFormat as I } from "../AmountInput/useNumberParseAndFormat.js";
import { DEFAULT_LOCALE as k } from "../../../data/locale.js";
const O = (f) => {
  try {
    return B(f);
  } catch {
    return null;
  }
}, z = ({
  cryptoAmount: f,
  cryptoCurrency: F,
  cryptoDecimalPlaces: n = 8,
  exchangeRate: m,
  fiatAmount: V,
  fiatCurrency: h,
  fiatDecimalPlaces: i = 2,
  inputType: o,
  locale: E = k,
  setFiatAmount: s,
  setCryptoAmount: d
}) => {
  const L = u(
    (r) => {
      s(r);
      const t = O(r);
      if (r === 0 || isNaN(r) || t === null) {
        d(NaN);
        return;
      }
      const N = t.times(m ?? 0);
      d(
        N.eq(0) ? 0 : +N.toFixed(n, B.roundDown)
      );
    },
    [m, s, d, n]
  ), b = u(
    (r) => {
      d(r);
      const t = O(r);
      if (r === 0 || isNaN(r) || t === null) {
        s(NaN);
        return;
      }
      const N = m ? t.div(m) : B(0);
      s(
        N.eq(0) ? 0 : +N.toFixed(i, B.roundDown)
      );
    },
    [m, s, d, i]
  ), q = u(
    (r, t) => {
      o === "fiat" ? L(r) : b(r), t?.(r);
    },
    [L, b, o]
  ), { formatter: a } = I({
    inputValue: "",
    locale: E,
    options: {
      maximumFractionDigits: i,
      minimumFractionDigits: 0
    }
  }), { formatter: g } = I({
    inputValue: "",
    locale: E,
    options: {
      maximumFractionDigits: n,
      minimumFractionDigits: 0
    }
  }), w = u(
    (r) => o === "fiat" ? a.format(r) : g.format(r),
    [g, a, o]
  ), x = u(
    (r) => o === "fiat" ? g.format(r) : a.format(r),
    [g, a, o]
  );
  return _(
    () => ({
      inputCurrency: o === "fiat" ? h : F,
      inputDecimalPlaces: o === "fiat" ? i : n,
      inputValue: o === "fiat" ? V : f,
      formatInputValue: w,
      formatOutputValue: x,
      handleChange: q,
      outputCurrency: o === "fiat" ? F : h,
      outputDecimalPlaces: o === "fiat" ? n : i,
      outputValue: o === "fiat" ? f : V
    }),
    [
      f,
      F,
      n,
      V,
      h,
      i,
      w,
      x,
      q,
      o
    ]
  );
};
export {
  z as useExchangeAmountInput
};
