// @coinbase/cdp-react/dist/components/SignInModal/index.js
import { jsx as n, jsxs as t, Fragment as c } from "react/jsx-runtime";
import { u as C } from "../../chunks/useSendComponentCallOnce.BHZMuo6E.js";
import { createContext as b, useMemo as x, useContext as v } from "react";
import { c as N } from "../../chunks/lite.1fxw3LjI.js";
import { useAppConfig as w } from "../CDPReactProvider/index.js";
import { SignIn as T } from "../SignIn/index.js";
import "@coinbase/cdp-hooks";
import { Button as h } from "../ui/Button/index.js";
import { ModalTrigger as k, ModalContent as B, ModalClose as D, Modal as F, ModalTitle as j, ModalDescription as y } from "../ui/Modal/index.js";
import { VisuallyHidden as g } from "../ui/VisuallyHidden/index.js";
import { IconXMark as A } from "../../icons/IconXMark.js";
import { childrenHasComponent as f } from "../../utils/childrenHasComponent.js";
import { SignInBackButton as V } from "../SignIn/SignInBackButton.js";
import { SignInImage as z } from "../SignIn/SignInImage.js";
import { SignInForm as E } from "../SignIn/SignInForm.js";
import { SignInTitle as _ } from "../SignIn/SignInTitle.js";
import { SignInDescription as u } from "../SignIn/SignInDescription.js";
import { SignInAuthMethodButtons as H } from "../SignIn/SignInAuthMethodButtons.js";
import { SignInFooter as J } from "../SignIn/SignInFooter.js";
import '../../assets/SignInModal.css';const O = "SignInModal-module__trigger___IcJ8x", X = "SignInModal-module__footer___6qEo2", $ = "SignInModal-module__buttons___jhaVn", e = {
  trigger: O,
  "sign-in": "SignInModal-module__sign-in___n05-5",
  "no-footer": "SignInModal-module__no-footer___meSt1",
  "title-desc-wrapper": "SignInModal-module__title-desc-wrapper___--XLa",
  "sign-in-form": "SignInModal-module__sign-in-form___Jzx1-",
  "sign-in-image": "SignInModal-module__sign-in-image___DGg7B",
  footer: X,
  buttons: $,
  "back-button": "SignInModal-module__back-button___nMh2-",
  "close-icon": "SignInModal-module__close-icon___b-gMh"
}, M = b(null), q = () => {
  const i = v(M);
  if (!i)
    throw new Error("useSignInModalContext must be used within a SignInModal");
  return i;
}, p = ({
  children: i,
  className: o = "",
  label: s,
  ...r
}) => /* @__PURE__ */ n(k, { asChild: !0, children: i || /* @__PURE__ */ n(h, { className: N(e.trigger, o), ...r, children: s || "Sign in" }) }), m = (i) => {
  const { showCoinbaseFooter: o } = w(), { authMethods: s, onSuccess: r } = q();
  return /* @__PURE__ */ n(B, { ...i, children: /* @__PURE__ */ n(
    T,
    {
      authMethods: s,
      onSuccess: r,
      className: `${e["sign-in"]} ${o ? "" : e["no-footer"]}`,
      ...i,
      children: /* @__PURE__ */ t(c, { children: [
        /* @__PURE__ */ n(G, {}),
        /* @__PURE__ */ t("div", { className: e.buttons, children: [
          /* @__PURE__ */ n(V, { className: e["back-button"] }),
          /* @__PURE__ */ n(D, { asChild: !0, children: /* @__PURE__ */ n(h, { "aria-label": "Close", size: "md", variant: "transparentSecondary", children: /* @__PURE__ */ n(A, { className: e["close-icon"] }) }) })
        ] }),
        /* @__PURE__ */ n("div", { className: e["sign-in-image"], children: /* @__PURE__ */ n(z, {}) }),
        /* @__PURE__ */ n(E, { onSuccess: r, children: ({ authMethod: l, step: a, Form: d }) => /* @__PURE__ */ t(c, { children: [
          /* @__PURE__ */ t("div", { className: e["title-desc-wrapper"], children: [
            a === "verification" ? /* @__PURE__ */ n(g, { as: "div", children: /* @__PURE__ */ n(_, {}) }) : /* @__PURE__ */ n(_, {}),
            a === "credentials" ? /* @__PURE__ */ n(g, { as: "div", children: /* @__PURE__ */ n(u, { authMethod: l }) }) : /* @__PURE__ */ n(u, { authMethod: l })
          ] }),
          /* @__PURE__ */ n("div", { className: e["sign-in-form"], children: d }),
          a === "credentials" && /* @__PURE__ */ n(H, { activeMethod: l })
        ] }) }),
        o && /* @__PURE__ */ n(J, { className: e.footer })
      ] })
    }
  ) });
}, mn = ({
  authMethods: i,
  children: o,
  open: s,
  setIsOpen: r,
  onSuccess: l
}) => {
  C("sign_in_modal");
  const a = o ? f(o, p) : !1, d = o ? f(o, m) : !1, S = !a && !d, I = x(() => ({ authMethods: i, onSuccess: l }), [i, l]);
  return /* @__PURE__ */ n(M.Provider, { value: I, children: /* @__PURE__ */ n(F, { open: s, onOpenChange: r, children: S ? /* @__PURE__ */ t(c, { children: [
    /* @__PURE__ */ n(p, { children: o }),
    /* @__PURE__ */ n(m, {})
  ] }) : /* @__PURE__ */ t(c, { children: [
    o,
    !d && /* @__PURE__ */ n(m, {})
  ] }) }) });
}, G = () => /* @__PURE__ */ t(g, { children: [
  /* @__PURE__ */ n(j, { asChild: !0, children: /* @__PURE__ */ n(_, { as: "span" }) }),
  /* @__PURE__ */ n(y, { asChild: !0, children: /* @__PURE__ */ n(u, { as: "span" }) })
] });
export {
  mn as SignInModal,
  m as SignInModalContent,
  p as SignInModalTrigger
};
