// @coinbase/cdp-react/dist/components/forms/EmailForm/index.js
import { jsxs as u, jsx as r } from "react/jsx-runtime";
import "@coinbase/cdp-hooks";
import { Form as c, FormSubmit as v } from "@radix-ui/react-form";
import "react";
import { Field as f } from "../Field/index.js";
import { withValidityState as h } from "../Input/index.js";
import { Button as _ } from "../../ui/Button/index.js";
import { ServerError as E } from "../../ui/ServerError/index.js";
import { isEmailInvalid as F } from "../../../utils/isEmailInvalid.js";
import '../../../assets/EmailForm.css';const x = "EmailForm-module__form___L0wVX", o = {
  form: x,
  "server-error": "EmailForm-module__server-error___0TSPx"
}, N = ({
  className: a = "",
  email: t,
  error: e,
  onEmailChange: m,
  isPending: l,
  submitLabel: s = "Continue",
  ...d
}) => {
  const i = {
    autoComplete: "email",
    type: "email",
    placeholder: "name@example.com",
    required: !0
  }, n = (p) => {
    m?.(p.target.value?.trim());
  };
  return /* @__PURE__ */ u(c, { className: `${o.form} ${a}`, ...d, children: [
    /* @__PURE__ */ r(
      f,
      {
        label: "Email address",
        name: "email",
        validators: [
          ["valueMissing", "Email is required"],
          ["typeMismatch", "Use a valid email address"],
          [F, "Use a valid email address"]
        ],
        children: h(
          m !== void 0 ? {
            ...i,
            value: t,
            onChange: n
          } : i
        )
      }
    ),
    e && /* @__PURE__ */ r(E, { error: e, className: o["server-error"] }),
    /* @__PURE__ */ r(v, { asChild: !0, children: /* @__PURE__ */ r(_, { type: "submit", isPending: l, children: s }) })
  ] });
};
export {
  N as EmailForm
};
