// @coinbase/cdp-react/dist/components/forms/Field/index.js
import { jsxs as n, jsx as r } from "react/jsx-runtime";
import { FormField as F, FormLabel as f, FormValidityState as u, FormControl as C, FormMessage as d } from "@radix-ui/react-form";
import "react";
import { Label as _ } from "../Label/index.js";
import { Error as o } from "../../ui/Error/index.js";
import '../../../assets/Field.css';const b = {
  "field-wrapper": "Field-module__field-wrapper___SCcbl"
}, S = ({
  children: l,
  className: t = "",
  inlineError: i,
  label: m,
  name: a,
  style: s,
  validators: p
}) => /* @__PURE__ */ n(F, { name: a, className: `${t} ${b["field-wrapper"]}`, style: s, children: [
  /* @__PURE__ */ r(f, { asChild: !0, children: /* @__PURE__ */ r(_, { children: m }) }),
  /* @__PURE__ */ r(u, { children: (e) => l && /* @__PURE__ */ r(C, { asChild: !0, children: l(e) }) }),
  p?.map(([e, c], h) => /* @__PURE__ */ r(d, { match: e, asChild: !0, children: /* @__PURE__ */ r(o, { children: c }) }, h)),
  i && /* @__PURE__ */ r(d, { asChild: !0, children: /* @__PURE__ */ r(o, { children: i }) })
] });
export {
  S as Field
};
