// @coinbase/cdp-react/dist/components/SendEvmTransactionButton/index.js
import { jsx as E } from "react/jsx-runtime";
import { useSendEvmTransaction as _, APIError as S } from "@coinbase/cdp-hooks";
import { u as b } from "../../chunks/useSendComponentCallOnce.BHZMuo6E.js";
import { useState as T, useCallback as y } from "react";
import { c as C } from "../../chunks/lite.1fxw3LjI.js";
import { Button as w } from "../ui/Button/index.js";
import { getMessageFromUnknownError as x } from "../../utils/getMessageFromUnknownError.js";
import '../../assets/SendEvmTransactionButton.css';const B = {
  "send-evm-transaction-button": "SendEvmTransactionButton-module__send-evm-transaction-button___74m57"
}, F = ({
  account: e,
  children: f,
  className: u = "",
  network: s,
  onClick: o,
  onError: a,
  onSuccess: i,
  pendingLabel: p,
  transaction: r,
  ...m
}) => {
  b("send_evm_transaction_button");
  const [v, c] = T(!1), { sendEvmTransaction: d } = _(), g = y(
    async (l) => {
      l.preventDefault(), o?.(l);
      try {
        if (!e || !s || !r) {
          const n = [];
          throw e || n.push("account"), s || n.push("network"), r || n.push("transaction"), new Error(`Missing required parameters: ${n.join(", ")}`);
        }
        c(!0);
        const { transactionHash: t } = await d({
          transaction: r,
          evmAccount: e,
          network: s
        });
        i?.(t);
      } catch (t) {
        let n;
        if (t instanceof S || t instanceof Error)
          n = t;
        else {
          const h = x(t);
          n = new Error(h);
        }
        a?.(n);
      } finally {
        c(!1);
      }
    },
    [e, s, o, a, i, d, r]
  );
  return /* @__PURE__ */ E(
    w,
    {
      className: m.asChild ? "" : C(B["send-evm-transaction-button"], u),
      isPending: v,
      pendingLabel: p || "Sending transaction...",
      onClick: g,
      ...m,
      children: f || "Send transaction"
    }
  );
};
export {
  F as SendEvmTransactionButton
};
