// @coinbase/cdp-react/dist/index.js
import "./chunks/useSendComponentCallOnce.BHZMuo6E.js";
import "react";
import "./version.js";
import { AuthButton as m } from "./components/AuthButton/index.js";
import { ALL_AUTH_METHODS as x, AUTH_METHODS as l, CDPReactProvider as i, OAUTH_METHODS as f, useAppConfig as s, useProviderName as u } from "./components/CDPReactProvider/index.js";
import { CopyAddress as c } from "./components/CopyAddress/index.js";
import { CopyEvmKeyButton as I } from "./components/CopyEvmKeyButton/index.js";
import { CopySolanaKeyButton as C } from "./components/CopySolanaKeyButton/index.js";
import { ExportWallet as h, ExportWalletCopyAddress as E, ExportWalletCopyKeyButton as S, ExportWalletFooter as T, E as k, a as F, b as L, c as B, d as W, ExportWalletTitle as y, ExportWalletWarning as b } from "./components/ExportWallet/index.js";
import { Fund as w, FundFooter as D } from "./components/Fund/index.js";
import { FundModal as H, FundModalContent as O, FundModalTrigger as R } from "./components/FundModal/index.js";
import { LinkAuth as _, LinkAuthError as K } from "./components/LinkAuth/index.js";
import { LinkAuthModal as N, LinkAuthModalContent as j, LinkAuthModalTrigger as G } from "./components/LinkAuthModal/index.js";
import { SendEvmTransactionButton as q } from "./components/SendEvmTransactionButton/index.js";
import { SendSolanaTransactionButton as J } from "./components/SendSolanaTransactionButton/index.js";
import { SignIn as Y } from "./components/SignIn/index.js";
import { useSignInReducer as $ } from "./components/SignIn/useSignInReducer.js";
import { SignInModal as ro, SignInModalContent as to, SignInModalTrigger as eo } from "./components/SignInModal/index.js";
import { SignOutButton as po } from "./components/SignOutButton/index.js";
import { ThemeProvider as ao, useTheme as xo } from "./components/ThemeProvider/index.js";
import { Button as io } from "./components/ui/Button/index.js";
import { Modal as so, ModalClose as uo, ModalContent as co, ModalDescription as go, ModalTitle as Io, ModalTrigger as Mo } from "./components/ui/Modal/index.js";
import { cssVariables as Ao } from "./theme/cssVariables.js";
import { theme as Eo } from "./theme/theme.js";
import { borderRadius as To, borderRadiusComponents as ko, borderRadiusSemantic as Fo, colors as Lo, colorsBase as Bo, colorsComponents as Wo, colorsSemantic as yo, font as bo, fontComponents as vo, fontSemantic as wo, tokens as Do } from "./theme/tokens.js";
import { flattenTokensObject as Ho, themeToCssVariables as Oo } from "./theme/utils.js";
import { IconAppleLogo as Uo } from "./icons/IconAppleLogo.js";
import { IconArrowLeft as Ko } from "./icons/IconArrowLeft.js";
import { IconArrowsUpDown as No } from "./icons/IconArrowsUpDown.js";
import { IconCheck as Go } from "./icons/IconCheck.js";
import { IconCheckCircle as qo } from "./icons/IconCheckCircle.js";
import { IconChevronDown as Jo } from "./icons/IconChevronDown.js";
import { IconCopy as Yo } from "./icons/IconCopy.js";
import { IconEnvelope as $o } from "./icons/IconEnvelope.js";
import { IconExclamationCircle as rr } from "./icons/IconExclamationCircle.js";
import { IconExclamationTriangle as er } from "./icons/IconExclamationTriangle.js";
import { IconGoogleLogo as pr } from "./icons/IconGoogleLogo.js";
import { IconLock as ar } from "./icons/IconLock.js";
import { IconPhone as lr } from "./icons/IconPhone.js";
import { IconXMark as fr } from "./icons/IconXMark.js";
import { IconMinus as ur } from "./icons/IconMinus.js";
import { IconPlus as cr } from "./icons/IconPlus.js";
import { clamp as Ir } from "./utils/clamp.js";
import { getMessageFromUnknownError as Cr } from "./utils/getMessageFromUnknownError.js";
import { isApiError as hr } from "./utils/isApiError.js";
import { isEmailInvalid as Sr } from "./utils/isEmailInvalid.js";
import { parseValuesFromPhoneNumber as kr } from "./utils/parseValuesFromPhoneNumber.js";
import { sendIframeMessage as Lr } from "./utils/sendIframeMessage.js";
import "@coinbase/cdp-core";
import { FundForm as Wr } from "./components/Fund/FundForm.js";
import { FundTitle as br } from "./components/Fund/FundTitle.js";
import { useFundContext as wr } from "./components/Fund/FundProvider.js";
import { a as Pr, u as Hr } from "./chunks/LinkAuthFlow.3HN888DV.js";
import { LinkAuthFlowBackButton as Rr } from "./components/LinkAuth/LinkAuthFlowBackButton.js";
import { LinkAuthItem as _r } from "./components/LinkAuth/LinkAuthItem.js";
import { LinkAuthItems as Vr } from "./components/LinkAuth/LinkAuthItems.js";
import { LinkAuthTitle as jr } from "./components/LinkAuth/LinkAuthTitle.js";
import { useLinkAuthContext as Xr } from "./components/LinkAuth/LinkAuthProvider.js";
import { SignInAuthMethodButtons as zr } from "./components/SignIn/SignInAuthMethodButtons.js";
import { SignInBackButton as Qr } from "./components/SignIn/SignInBackButton.js";
import { SignInDescription as Zr } from "./components/SignIn/SignInDescription.js";
import { SignInFooter as ot } from "./components/SignIn/SignInFooter.js";
import { SignInForm as tt } from "./components/SignIn/SignInForm.js";
import { SignInImage as nt } from "./components/SignIn/SignInImage.js";
import { SignInTitle as mt } from "./components/SignIn/SignInTitle.js";
import { useSignInContext as xt } from "./components/SignIn/SignInProvider.js";
export {
  x as ALL_AUTH_METHODS,
  l as AUTH_METHODS,
  m as AuthButton,
  io as Button,
  i as CDPReactProvider,
  c as CopyAddress,
  I as CopyEvmKeyButton,
  C as CopySolanaKeyButton,
  h as ExportWallet,
  E as ExportWalletCopyAddress,
  S as ExportWalletCopyKeyButton,
  T as ExportWalletFooter,
  k as ExportWalletModal,
  F as ExportWalletModalClose,
  L as ExportWalletModalContent,
  B as ExportWalletModalTitle,
  W as ExportWalletModalTrigger,
  y as ExportWalletTitle,
  b as ExportWalletWarning,
  w as Fund,
  D as FundFooter,
  Wr as FundForm,
  H as FundModal,
  O as FundModalContent,
  R as FundModalTrigger,
  br as FundTitle,
  Uo as IconAppleLogo,
  Ko as IconArrowLeft,
  No as IconArrowsUpDown,
  Go as IconCheck,
  qo as IconCheckCircle,
  Jo as IconChevronDown,
  Yo as IconCopy,
  $o as IconEnvelope,
  rr as IconExclamationCircle,
  er as IconExclamationTriangle,
  pr as IconGoogleLogo,
  ar as IconLock,
  ur as IconMinus,
  lr as IconPhone,
  cr as IconPlus,
  fr as IconXMark,
  _ as LinkAuth,
  K as LinkAuthError,
  Pr as LinkAuthFlow,
  Rr as LinkAuthFlowBackButton,
  _r as LinkAuthItem,
  Vr as LinkAuthItems,
  N as LinkAuthModal,
  j as LinkAuthModalContent,
  G as LinkAuthModalTrigger,
  jr as LinkAuthTitle,
  so as Modal,
  uo as ModalClose,
  co as ModalContent,
  go as ModalDescription,
  Io as ModalTitle,
  Mo as ModalTrigger,
  f as OAUTH_METHODS,
  q as SendEvmTransactionButton,
  J as SendSolanaTransactionButton,
  Y as SignIn,
  zr as SignInAuthMethodButtons,
  Qr as SignInBackButton,
  Zr as SignInDescription,
  ot as SignInFooter,
  tt as SignInForm,
  nt as SignInImage,
  ro as SignInModal,
  to as SignInModalContent,
  eo as SignInModalTrigger,
  mt as SignInTitle,
  po as SignOutButton,
  ao as ThemeProvider,
  To as borderRadius,
  ko as borderRadiusComponents,
  Fo as borderRadiusSemantic,
  Ir as clamp,
  Lo as colors,
  Bo as colorsBase,
  Wo as colorsComponents,
  yo as colorsSemantic,
  Ao as cssVariables,
  Ho as flattenTokensObject,
  bo as font,
  vo as fontComponents,
  wo as fontSemantic,
  Cr as getMessageFromUnknownError,
  hr as isApiError,
  Sr as isEmailInvalid,
  kr as parseValuesFromPhoneNumber,
  Lr as sendIframeMessage,
  Eo as theme,
  Oo as themeToCssVariables,
  Do as tokens,
  s as useAppConfig,
  wr as useFundContext,
  Xr as useLinkAuthContext,
  Hr as useLinkAuthFlow,
  u as useProviderName,
  xt as useSignInContext,
  $ as useSignInReducer,
  xo as useTheme
};
