// @coinbase/cdp-core/dist/web/index.web37.js
import "./index.web2.js";
import { useSendHookCallOnce as l } from "./index.web62.js";
import "./index.web3.js";
import { getCurrentUserSync as T } from "./index.web13.js";
import { getAccessToken as g, getAccessTokenExpiration as I } from "./index.web9.js";
import { DEFAULT_SECURE_IFRAME_BASE_PATH as S, SECURE_IFRAME_KEY_EXPORT_EVENT_TYPE as s, DEFAULT_SECURE_IFRAME_THEME as h } from "./index.web36.js";
import { validateSecureIframeEnvironment as f, createIframe as x, calculateSecureIframeHeight as A, buildKeyExportIframeUrl as U, getSecureIframeOrigin as k, isSecureIframeKeyExportMessage as b, sendSecureIframeMessage as E } from "./index.web35.js";
import { getConfig as w } from "./index.web49.js";
async function u(t) {
  const e = x(t.target), y = A(t.theme), c = w().secureIframeBasePath || S;
  e.setAttribute("allow", "clipboard-read; clipboard-write"), e.style.border = "none", e.style.display = "none", e.style.height = `${y}px`, e.src = U({ ...t, basePath: c });
  const i = k(c);
  let m = !1;
  const p = () => {
    m || (m = !0, window.removeEventListener("message", d), e.parentNode && e.parentNode.removeChild(e));
  }, d = async (a) => {
    if (!i || a.origin !== i) return;
    const n = a.data;
    if (b(n))
      switch (n.type) {
        case s.LISTENING: {
          const r = T();
          if (!r) {
            t.onStatusUpdate?.("error", "User not authenticated");
            return;
          }
          const o = await g(), _ = await I();
          E(e, {
            type: s.GET_PRIVATE_KEY,
            payload: {
              address: t.address,
              type: t.type,
              authState: {
                accessToken: o || "",
                expiresAt: _ || 0,
                user: r
              },
              theme: {
                ...h,
                ...t.theme || {}
              }
            }
          });
          break;
        }
        case s.STATUS: {
          const { status: r, message: o } = n.payload;
          r === "ready" && (e.style.display = "block"), r === "expired" && (e.style.display = "none", p()), t.onStatusUpdate?.(r, o);
          break;
        }
      }
  };
  return window.addEventListener("message", d), {
    iframe: e,
    cleanup: p,
    updateTheme: (a) => {
      E(e, {
        type: s.THEME,
        payload: { theme: a }
      });
    }
  };
}
async function N(t) {
  return l("create_evm_key_export_iframe"), f(), u({ ...t, type: "evm" });
}
async function O(t) {
  return l("create_solana_key_export_iframe"), f(), u({ ...t, type: "solana" });
}
export {
  N as createEvmKeyExportIframe,
  O as createSolanaKeyExportIframe
};
