// @coinbase/cdp-core/dist/web/index.web25.js
import { initiateMfaEnrollment as m } from "@coinbase/cdp-api-client";
import { Analytics as u } from "./index.web2.js";
import "./index.web3.js";
import { withAuth as c } from "./index.web44.js";
import { validateUserAuthState as l } from "./index.web53.js";
import { getConfig as a, getCoreAuthManager as p } from "./index.web49.js";
const S = async (o) => {
  if (u.sendActionCallEvent({ name: "initiate_mfa_enrollment" }), a().useMock)
    return {
      authUrl: "otpauth://totp/MockApp:mock@example.com?secret=MOCKTOTP&issuer=MockApp",
      secret: "MOCKTOTP"
    };
  const t = p(), { user: e, authState: n } = l(
    t.getUser(),
    t.getAuthState()
  ), i = await c({}, t, async ({ user: s }) => {
    const r = await m(
      a().projectId,
      s.userId,
      o.mfaMethod
    );
    return {
      authUrl: r.authUrl,
      secret: r.secret
    };
  });
  return await t.setAuthState({
    ...n,
    user: {
      ...e,
      mfaMethods: {
        enrollmentPromptedAt: (/* @__PURE__ */ new Date()).toISOString(),
        ...e.mfaMethods
      }
    }
  }), i;
};
export {
  S as initiateMfaEnrollment
};
