// @coinbase/cdp-react/dist/components/ui/Modal/index.js
import { jsx as o, jsxs as n } from "react/jsx-runtime";
import { Dialog as _, DialogTitle as p, DialogDescription as g, DialogPortal as M, DialogOverlay as D, DialogContent as f, DialogClose as u, DialogTrigger as y } from "@radix-ui/react-dialog";
import { c as e } from "../../../chunks/lite.1fxw3LjI.js";
import { forwardRef as v, useMemo as C } from "react";
import { useTheme as T } from "../../ThemeProvider/index.js";
import '../../../assets/Modal.css';const x = "Modal-module__modal___MUsZA", N = "Modal-module__overlay___-RqGQ", h = "Modal-module__spacer___dIfOA", a = {
  modal: x,
  overlay: N,
  spacer: h
}, j = v(
  ({ children: r, className: t = "", overlayClassName: c = "", overlayStyle: l, ...i }, d) => {
    const { cssVariables: s } = T(), m = C(
      () => ({ ...s, ...l }),
      [s, l]
    );
    return /* @__PURE__ */ o(M, { children: /* @__PURE__ */ n(D, { className: e(a.overlay, c), style: m, children: [
      /* @__PURE__ */ o("div", { className: a.spacer }),
      /* @__PURE__ */ o(f, { ...i, ref: d, className: e(a.modal, t), children: r })
    ] }) });
  }
);
j.displayName = "ModalContent";
const w = _, G = u, I = g, P = p, Q = y;
export {
  w as Modal,
  G as ModalClose,
  j as ModalContent,
  I as ModalDescription,
  P as ModalTitle,
  Q as ModalTrigger
};
