// @coinbase/cdp-react/dist/components/forms/AmountInput/index.js
import { jsxs as s, jsx as a } from "react/jsx-runtime";
import { forwardRef as g, useRef as u, useMemo as j, useCallback as q, useEffect as h } from "react";
import { useAmountInput as z } from "./useAmountInput.js";
import { useInputScale as M } from "./useInputScale.js";
import '../../../assets/AmountInput.css';const w = "AmountInput-module__input___Af-R1", E = "AmountInput-module__visible___qmvT0", G = "AmountInput-module__hidden___cO0sQ", O = "AmountInput-module__unit___-zmGz", n = {
  "amount-input": "AmountInput-module__amount-input___24NlC",
  input: w,
  "amount-unit": "AmountInput-module__amount-unit___48d60",
  visible: E,
  hidden: G,
  unit: O
}, P = g(
  ({
    amount: I,
    className: A = "",
    unit: p,
    decimalPlaces: r = 2,
    locale: R,
    max: v,
    min: C,
    setAmount: N,
    onBlur: y,
    onPaste: $,
    placeholder: c = "0",
    ...b
  }, t) => {
    const o = u(null), i = u(null), m = u(null), d = u(null), _ = u(null), e = j(() => p?.toUpperCase(), [p]), { handleBlur: x, handleChange: B, handleBeforeInput: k, handlePaste: S, inputValue: l } = z(
      {
        amount: I,
        decimalPlaces: r,
        locale: R,
        max: v,
        min: C,
        setAmount: N,
        onBlur: y,
        onPaste: $
      }
    ), { updateScale: f } = M({
      containerRef: i,
      measureRef: m,
      valueRef: d,
      inputRef: o,
      scaleWrapperRef: _
    }), U = q(() => {
      i.current?.querySelector("input[type='text']")?.focus();
    }, []);
    return h(() => {
      f();
    }, [l, e, r, f]), h(() => {
      t && (typeof t == "function" ? t(o.current) : t.current = o.current);
    }, [t]), /* @__PURE__ */ s("span", { ref: i, className: n["amount-input"], children: [
      /* @__PURE__ */ s("span", { ref: _, className: `${n["amount-unit"]} ${n.visible}`, children: [
        /* @__PURE__ */ a(
          "input",
          {
            className: `${n.input} ${A}`,
            inputMode: "decimal",
            onBlur: x,
            onChange: B,
            onBeforeInput: k,
            onPaste: S,
            placeholder: c,
            ref: o,
            type: "text",
            value: l,
            ...b
          }
        ),
        e && /* @__PURE__ */ a("span", { className: n.unit, onClick: U, children: e })
      ] }),
      /* @__PURE__ */ s(
        "span",
        {
          ref: m,
          "aria-hidden": "true",
          className: `${n["amount-unit"]} ${n.hidden}`,
          children: [
            /* @__PURE__ */ s("span", { ref: d, children: [
              l || c,
              " "
            ] }),
            e && /* @__PURE__ */ a("span", { children: e })
          ]
        }
      )
    ] });
  }
);
P.displayName = "AmountInput";
export {
  P as AmountInput
};
