// @coinbase/cdp-react/dist/components/LinkAuth/index.js
import { jsx as r, jsxs as h, Fragment as w } from "react/jsx-runtime";
import { useLinkOAuth as C } from "@coinbase/cdp-hooks";
import { u as I } from "../../chunks/useSendComponentCallOnce.BHZMuo6E.js";
import { useRef as d, useState as N, useCallback as u } from "react";
import "../CDPReactProvider/index.js";
import { useOauthStatusModal as P } from "../OAuthStatusModal/index.js";
import { Banner as U } from "../ui/Banner/index.js";
import { getMessageFromUnknownError as D } from "../../utils/getMessageFromUnknownError.js";
import "libphonenumber-js";
import { toOAuthProviderType as H } from "../../utils/toOAuthProviderType.js";
import { L as M, a as v } from "../../chunks/LinkAuthFlow.3HN888DV.js";
import { u as mt } from "../../chunks/LinkAuthFlow.3HN888DV.js";
import { LinkAuthFlowBackButton as V } from "./LinkAuthFlowBackButton.js";
import { LinkAuthItem as dt } from "./LinkAuthItem.js";
import { LinkAuthItems as ft } from "./LinkAuthItems.js";
import { useLinkAuthContext as y, LinkAuthProvider as j } from "./LinkAuthProvider.js";
import { LinkAuthTitle as K } from "./LinkAuthTitle.js";
import { methodToView as g } from "./utils.js";
import '../../assets/LinkAuth.css';const W = "LinkAuth-module__header___0UuxP", $ = "LinkAuth-module__error___XUo4W", k = {
  "link-auth": "LinkAuth-module__link-auth___kPJao",
  header: W,
  error: $
}, b = ({ role: o = "alert", ...c }) => {
  const { state: i } = y(), { error: s } = i;
  return s ? /* @__PURE__ */ r(U, { variant: "error", role: o, ...c, children: s }) : null;
}, G = ({ children: o, className: c = "", onLinkSuccess: i, ...s }) => {
  const { linkOAuth: f } = C(), { persistOAuthState: p } = P(), { state: a, dispatch: n } = y(), { methodToLink: O } = a, L = d(null), l = d(null), x = g(O), A = d(x), [E, m] = N("left"), e = u(
    (t) => {
      A.current !== t && (A.current = t, L.current?.transition.toggle(t), t === "list" && n({ type: "RESET_STATE" }), (t === "email" || t === "sms") && n({ type: "LINK_AUTH_METHOD", payload: { method: t } }));
    },
    [n]
  ), B = u(() => {
    if (m("right"), l.current?.canGoBack) {
      l.current.goBack();
      return;
    }
    e("list");
  }, [e]), F = u(() => {
    m("left"), i?.(a.methodToLink), setTimeout(() => {
      e("list");
    }, 800);
  }, [e, a.methodToLink, i]), R = u(
    async (t) => {
      if (m("left"), t?.startsWith("oauth:")) {
        const T = H(t);
        if (T)
          try {
            await f(T), p(!0);
          } catch (S) {
            n({
              type: "LINK_AUTH_METHOD_ERROR",
              payload: { error: D(S) }
            });
          }
      } else
        e(g(t));
    },
    [f, e, p, n]
  ), _ = typeof o == "function" ? o(a) : o;
  return /* @__PURE__ */ r("div", { className: `${k["link-auth"]} ${c}`, ...s, children: /* @__PURE__ */ h(
    M,
    {
      direction: E,
      onBack: B,
      onLink: R,
      onLinkSuccess: F,
      signInRef: l,
      transitionRef: L,
      children: [
        _,
        !_ && /* @__PURE__ */ h(w, { children: [
          /* @__PURE__ */ h("div", { className: k.header, children: [
            /* @__PURE__ */ r(K, {}),
            /* @__PURE__ */ r(V, {})
          ] }),
          /* @__PURE__ */ r("div", { className: k.error, children: /* @__PURE__ */ r(b, {}) }),
          /* @__PURE__ */ r(v, {})
        ] })
      ]
    }
  ) });
}, ut = (o) => (I("manage_auth"), /* @__PURE__ */ r(j, { children: /* @__PURE__ */ r(G, { ...o }) }));
export {
  ut as LinkAuth,
  b as LinkAuthError,
  v as LinkAuthFlow,
  V as LinkAuthFlowBackButton,
  dt as LinkAuthItem,
  ft as LinkAuthItems,
  K as LinkAuthTitle,
  y as useLinkAuthContext,
  mt as useLinkAuthFlow
};
