// @coinbase/cdp-react/dist/components/LinkAuth/LinkAuthFlowBackButton.js
import { jsx as t } from "react/jsx-runtime";
import { c as k } from "../../chunks/lite.1fxw3LjI.js";
import "react";
import { Button as p } from "../ui/Button/index.js";
import { IconArrowLeft as f } from "../../icons/IconArrowLeft.js";
import { b as w } from "../../chunks/LinkAuthFlow.3HN888DV.js";
import { useLinkAuthContext as _ } from "./LinkAuthProvider.js";
import "../CDPReactProvider/index.js";
import { methodToView as b } from "./utils.js";
import '../../assets/LinkAuthFlowBackButton.css';const h = "LinkAuthFlowBackButton-module__button___wPKfW", B = "LinkAuthFlowBackButton-module__icon___OUmzY", o = {
  button: h,
  icon: B
}, T = ({
  ["aria-label"]: n = "Back",
  children: r,
  className: i = "",
  size: e = "md",
  onClick: c,
  variant: m = "transparentSecondary",
  view: a,
  ...s
}) => {
  const { back: u } = w(), { state: l } = _();
  return (a || b(l.methodToLink)) === "list" ? null : /* @__PURE__ */ t(
    p,
    {
      type: "button",
      "aria-label": n,
      className: k(o.button, i),
      onClick: c ?? u,
      size: e,
      variant: m,
      ...s,
      children: r || /* @__PURE__ */ t(f, { className: o.icon })
    }
  );
};
export {
  T as LinkAuthFlowBackButton
};
