// @coinbase/cdp-react/dist/components/LinkAuth/LinkAuthItem.js
import { jsxs as _, jsx as t } from "react/jsx-runtime";
import "react";
import { Button as o } from "../ui/Button/index.js";
import { IconMinus as u } from "../../icons/IconMinus.js";
import { IconPlus as s } from "../../icons/IconPlus.js";
import "../CDPReactProvider/index.js";
import '../../assets/LinkAuthItem.css';const d = "LinkAuthItem-module__item___NE3CT", l = "LinkAuthItem-module__button___nM5Tc", i = {
  item: d,
  "item-icon": "LinkAuthItem-module__item-icon___ZIQny",
  "item-text": "LinkAuthItem-module__item-text___-5DVL",
  "item-action": "LinkAuthItem-module__item-action___4wOIN",
  button: l
}, v = ({
  authMethod: h,
  userAlias: a,
  icon: m,
  isLinked: r,
  isPending: n,
  label: e,
  onLink: c
}) => /* @__PURE__ */ _("div", { className: i.item, children: [
  m && /* @__PURE__ */ t("div", { className: i["item-icon"], children: m }),
  /* @__PURE__ */ t("div", { className: i["item-text"], children: a || e }),
  /* @__PURE__ */ t("div", { className: i["item-action"], children: r ? /* @__PURE__ */ t(
    o,
    {
      variant: "secondary",
      className: i.button,
      isPending: n,
      disabled: !0,
      "aria-label": `Unlink ${e}`,
      children: /* @__PURE__ */ t(u, { "aria-hidden": "true" })
    }
  ) : /* @__PURE__ */ t(
    o,
    {
      variant: "primary",
      className: i.button,
      onClick: c,
      "aria-label": `Link ${e}`,
      isPending: n,
      children: /* @__PURE__ */ t(s, { "aria-hidden": "true" })
    }
  ) })
] });
export {
  v as LinkAuthItem
};
