// @coinbase/cdp-react/dist/components/LinkAuth/LinkAuthItems.js
import { jsx as l } from "react/jsx-runtime";
import { c as I } from "../../chunks/lite.1fxw3LjI.js";
import { useMemo as b } from "react";
import "../CDPReactProvider/index.js";
import { IconAppleLogo as d } from "../../icons/IconAppleLogo.js";
import { IconEnvelope as f } from "../../icons/IconEnvelope.js";
import { IconGoogleColorLogo as L } from "../../icons/IconGoogleColorLogo.js";
import { IconPhone as g } from "../../icons/IconPhone.js";
import { IconXLogo as _ } from "../../icons/IconXLogo.js";
import { LinkAuthItem as A } from "./LinkAuthItem.js";
import { useLinkAuthContext as P } from "./LinkAuthProvider.js";
import '../../assets/LinkAuthItems.css';const k = "LinkAuthItems-module__list___5AimM", x = {
  list: k
}, M = {
  email: { Icon: /* @__PURE__ */ l(f, { "aria-label": "Email" }), label: "Email" },
  sms: { Icon: /* @__PURE__ */ l(g, { "aria-label": "Phone" }), label: "Phone" },
  "oauth:google": { Icon: /* @__PURE__ */ l(L, { "aria-label": "Google" }), label: "Google" },
  "oauth:apple": { Icon: /* @__PURE__ */ l(d, { "aria-label": "Apple" }), label: "Apple" },
  "oauth:x": { Icon: /* @__PURE__ */ l(_, { "aria-label": "X" }), label: "X" }
}, D = ({ children: m, className: s = "", onLink: e, ...c }) => {
  const { state: p, dispatch: i } = P(), { authMethods: a, isPending: n, methodToLink: r } = p, h = b(() => a.map((o) => {
    const { Icon: t, label: u } = M[o.method];
    return {
      ...o,
      icon: t,
      label: u,
      onLink: async () => {
        i({ type: "LINK_AUTH_METHOD", payload: { method: o.method } }), await e?.(o.method);
      }
    };
  }), [a, i, e]);
  return /* @__PURE__ */ l("ul", { className: I(x.list, s), ...c, children: h.map(({ method: o, ...t }) => /* @__PURE__ */ l("li", { children: m?.({
    ...t,
    authMethod: o,
    isPending: n && r === o
  }) || /* @__PURE__ */ l(
    A,
    {
      ...t,
      authMethod: o,
      isPending: n && r === o
    }
  ) }, o)) });
};
export {
  D as LinkAuthItems
};
