// @coinbase/cdp-react/dist/components/SignIn/flows/SignInWithEmail.js
import { jsxs as m, Fragment as a, jsx as o } from "react/jsx-runtime";
import { useIsSignedIn as L, useSignInWithEmail as j, useLinkEmail as k, useVerifyEmailOTP as H } from "@coinbase/cdp-hooks";
import { useRef as g, useLayoutEffect as M } from "react";
import { EmailForm as V } from "../../forms/EmailForm/index.js";
import { OTPForm as q } from "../../forms/OTPForm/index.js";
import { SwitchSlideTransition as z } from "../../ui/SwitchSlideTransition/index.js";
import { useEmailForm as A } from "../hooks/useEmailForm.js";
import { useOTPForm as B } from "../hooks/useOTPForm.js";
import { SignInCredentials as G } from "../SignInCredentials.js";
import { useSignInContext as S } from "../SignInProvider.js";
import "../../CDPReactProvider/index.js";
const J = ["email", "otp"], N = ({
  step: n,
  autoFocus: r,
  onSuccess: E,
  children: c
}) => {
  const l = g(null), h = 6, { isSignedIn: P } = L(), { state: i } = S(), { signInWithEmail: T } = j(), { linkEmail: O } = k(), { verifyEmailOTP: I } = H(), u = n || i.step, t = u === "credentials" ? "email" : "otp", d = g(t), { setEmail: v, submitEmail: f } = A({
    submit: (e) => P ? O(e) : T({ email: e })
  }), { resendCountdown: b, resetOTP: R, setOTP: w, startResendTimer: F, submitOtp: W } = B({
    passwordLength: h,
    submit: (e) => I({ flowId: i.flowId, otp: e })
  }), p = () => {
    F(60);
  }, C = () => {
    R(), f({ email: i.email, onSuccess: p });
  }, x = (e) => {
    e.preventDefault(), f({ email: i.email, onSuccess: p });
  }, y = (e) => {
    e.preventDefault(), W({ otp: i.otp, onSuccess: E });
  };
  return M(() => {
    d.current !== t && (l.current?.transition.toggle(t), d.current = t);
  }, [t]), /* @__PURE__ */ o(
    z,
    {
      autoFocus: r,
      animateHeight: !1,
      items: J,
      initialEntered: !0,
      direction: t === "otp" ? "left" : "right",
      transitionRef: l,
      children: ({ itemKey: e, ...D }) => {
        let s = null;
        return e === "email" && (s = /* @__PURE__ */ o(
          V,
          {
            email: i.email,
            error: t === "email" && i.error || "",
            isPending: i.isPending,
            onEmailChange: v,
            onSubmit: x
          }
        )), e === "otp" && (s = /* @__PURE__ */ o(
          q,
          {
            canResetOTP: i.canResetOTP,
            error: t === "otp" && i.error || "",
            isPending: i.isPending,
            onOTPChange: w,
            onResendOTP: C,
            onSubmit: y,
            otp: i.otp,
            resendCountdown: b,
            successMessage: i.isSuccess ? "Success!" : void 0
          }
        )), /* @__PURE__ */ o("div", { ...D, children: c ? c({ step: u, Form: s }) : s });
      }
    }
  );
}, Q = ({ step: n }) => /* @__PURE__ */ m(a, { children: [
  n === "credentials" && "Sign in",
  n === "verification" && "Enter verification code"
] }), U = ({ step: n }) => {
  const { state: r } = S();
  return /* @__PURE__ */ m(a, { children: [
    n === "credentials" && "We’ll send you a verification code via email.",
    n === "verification" && /* @__PURE__ */ m(a, { children: [
      "Enter the 6 digit code sent to ",
      /* @__PURE__ */ o(G, { children: r.email })
    ] })
  ] });
}, si = {
  description: U,
  forms: N,
  title: Q
};
export {
  N as SignInWithEmail,
  U as SignInWithEmailDescription,
  Q as SignInWithEmailTitle,
  si as config
};
