// @coinbase/cdp-react/dist/components/SignIn/flows/SignInWithOAuth.js
import { jsxs as h, Fragment as a, jsx as e } from "react/jsx-runtime";
import "@coinbase/cdp-hooks";
import { useRef as A, useCallback as S, useEffect as v } from "react";
import { useAppConfig as m } from "../../CDPReactProvider/index.js";
import { useSignInWithOAuth as O } from "../hooks/useSignInWithOAuth.js";
import { Button as W } from "../../ui/Button/index.js";
import { ServerError as I } from "../../ui/ServerError/index.js";
import { capitalize as p } from "../../../utils/capitalize.js";
import { toOAuthProviderType as f } from "../../../utils/toOAuthProviderType.js";
const y = ({ step: i, children: r }) => {
  const { authMethods: n } = m(), d = n?.[0], t = f(d), { signInWithOAuth: s, isPending: l, error: u } = O(), o = A(t), g = S(() => {
    t && s(t);
  }, [t, s]);
  v(() => {
    t && (o.current = t);
  }, [t]);
  const c = o.current ? /* @__PURE__ */ h(a, { children: [
    /* @__PURE__ */ h(W, { fullWidth: !0, onClick: g, type: "submit", isPending: l, children: [
      "Continue with ",
      p(o.current)
    ] }),
    u && /* @__PURE__ */ e("div", { style: { marginTop: "0.5em" }, children: /* @__PURE__ */ e(I, { error: u }) })
  ] }) : null;
  return /* @__PURE__ */ e("div", { children: r ? r({ step: i, Form: c }) : c });
}, C = () => "Sign in", M = () => {
  const { authMethods: i } = m(), r = i?.[0], n = f(r);
  return `Sign in with your ${p(n ?? "")} account.`;
}, R = {
  description: M,
  forms: y,
  title: C
};
export {
  y as SignInWithOAuth,
  M as SignInWithOAuthDescription,
  C as SignInWithOAuthTitle,
  R as config
};
