// @coinbase/cdp-react/dist/components/SignIn/flows/SignInWithSms.js
import { jsxs as h, Fragment as p, jsx as r } from "react/jsx-runtime";
import { useIsSignedIn as H, useSignInWithSms as U, useLinkSms as q, useVerifySmsOTP as z } from "@coinbase/cdp-hooks";
import "libphonenumber-js";
import { useRef as g, useState as N, useEffect as A, useLayoutEffect as B, useMemo as G } from "react";
import { OTPForm as J } from "../../forms/OTPForm/index.js";
import { PhoneNumberForm as Q } from "../../forms/PhoneNumberForm/index.js";
import { SwitchSlideTransition as X } from "../../ui/SwitchSlideTransition/index.js";
import { formatPhoneNumber as Y } from "../../../utils/formatPhoneNumber.js";
import { parseValuesFromPhoneNumber as Z } from "../../../utils/parseValuesFromPhoneNumber.js";
import { useOTPForm as _ } from "../hooks/useOTPForm.js";
import { usePhoneNumberForm as $ } from "../hooks/usePhoneNumberForm.js";
import { SignInCredentials as K } from "../SignInCredentials.js";
import { useSignInContext as T } from "../SignInProvider.js";
import "../../CDPReactProvider/index.js";
const ee = ["phoneNumber", "otp"], ne = ({
  step: o,
  autoFocus: s,
  onSuccess: i,
  children: a
}) => {
  const d = g(null), O = 6, { isSignedIn: v } = H(), { state: n } = T(), { signInWithSms: C } = U(), { linkSms: I } = q(), { verifySmsOTP: y } = z(), [c, F] = N("US"), [m, f] = N({
    value: "",
    e164: ""
  }), l = o || n.step, t = l === "credentials" ? "phoneNumber" : "otp", b = g(t), { setPhoneNumber: R, submitPhoneNumber: S } = $({
    submit: (e) => v ? I(e) : C({ phoneNumber: e })
  }), { resendCountdown: w, resetOTP: E, setOTP: W, startResendTimer: x, submitOtp: j } = _({
    passwordLength: O,
    submit: (e) => y({ flowId: n.flowId, otp: e })
  }), D = (e) => {
    e.value !== m.value && (f(e), R(e.e164));
  };
  A(() => {
    if (n.phoneNumber === m.e164)
      return;
    const e = Z(n.phoneNumber, c);
    f(e);
  }, [n.phoneNumber, c, m.e164]);
  const P = () => {
    x(60);
  }, L = () => {
    E(), S({ phoneNumber: n.phoneNumber, onSuccess: P });
  }, k = (e) => {
    e.preventDefault(), S({ phoneNumber: n.phoneNumber, onSuccess: P });
  }, M = (e) => {
    e.preventDefault(), j({ otp: n.otp, onSuccess: i });
  };
  return B(() => {
    b.current !== t && (d.current?.transition.toggle(t), b.current = t);
  }, [t]), /* @__PURE__ */ r(
    X,
    {
      autoFocus: s ? "input[type='tel']" : !1,
      animateHeight: !1,
      items: ee,
      initialEntered: !0,
      direction: t === "otp" ? "left" : "right",
      transitionRef: d,
      children: ({ itemKey: e, ...V }) => {
        let u = null;
        return e === "phoneNumber" && (u = /* @__PURE__ */ r(
          Q,
          {
            countryCode: c,
            error: t === "phoneNumber" && n.error || "",
            isPending: n.isPending,
            onCountryCodeChange: F,
            onPhoneNumberChange: D,
            onSubmit: k,
            phoneNumber: m
          }
        )), e === "otp" && (u = /* @__PURE__ */ r(
          J,
          {
            canResetOTP: n.canResetOTP,
            error: t === "otp" && n.error || "",
            isPending: n.isPending,
            onOTPChange: W,
            onResendOTP: L,
            onSubmit: M,
            otp: n.otp,
            resendCountdown: w,
            successMessage: n.isSuccess ? "Success!" : void 0
          }
        )), /* @__PURE__ */ r("div", { ...V, children: a ? a({ step: l, Form: u }) : u });
      }
    }
  );
}, te = ({ step: o }) => /* @__PURE__ */ h(p, { children: [
  o === "credentials" && "Sign in",
  o === "verification" && "Enter verification code"
] }), oe = ({ step: o }) => {
  const { state: s } = T(), i = G(
    () => Y(s.phoneNumber),
    [s.phoneNumber]
  );
  return /* @__PURE__ */ h(p, { children: [
    o === "credentials" && "We’ll send you a verification code via text.",
    o === "verification" && /* @__PURE__ */ h(p, { children: [
      "Enter the 6 digit code sent to",
      " ",
      i ? /* @__PURE__ */ r(K, { children: i }) : "your phone number"
    ] })
  ] });
}, Pe = {
  description: oe,
  forms: ne,
  title: te
};
export {
  ne as SignInWithSms,
  oe as SignInWithSmsDescription,
  te as SignInWithSmsTitle,
  Pe as config
};
