// @coinbase/cdp-react/dist/components/SignIn/index.js
import { jsx as n, jsxs as r, Fragment as c } from "react/jsx-runtime";
import { u as d } from "../../chunks/useSendComponentCallOnce.BHZMuo6E.js";
import "react";
import { useAppConfig as u } from "../CDPReactProvider/index.js";
import { SignInAuthMethodButtons as f } from "./SignInAuthMethodButtons.js";
import { SignInBackButton as I } from "./SignInBackButton.js";
import { SignInDescription as S } from "./SignInDescription.js";
import { SignInFooter as b } from "./SignInFooter.js";
import { SignInForm as k } from "./SignInForm.js";
import { SignInImage as w } from "./SignInImage.js";
import { SignInProvider as C, useSignInContext as h } from "./SignInProvider.js";
import { SignInTitle as v } from "./SignInTitle.js";
import "@coinbase/cdp-hooks";
import '../../assets/SignIn.css';const F = "SignIn-module__footer___3Gkbc", t = {
  "sign-in": "SignIn-module__sign-in___cYpee",
  "title-desc-wrapper": "SignIn-module__title-desc-wrapper___VpPcA",
  "back-button": "SignIn-module__back-button___bmE-y",
  "back-button-wrapper": "SignIn-module__back-button-wrapper___Q9FA1",
  footer: F
}, N = ({
  className: e = "",
  children: o,
  onSuccess: p,
  ...a
}) => {
  const { state: _ } = h(), { showCoinbaseFooter: g } = u(), s = typeof o == "function" ? o(_) : o;
  return /* @__PURE__ */ r("div", { className: `${t["sign-in"]} ${e}`, ...a, children: [
    s,
    !s && /* @__PURE__ */ r(c, { children: [
      /* @__PURE__ */ n("div", { className: t["back-button-wrapper"], children: /* @__PURE__ */ n(I, { className: t["back-button"] }) }),
      /* @__PURE__ */ n(w, {}),
      /* @__PURE__ */ n(k, { onSuccess: p, children: ({ authMethod: i, step: m, Form: l }) => /* @__PURE__ */ r(c, { children: [
        /* @__PURE__ */ r("div", { className: t["title-desc-wrapper"], children: [
          /* @__PURE__ */ n(v, { step: m, authMethod: i }),
          /* @__PURE__ */ n(S, { step: m, authMethod: i })
        ] }),
        l,
        m === "credentials" && /* @__PURE__ */ n(f, { activeMethod: i })
      ] }) }),
      g && /* @__PURE__ */ n(b, { className: t.footer })
    ] })
  ] });
}, V = ({ authMethods: e, ...o }) => (d("sign_in"), /* @__PURE__ */ n(C, { authMethods: e, children: /* @__PURE__ */ n(N, { ...o }) }));
export {
  V as SignIn,
  f as SignInAuthMethodButtons,
  I as SignInBackButton,
  S as SignInDescription,
  b as SignInFooter,
  k as SignInForm,
  w as SignInImage,
  v as SignInTitle,
  h as useSignInContext
};
