// @coinbase/cdp-react/dist/components/SignIn/SignInForm.js
import { jsx as a } from "react/jsx-runtime";
import { useRef as d, useState as I, useLayoutEffect as p } from "react";
import "../CDPReactProvider/index.js";
import { SwitchFadeTransition as T } from "../ui/SwitchFadeTransition/index.js";
import { a as y } from "../../chunks/index.Bvudzh_y.js";
import { useSignInContext as v } from "./SignInProvider.js";
import "@coinbase/cdp-hooks";
import '../../assets/SignInForm.css';const E = {
  "auth-method-wrapper": "SignInForm-module__auth-method-wrapper___uIOAB"
}, B = ({
  autoFocus: l = !0,
  onSuccess: f,
  step: M,
  children: g,
  ...S
}) => {
  const { state: r, dispatch: c } = v(), { authMethods: e } = r, w = M || r.step, m = d(r.authMethod), i = d(e), u = d(null), [_, A] = I(e || []);
  return p(() => {
    let t = r.authMethod;
    const n = !e?.includes(r.authMethod), o = i.current && i.current.length === e?.length && i.current.some((h, s) => h !== e?.[s]);
    if ((n || o) && (t = e?.[0]), t && t !== r.authMethod) {
      c({
        type: "SET_AUTH_METHOD",
        payload: { authMethod: t }
      });
      return;
    }
    u.current?.transition && t && m.current !== t && (u.current.transition.toggle(t), m.current = t);
  }, [e, c, r.authMethod]), p(() => {
    i.current = e, A((t) => {
      const n = e?.filter((o) => !t.includes(o)) || [];
      return n.length ? [...t, ...n] : t;
    });
  }, [e]), /* @__PURE__ */ a(
    T,
    {
      animateHeight: !0,
      timeout: 250,
      items: _,
      initialEntered: !0,
      transitionRef: u,
      ...S,
      children: ({ itemKey: t, ...n }) => {
        const o = y[t].forms;
        return /* @__PURE__ */ a("div", { ...n, className: E["auth-method-wrapper"], children: /* @__PURE__ */ a(o, { autoFocus: l, step: w, onSuccess: f, children: ({ step: h, Form: s }) => g?.({ step: h, authMethod: t, Form: s }) || s }) });
      }
    }
  );
};
export {
  B as SignInForm
};
