// @coinbase/cdp-react/dist/components/SignIn/SignInAuthMethodButtons.js
import { jsx as n, jsxs as r } from "react/jsx-runtime";
import { useCallback as v, useMemo as y } from "react";
import "../CDPReactProvider/index.js";
import { useSignInWithOAuth as S } from "./hooks/useSignInWithOAuth.js";
import { useSignInContext as _ } from "./SignInProvider.js";
import { Button as C } from "../ui/Button/index.js";
import { ServerError as M } from "../ui/ServerError/index.js";
import { IconAppleLogo as k } from "../../icons/IconAppleLogo.js";
import { IconEnvelope as B } from "../../icons/IconEnvelope.js";
import { IconGoogleLogo as E } from "../../icons/IconGoogleLogo.js";
import { IconPhone as P } from "../../icons/IconPhone.js";
import { IconXLogo as O } from "../../icons/IconXLogo.js";
import { toOAuthProviderType as b } from "../../utils/toOAuthProviderType.js";
import '../../assets/SignInAuthMethodButtons.css';const w = "SignInAuthMethodButtons-module__divider___zphEh", o = {
  "auth-method-buttons": "SignInAuthMethodButtons-module__auth-method-buttons___jYEH7",
  "auth-btn": "SignInAuthMethodButtons-module__auth-btn___o1a09",
  "auth-btn-icon": "SignInAuthMethodButtons-module__auth-btn-icon___VjvFY",
  "auth-btn-label": "SignInAuthMethodButtons-module__auth-btn-label___mEWd3",
  divider: w
}, g = {
  email: {
    label: "Continue with email",
    icon: /* @__PURE__ */ n(B, {})
  },
  sms: {
    label: "Continue with phone",
    icon: /* @__PURE__ */ n(P, {})
  },
  "oauth:google": {
    label: "Continue with Google",
    icon: /* @__PURE__ */ n(E, {})
  },
  "oauth:apple": {
    label: "Continue with Apple",
    icon: /* @__PURE__ */ n(k, {})
  },
  "oauth:x": {
    label: "Continue with X",
    icon: /* @__PURE__ */ n(O, {})
  }
}, U = ({ activeMethod: u }) => {
  const {
    state: { authMethods: a }
  } = _(), {
    signInWithOAuth: l,
    isPending: s,
    error: h,
    oauthProvider: c
  } = S(), m = v(
    async (t) => {
      const e = b(t);
      e && l(e);
    },
    [l]
  ), { dispatch: d } = _(), i = y(() => (a || ["email"]).map((t) => {
    if (!g[t])
      return null;
    const { label: e, icon: f } = g[t], p = t.startsWith("oauth:"), A = p ? () => m(t) : () => d({ type: "SET_AUTH_METHOD", payload: { authMethod: t } }), I = p && c.current === b(t) && s;
    return {
      key: t,
      label: e,
      icon: f,
      onClick: A,
      isPending: I
    };
  }).filter((t) => t !== null), [a, d, m, s, c]);
  return !i.length || i.length === 1 && i[0].key === u ? null : /* @__PURE__ */ r("div", { className: o["auth-method-buttons"], children: [
    /* @__PURE__ */ r("div", { className: o.divider, children: [
      /* @__PURE__ */ n("hr", {}),
      /* @__PURE__ */ n("span", { children: "or" })
    ] }),
    i.map((t) => t.key === u ? null : /* @__PURE__ */ r(
      C,
      {
        "aria-label": t.ariaLabel,
        className: o["auth-btn"],
        type: "button",
        variant: "secondary",
        onClick: t.onClick,
        isPending: t.isPending,
        children: [
          /* @__PURE__ */ n("span", { className: o["auth-btn-icon"], children: t.icon }),
          /* @__PURE__ */ n("span", { className: o["auth-btn-label"], children: t.label })
        ]
      },
      t.key
    )),
    h && /* @__PURE__ */ n(M, { error: h })
  ] });
};
export {
  U as SignInAuthMethodButtons
};
