// @coinbase/cdp-react/dist/components/SignIn/SignInProvider.js
import { jsx as h } from "react/jsx-runtime";
import { createContext as m, useContext as l, useMemo as i } from "react";
import { useAppConfig as f } from "../CDPReactProvider/index.js";
import "@coinbase/cdp-hooks";
import { useSignInReducer as p } from "./useSignInReducer.js";
const a = {
  authMethod: "email",
  authMethods: ["email"],
  canResetOTP: !1,
  email: "",
  error: null,
  flowId: "",
  isPending: !1,
  isSuccess: !1,
  otp: "",
  phoneNumber: "",
  step: "credentials"
}, u = m({
  state: a,
  dispatch: () => {
  }
}), M = () => {
  const t = l(u);
  if (!t)
    throw new Error("useSignInContext must be used within a SignInProvider");
  return t;
}, v = ({
  children: t,
  authMethods: n
}) => {
  const { authMethods: o } = f(), e = i(() => n?.filter((d) => o.includes(d)) ?? o, [n, o]), [s, r] = p({
    ...a,
    authMethod: e?.[0] || "email",
    authMethods: e
  }), c = i(
    () => ({ state: { ...s, authMethods: e }, dispatch: r }),
    [s, r, e]
  );
  return /* @__PURE__ */ h(u.Provider, { value: c, children: t });
};
export {
  u as SignInContext,
  v as SignInProvider,
  M as useSignInContext
};
