// @coinbase/cdp-react/dist/components/SignIn/SignInBackButton.js
import { jsx as n } from "react/jsx-runtime";
import { useMemo as f } from "react";
import { Button as S } from "../ui/Button/index.js";
import { IconArrowLeft as d } from "../../icons/IconArrowLeft.js";
import "../../theme/theme.js";
import { useSignInContext as B } from "./SignInProvider.js";
import "@coinbase/cdp-hooks";
import "../CDPReactProvider/index.js";
import '../../assets/SignInBackButton.css';const _ = "SignInBackButton-module__icon___stNFS", k = {
  icon: _
}, T = ({
  step: t,
  children: i,
  onClick: r,
  size: e = "md",
  variant: c = "transparentSecondary",
  ["aria-label"]: s = "Back",
  ...a
}) => {
  const { state: o, dispatch: m } = B(), p = f(() => t || o.step, [t, o.step]), l = (u) => {
    r?.(u), m({ type: "RESET_STATE" });
  };
  return p !== "verification" ? null : /* @__PURE__ */ n(
    S,
    {
      type: "button",
      "aria-label": s,
      onClick: l,
      size: e,
      variant: c,
      ...a,
      children: i || /* @__PURE__ */ n(d, { className: k.icon })
    }
  );
};
export {
  T as SignInBackButton
};
