// @coinbase/cdp-react/dist/components/SignIn/useSignInReducer.js
import { useReducer as u } from "react";
import "@coinbase/cdp-hooks";
import "../CDPReactProvider/index.js";
function l(e, r) {
  switch (r.type) {
    case "SET_AUTH_METHODS":
      return {
        ...e,
        authMethods: r.payload.authMethods
      };
    case "SET_AUTH_METHOD":
      return {
        ...e,
        authMethod: r.payload.authMethod,
        step: "credentials",
        email: "",
        phoneNumber: "",
        error: null
      };
    case "SET_EMAIL":
      return { ...e, email: r.payload.email, phoneNumber: "", error: null };
    case "SUBMIT_EMAIL":
      return {
        ...e,
        isPending: !0,
        email: r.payload.email,
        phoneNumber: "",
        error: null
      };
    case "SET_PHONE_NUMBER":
      return { ...e, phoneNumber: r.payload.phoneNumber, email: "", error: null };
    case "SUBMIT_PHONE_NUMBER":
      return {
        ...e,
        isPending: !0,
        phoneNumber: r.payload.phoneNumber,
        email: "",
        error: null
      };
    case "SUBMIT_EMAIL_SUCCESS":
    case "SUBMIT_PHONE_NUMBER_SUCCESS":
      return {
        ...e,
        error: null,
        flowId: r.payload.flowId,
        canResetOTP: !1,
        isPending: !1,
        otp: "",
        step: "verification"
      };
    case "SUBMIT_EMAIL_FAILURE":
    case "SUBMIT_PHONE_NUMBER_FAILURE":
      return {
        ...e,
        error: r.payload.error,
        isPending: !1
      };
    case "SET_OTP":
      return { ...e, error: null, otp: r.payload.otp };
    case "SUBMIT_OTP":
      return { ...e, error: null, isPending: !0, otp: r.payload.otp };
    case "SUBMIT_OTP_SUCCESS":
      return { ...e, error: null, isPending: !1, isSuccess: !0 };
    case "SUBMIT_OTP_FAILURE":
      return {
        ...e,
        error: r.payload.error,
        isPending: !1,
        isSuccess: !1
      };
    case "ALLOW_RESET_OTP":
      return { ...e, canResetOTP: !0 };
    case "RESET_OTP":
      return { ...e, error: null, canResetOTP: !1 };
    case "CLEAR_ERROR":
      return { ...e, error: null };
    case "RESET_STATE":
      return {
        authMethod: e.authMethod,
        authMethods: e.authMethods,
        canResetOTP: !1,
        error: null,
        isPending: !1,
        isSuccess: !1,
        email: "",
        flowId: "",
        otp: "",
        phoneNumber: "",
        step: "credentials"
      };
    default:
      throw new Error("Unknown action type");
  }
}
const s = (e) => u(l, e);
export {
  s as useSignInReducer
};
