// @coinbase/cdp-react/dist/components/OAuthStatusModal/index.js
import { jsx as s, jsxs as r, Fragment as S } from "react/jsx-runtime";
import "@coinbase/cdp-core";
import { useOAuthState as I } from "@coinbase/cdp-hooks";
import { u as b } from "../../chunks/useSendComponentCallOnce.BHZMuo6E.js";
import { useCallback as N, useState as O, useEffect as h, useRef as A } from "react";
import { useProviderName as $ } from "../CDPReactProvider/index.js";
import { CoinbaseFooter as k } from "../ui/CoinbaseFooter/index.js";
import { LoadingSpinner as w } from "../ui/LoadingSpinner/index.js";
import { Modal as E, ModalContent as L, ModalTitle as y, ModalDescription as R } from "../ui/Modal/index.js";
import { SwitchFadeTransition as P } from "../ui/SwitchFadeTransition/index.js";
import { VisuallyHidden as M } from "../ui/VisuallyHidden/index.js";
import { useTimer as U } from "../../hooks/useTimer.js";
import { IconAppleLogo as K } from "../../icons/IconAppleLogo.js";
import { IconCheck as x } from "../../icons/IconCheck.js";
import { IconGoogleColorLogo as G } from "../../icons/IconGoogleColorLogo.js";
import { IconXLogo as H } from "../../icons/IconXLogo.js";
import { IconXMark as q } from "../../icons/IconXMark.js";
import { capitalize as _ } from "../../utils/capitalize.js";
import '../../assets/OAuthStatusModal.css';const Y = "OAuthStatusModal-module__status___vU6x7", j = "OAuthStatusModal-module__graphic___2fKoa", D = "OAuthStatusModal-module__pad___oooMz", F = "OAuthStatusModal-module__spinner___Q3dYq", z = "OAuthStatusModal-module__icon___Lj5Gf", V = "OAuthStatusModal-module__subhead___Pg9GI", W = "OAuthStatusModal-module__description___0qOZY", X = "OAuthStatusModal-module__ring___mKwSb", J = "OAuthStatusModal-module__invisible___IeRd0", t = {
  "oauth-status-modal": "OAuthStatusModal-module__oauth-status-modal___I69Ii",
  "transition-wrapper": "OAuthStatusModal-module__transition-wrapper___TmPRh",
  status: Y,
  graphic: j,
  pad: D,
  spinner: F,
  icon: z,
  "icon-success": "OAuthStatusModal-module__icon-success___gqMWr",
  "icon-error": "OAuthStatusModal-module__icon-error___psI7m",
  subhead: V,
  description: W,
  ring: X,
  "ring-success": "OAuthStatusModal-module__ring-success___unyM9",
  "ring-error": "OAuthStatusModal-module__ring-error___OsgqU",
  invisible: J
}, p = "cdp-react-sign-in-with-oauth-provider-name", g = "cdp-react-sign-in-with-oauth-is-account-link", Q = ["pending", "success", "error"], It = () => {
  const n = $() || "";
  return { persistOAuthState: N(
    (e) => {
      sessionStorage.setItem(p, n), sessionStorage.setItem(
        g,
        e ? "true" : "false"
      );
    },
    [n]
  ) };
}, Nt = () => {
  b("oauth_status_modal");
  const [n, o] = O(!1), { oauthState: e } = I(), [d, a] = O(!1), c = $(), u = e?.providerType || "google";
  return h(() => {
    if (e?.status === "pending" || e?.status === "error" || e?.status === "success") {
      const l = sessionStorage.getItem(p);
      if (l === null || l !== c)
        return;
      sessionStorage.removeItem(p), o(!0);
      const m = sessionStorage.getItem(g) === "true";
      sessionStorage.removeItem(g), a(m);
    }
  }, [e?.status, c]), /* @__PURE__ */ s(E, { open: n, onOpenChange: o, children: /* @__PURE__ */ r(L, { children: [
    /* @__PURE__ */ s(M, { children: /* @__PURE__ */ s(y, { children: "OAuth Status" }) }),
    /* @__PURE__ */ s(M, { children: /* @__PURE__ */ r(R, { children: [
      d ? `Linking your ${u} account` : `Signing in with ${u}`,
      " ",
      e?.status ? `${e?.status === "success" ? "successful" : e?.status === "error" ? "failed" : "pending"}` : ""
    ] }) }),
    /* @__PURE__ */ s(Z, { isAccountLink: d, handleClose: () => o(!1) })
  ] }) });
}, Z = ({
  isAccountLink: n,
  handleClose: o
}) => {
  const { oauthState: e } = I(), d = e?.providerType || "google", { timeRemaining: a, start: c, reset: u } = U(), l = A(null), m = A(null), v = 250, f = N((i) => {
    m.current === i || i === void 0 || (m.current = i, l.current?.transition.toggle(i));
  }, []);
  return h(() => {
    f(e?.status), e?.status === "success" && c(3);
  }, [e?.status, f, c]), h(() => {
    a !== null && a <= 0 && (u(), o());
  }, [a, u, o]), /* @__PURE__ */ r("div", { className: t["oauth-status-modal"], children: [
    /* @__PURE__ */ s(
      P,
      {
        animateHeight: !1,
        timeout: v,
        items: Q,
        initialEntered: !1,
        transitionRef: l,
        className: t["transition-wrapper"],
        children: ({ itemKey: i, ...C }) => {
          const T = et[i];
          return /* @__PURE__ */ s("div", { ...C, className: t.status, children: /* @__PURE__ */ s(
            T,
            {
              isAccountLink: n,
              timeRemaining: a,
              provider: d
            }
          ) });
        }
      }
    ),
    /* @__PURE__ */ s(k, {})
  ] });
}, B = ({ isAccountLink: n, provider: o }) => {
  const e = ot[o];
  return /* @__PURE__ */ r(S, { children: [
    /* @__PURE__ */ r("div", { className: t.graphic, children: [
      /* @__PURE__ */ s(
        w,
        {
          strokeWidth: "5%",
          staticStroke: !0,
          staticStrokeColor: "var(--cdp-web-colors-line-default)",
          className: t.spinner
        }
      ),
      e && /* @__PURE__ */ s("span", { className: t.icon, children: /* @__PURE__ */ s(e, {}) })
    ] }),
    /* @__PURE__ */ s("p", { className: t.subhead, children: n ? `Linking your ${_(o)} account` : `Signing in with ${_(o)}` }),
    /* @__PURE__ */ s("p", { className: t.description, children: "Just a moment..." })
  ] });
}, tt = ({
  isAccountLink: n,
  timeRemaining: o,
  provider: e
}) => /* @__PURE__ */ r(S, { children: [
  /* @__PURE__ */ r("div", { className: t.graphic, children: [
    /* @__PURE__ */ s("span", { className: `${t.ring} ${t["ring-success"]}` }),
    /* @__PURE__ */ s("span", { className: `${t.icon} ${t["icon-success"]}`, children: /* @__PURE__ */ s(x, {}) })
  ] }),
  /* @__PURE__ */ s("p", { className: t.subhead, children: n ? `${_(e)} account linked` : `Signed in with ${_(e)}` }),
  /* @__PURE__ */ r("p", { className: `${t.description} ${o === null ? t.invisible : ""}`, children: [
    "This window will automatically close in ",
    o ?? "soon"
  ] })
] }), st = ({ isAccountLink: n }) => /* @__PURE__ */ r(S, { children: [
  /* @__PURE__ */ r("div", { className: `${t.graphic} ${t.pad}`, children: [
    /* @__PURE__ */ s("span", { className: `${t.ring} ${t["ring-error"]}` }),
    /* @__PURE__ */ s("span", { className: `${t.icon} ${t["icon-error"]}`, children: /* @__PURE__ */ s(q, {}) })
  ] }),
  /* @__PURE__ */ s("p", { className: t.subhead, children: n ? "Account linking failed" : "Sign in failed" }),
  /* @__PURE__ */ s("p", { className: t.description, children: "Something went wrong. Please try again." })
] }), et = {
  pending: B,
  success: tt,
  error: st
}, ot = {
  google: G,
  apple: K,
  x: H
};
export {
  g as OAUTH_IS_ACCOUNT_LINK_SESSION_STORAGE_KEY,
  p as OAUTH_PROVIDER_SESSION_STORAGE_KEY,
  Nt as OAuthStatusModal,
  Z as OAuthStatusModalContent,
  Nt as default,
  It as useOauthStatusModal
};
