// @coinbase/cdp-hooks/dist/esm/index43.js
import { getEnsAddress as a } from "./index44.js";
import { getEnsAvatar as m } from "./index45.js";
import { getEnsName as s } from "./index46.js";
import { getEnsResolver as i } from "./index47.js";
import { getEnsText as g } from "./index48.js";
import { call as p } from "./index49.js";
import { createAccessList as f } from "./index50.js";
import { createBlockFilter as n } from "./index51.js";
import { createContractEventFilter as c } from "./index52.js";
import { createEventFilter as C } from "./index53.js";
import { createPendingTransactionFilter as u } from "./index54.js";
import { estimateContractGas as F } from "./index55.js";
import { estimateFeesPerGas as T } from "./index56.js";
import { estimateGas as E } from "./index57.js";
import { estimateMaxPriorityFeePerGas as B } from "./index58.js";
import { getBalance as v } from "./index59.js";
import { getBlobBaseFee as d } from "./index60.js";
import { getBlock as w } from "./index61.js";
import { getBlockNumber as h } from "./index62.js";
import { getBlockTransactionCount as k } from "./index63.js";
import { getChainId as P } from "./index64.js";
import { getCode as o } from "./index65.js";
import { getContractEvents as y } from "./index66.js";
import { getEip712Domain as G } from "./index67.js";
import { getFeeHistory as R } from "./index68.js";
import { getFilterChanges as A } from "./index69.js";
import { getFilterLogs as b } from "./index70.js";
import { getGasPrice as l } from "./index71.js";
import { getLogs as L } from "./index72.js";
import { getProof as M } from "./index73.js";
import { getStorageAt as N } from "./index74.js";
import { getTransaction as x } from "./index75.js";
import { getTransactionConfirmations as D } from "./index76.js";
import { getTransactionCount as S } from "./index77.js";
import { getTransactionReceipt as q } from "./index78.js";
import { multicall as H } from "./index79.js";
import { readContract as I } from "./index80.js";
import { simulateBlocks as e } from "./index81.js";
import { simulateCalls as j } from "./index82.js";
import { simulateContract as z } from "./index83.js";
import { uninstallFilter as J } from "./index84.js";
import { verifyMessage as K } from "./index85.js";
import { verifyTypedData as O } from "./index86.js";
import { waitForTransactionReceipt as Q } from "./index87.js";
import { watchBlockNumber as U } from "./index88.js";
import { watchBlocks as V } from "./index89.js";
import { watchContractEvent as W } from "./index90.js";
import { watchEvent as X } from "./index91.js";
import { watchPendingTransactions as Y } from "./index92.js";
import { verifySiweMessage as Z } from "./index93.js";
import { prepareTransactionRequest as _ } from "./index94.js";
import { sendRawTransaction as $ } from "./index95.js";
function _r(t) {
  return {
    call: (r) => p(t, r),
    createAccessList: (r) => f(t, r),
    createBlockFilter: () => n(t),
    createContractEventFilter: (r) => c(t, r),
    createEventFilter: (r) => C(t, r),
    createPendingTransactionFilter: () => u(t),
    estimateContractGas: (r) => F(t, r),
    estimateGas: (r) => E(t, r),
    getBalance: (r) => v(t, r),
    getBlobBaseFee: () => d(t),
    getBlock: (r) => w(t, r),
    getBlockNumber: (r) => h(t, r),
    getBlockTransactionCount: (r) => k(t, r),
    getBytecode: (r) => o(t, r),
    getChainId: () => P(t),
    getCode: (r) => o(t, r),
    getContractEvents: (r) => y(t, r),
    getEip712Domain: (r) => G(t, r),
    getEnsAddress: (r) => a(t, r),
    getEnsAvatar: (r) => m(t, r),
    getEnsName: (r) => s(t, r),
    getEnsResolver: (r) => i(t, r),
    getEnsText: (r) => g(t, r),
    getFeeHistory: (r) => R(t, r),
    estimateFeesPerGas: (r) => T(t, r),
    getFilterChanges: (r) => A(t, r),
    getFilterLogs: (r) => b(t, r),
    getGasPrice: () => l(t),
    getLogs: (r) => L(t, r),
    getProof: (r) => M(t, r),
    estimateMaxPriorityFeePerGas: (r) => B(t, r),
    getStorageAt: (r) => N(t, r),
    getTransaction: (r) => x(t, r),
    getTransactionConfirmations: (r) => D(t, r),
    getTransactionCount: (r) => S(t, r),
    getTransactionReceipt: (r) => q(t, r),
    multicall: (r) => H(t, r),
    prepareTransactionRequest: (r) => _(t, r),
    readContract: (r) => I(t, r),
    sendRawTransaction: (r) => $(t, r),
    simulate: (r) => e(t, r),
    simulateBlocks: (r) => e(t, r),
    simulateCalls: (r) => j(t, r),
    simulateContract: (r) => z(t, r),
    verifyMessage: (r) => K(t, r),
    verifySiweMessage: (r) => Z(t, r),
    verifyTypedData: (r) => O(t, r),
    uninstallFilter: (r) => J(t, r),
    waitForTransactionReceipt: (r) => Q(t, r),
    watchBlocks: (r) => V(t, r),
    watchBlockNumber: (r) => U(t, r),
    watchContractEvent: (r) => W(t, r),
    watchEvent: (r) => X(t, r),
    watchPendingTransactions: (r) => Y(t, r)
  };
}
export {
  _r as publicActions
};
