// @coinbase/cdp-hooks/dist/esm/index228.js
import { execTyped as t } from "./index226.js";
const n = /^error (?<name>[a-zA-Z$_][a-zA-Z0-9$_]*)\((?<parameters>.*?)\)$/;
function g(e) {
  return n.test(e);
}
function S(e) {
  return t(n, e);
}
const r = /^event (?<name>[a-zA-Z$_][a-zA-Z0-9$_]*)\((?<parameters>.*?)\)$/;
function l(e) {
  return r.test(e);
}
function f(e) {
  return t(r, e);
}
const a = /^function (?<name>[a-zA-Z$_][a-zA-Z0-9$_]*)\((?<parameters>.*?)\)(?: (?<scope>external|public{1}))?(?: (?<stateMutability>pure|view|nonpayable|payable{1}))?(?: returns\s?\((?<returns>.*?)\))?$/;
function x(e) {
  return a.test(e);
}
function p(e) {
  return t(a, e);
}
const u = /^struct (?<name>[a-zA-Z$_][a-zA-Z0-9$_]*) \{(?<properties>.*?)\}$/;
function $(e) {
  return u.test(e);
}
function b(e) {
  return t(u, e);
}
const i = /^constructor\((?<parameters>.*?)\)(?:\s(?<stateMutability>payable{1}))?$/;
function m(e) {
  return i.test(e);
}
function y(e) {
  return t(i, e);
}
const c = /^fallback\(\) external(?:\s(?<stateMutability>payable{1}))?$/;
function v(e) {
  return c.test(e);
}
function z(e) {
  return t(c, e);
}
const o = /^receive\(\) external payable$/;
function A(e) {
  return o.test(e);
}
const R = /* @__PURE__ */ new Set(["indexed"]), Z = /* @__PURE__ */ new Set([
  "calldata",
  "memory",
  "storage"
]);
export {
  R as eventModifiers,
  y as execConstructorSignature,
  S as execErrorSignature,
  f as execEventSignature,
  z as execFallbackSignature,
  p as execFunctionSignature,
  b as execStructSignature,
  Z as functionModifiers,
  m as isConstructorSignature,
  g as isErrorSignature,
  l as isEventSignature,
  v as isFallbackSignature,
  x as isFunctionSignature,
  A as isReceiveSignature,
  $ as isStructSignature
};
