// @coinbase/cdp-hooks/dist/esm/index187.js
import { stringify as m } from "./index128.js";
import { BaseError as a } from "./index107.js";
import { getUrl as t } from "./index119.js";
class l extends a {
  constructor({ callbackSelector: r, cause: e, data: n, extraData: f, sender: c, urls: o }) {
    super(e.shortMessage || "An error occurred while fetching for an offchain result.", {
      cause: e,
      metaMessages: [
        ...e.metaMessages || [],
        e.metaMessages?.length ? "" : [],
        "Offchain Gateway Call:",
        o && [
          "  Gateway URL(s):",
          ...o.map((d) => `    ${t(d)}`)
        ],
        `  Sender: ${c}`,
        `  Data: ${n}`,
        `  Callback selector: ${r}`,
        `  Extra data: ${f}`
      ].flat(),
      name: "OffchainLookupError"
    });
  }
}
class u extends a {
  constructor({ result: r, url: e }) {
    super("Offchain gateway response is malformed. Response data must be a hex value.", {
      metaMessages: [
        `Gateway URL: ${t(e)}`,
        `Response: ${m(r)}`
      ],
      name: "OffchainLookupResponseMalformedError"
    });
  }
}
class g extends a {
  constructor({ sender: r, to: e }) {
    super("Reverted sender address does not match target contract address (`to`).", {
      metaMessages: [
        `Contract address: ${e}`,
        `OffchainLookup sender address: ${r}`
      ],
      name: "OffchainLookupSenderMismatchError"
    });
  }
}
export {
  l as OffchainLookupError,
  u as OffchainLookupResponseMalformedError,
  g as OffchainLookupSenderMismatchError
};
