// @coinbase/cdp-hooks/dist/esm/index272.js
import { hexToBigInt as n } from "./index167.js";
import { defineBlock as r } from "./index139.js";
import { defineTransaction as s, formatTransaction as m } from "./index141.js";
import { defineTransactionReceipt as l } from "./index168.js";
const p = {
  block: /* @__PURE__ */ r({
    format(e) {
      return {
        transactions: e.transactions?.map((t) => {
          if (typeof t == "string")
            return t;
          const i = m(t);
          return i.typeHex === "0x7e" && (i.isSystemTx = t.isSystemTx, i.mint = t.mint ? n(t.mint) : void 0, i.sourceHash = t.sourceHash, i.type = "deposit"), i;
        }),
        stateRoot: e.stateRoot
      };
    }
  }),
  transaction: /* @__PURE__ */ s({
    format(e) {
      const o = {};
      return e.type === "0x7e" && (o.isSystemTx = e.isSystemTx, o.mint = e.mint ? n(e.mint) : void 0, o.sourceHash = e.sourceHash, o.type = "deposit"), o;
    }
  }),
  transactionReceipt: /* @__PURE__ */ l({
    format(e) {
      return {
        l1GasPrice: e.l1GasPrice ? n(e.l1GasPrice) : null,
        l1GasUsed: e.l1GasUsed ? n(e.l1GasUsed) : null,
        l1Fee: e.l1Fee ? n(e.l1Fee) : null,
        l1FeeScalar: e.l1FeeScalar ? Number(e.l1FeeScalar) : null
      };
    }
  })
};
export {
  p as formatters
};
