// @coinbase/cdp-hooks/dist/esm/index122.js
import { BaseError as n } from "./index107.js";
class i extends n {
  constructor({ blockNumber: e, chain: t, contract: o }) {
    super(`Chain "${t.name}" does not support contract "${o.name}".`, {
      metaMessages: [
        "This could be due to any of the following:",
        ...e && o.blockCreated && o.blockCreated > e ? [
          `- The contract "${o.name}" was not deployed until block ${o.blockCreated} (current block ${e}).`
        ] : [
          `- The chain does not have the contract "${o.name}" configured.`
        ]
      ],
      name: "ChainDoesNotSupportContract"
    });
  }
}
class s extends n {
  constructor() {
    super("No chain was provided to the Client.", {
      name: "ClientChainNotConfiguredError"
    });
  }
}
class d extends n {
  constructor({ chainId: e }) {
    super(typeof e == "number" ? `Chain ID "${e}" is invalid.` : "Chain ID is invalid.", { name: "InvalidChainIdError" });
  }
}
export {
  i as ChainDoesNotSupportContract,
  s as ClientChainNotConfiguredError,
  d as InvalidChainIdError
};
