// @coinbase/cdp-hooks/dist/esm/index.js
import { APIError as i, OAuth2ProviderType as a } from "@coinbase/cdp-core";
import { Analytics as e } from "./index2.js";
import "react";
import { VERSION as o } from "./index3.js";
import { CDPContext as f, CDPHooksProvider as p } from "./index4.js";
import { useAuthenticateWithJWT as c, useConfig as x, useCurrentUser as E, useEnforceAuthenticated as l, useEnforceUnauthenticated as A, useEvmAccounts as d, useEvmAddress as g, useEvmSmartAccounts as v, useExportEvmAccount as h, useExportSolanaAccount as I, useGetAccessToken as P, useIsInitialized as k, useIsSignedIn as C, useOAuthState as O, useSendEvmTransaction as T, useSendSolanaTransaction as M, useSendUserOperation as y, useSignEvmHash as L, useSignEvmMessage as V, useSignEvmTransaction as W, useSignEvmTypedData as U, useSignInWithEmail as D, useSignInWithOAuth as G, useSignInWithSms as R, useSignOut as b, useSignSolanaMessage as H, useSignSolanaTransaction as K, useSolanaAccounts as z, useSolanaAddress as F, useVerifyEmailOTP as J, useVerifySmsOTP as N, useWaitForUserOperation as X } from "./index5.js";
import { useCreateSpendPermission as q } from "./index6.js";
import { useListSpendPermissions as B } from "./index7.js";
import { useRevokeSpendPermission as Y } from "./index8.js";
import { useLinkGoogle as _ } from "./index9.js";
import { useLinkApple as ee } from "./index10.js";
import { useLinkSms as re } from "./index11.js";
import { useLinkEmail as se } from "./index12.js";
import { useLinkOAuth as ue } from "./index13.js";
import { useCreateEvmEoaAccount as ae } from "./index14.js";
import { useCreateEvmSmartAccount as fe } from "./index15.js";
import { useCreateSolanaAccount as Se } from "./index16.js";
import { useX402 as xe } from "./index17.js";
import { useGetMfaConfig as le } from "./index18.js";
import { useRecordMfaEnrollmentPrompted as de } from "./index19.js";
import { useInitiateMfaEnrollment as ve } from "./index20.js";
import { useSubmitMfaEnrollment as Ie } from "./index21.js";
import { useInitiateMfaVerification as ke } from "./index22.js";
import { useSubmitMfaVerification as Oe } from "./index23.js";
import { useEvmKeyExportIframe as Me, useSolanaKeyExportIframe as ye } from "./index24.js";
e.registerPackageVersion("hooks", o);
export {
  i as APIError,
  f as CDPContext,
  p as CDPHooksProvider,
  a as OAuth2ProviderType,
  c as useAuthenticateWithJWT,
  x as useConfig,
  ae as useCreateEvmEoaAccount,
  fe as useCreateEvmSmartAccount,
  Se as useCreateSolanaAccount,
  q as useCreateSpendPermission,
  E as useCurrentUser,
  l as useEnforceAuthenticated,
  A as useEnforceUnauthenticated,
  d as useEvmAccounts,
  g as useEvmAddress,
  Me as useEvmKeyExportIframe,
  v as useEvmSmartAccounts,
  h as useExportEvmAccount,
  I as useExportSolanaAccount,
  P as useGetAccessToken,
  le as useGetMfaConfig,
  ve as useInitiateMfaEnrollment,
  ke as useInitiateMfaVerification,
  k as useIsInitialized,
  C as useIsSignedIn,
  ee as useLinkApple,
  se as useLinkEmail,
  _ as useLinkGoogle,
  ue as useLinkOAuth,
  re as useLinkSms,
  B as useListSpendPermissions,
  O as useOAuthState,
  de as useRecordMfaEnrollmentPrompted,
  Y as useRevokeSpendPermission,
  T as useSendEvmTransaction,
  M as useSendSolanaTransaction,
  y as useSendUserOperation,
  L as useSignEvmHash,
  V as useSignEvmMessage,
  W as useSignEvmTransaction,
  U as useSignEvmTypedData,
  D as useSignInWithEmail,
  G as useSignInWithOAuth,
  R as useSignInWithSms,
  b as useSignOut,
  H as useSignSolanaMessage,
  K as useSignSolanaTransaction,
  z as useSolanaAccounts,
  F as useSolanaAddress,
  ye as useSolanaKeyExportIframe,
  Ie as useSubmitMfaEnrollment,
  Oe as useSubmitMfaVerification,
  J as useVerifyEmailOTP,
  N as useVerifySmsOTP,
  X as useWaitForUserOperation,
  xe as useX402
};
