// @coinbase/cdp-hooks/dist/esm/index47.js
import { getChainContractAddress as l } from "./index198.js";
import { toHex as u } from "./index134.js";
import { packetToBytes as v } from "./index244.js";
import { getAction as f } from "./index237.js";
import { readContract as m } from "./index80.js";
async function A(o, r) {
  const { blockNumber: n, blockTag: i, name: s } = r, { chain: e } = o, a = (() => {
    if (r.universalResolverAddress)
      return r.universalResolverAddress;
    if (!e)
      throw new Error("client chain not configured. universalResolverAddress is required.");
    return l({
      blockNumber: n,
      chain: e,
      contract: "ensUniversalResolver"
    });
  })(), t = e?.ensTlds;
  if (t && !t.some((c) => s.endsWith(c)))
    throw new Error(`${s} is not a valid ENS TLD (${t?.join(", ")}) for chain "${e.name}" (id: ${e.id}).`);
  const [d] = await f(o, m, "readContract")({
    address: a,
    abi: [
      {
        inputs: [{ type: "bytes" }],
        name: "findResolver",
        outputs: [{ type: "address" }, { type: "bytes32" }],
        stateMutability: "view",
        type: "function"
      }
    ],
    functionName: "findResolver",
    args: [u(v(s))],
    blockNumber: n,
    blockTag: i
  });
  return d;
}
export {
  A as getEnsResolver
};
