// @coinbase/cdp-hooks/dist/esm/index250.js
import { InvalidAddressError as a } from "./index129.js";
import { InvalidBytesLengthError as d } from "./index130.js";
import { AccountStateConflictError as u, StateAssignmentConflictError as h } from "./index117.js";
import { isAddress as l } from "./index156.js";
import { numberToHex as c } from "./index134.js";
function s(r) {
  if (!(!r || r.length === 0))
    return r.reduce((t, { slot: e, value: n }) => {
      if (e.length !== 66)
        throw new d({
          size: e.length,
          targetSize: 66,
          type: "hex"
        });
      if (n.length !== 66)
        throw new d({
          size: n.length,
          targetSize: 66,
          type: "hex"
        });
      return t[e] = n, t;
    }, {});
}
function m(r) {
  const { balance: t, nonce: e, state: n, stateDiff: o, code: f } = r, i = {};
  if (f !== void 0 && (i.code = f), t !== void 0 && (i.balance = c(t)), e !== void 0 && (i.nonce = c(e)), n !== void 0 && (i.state = s(n)), o !== void 0) {
    if (i.state)
      throw new h();
    i.stateDiff = s(o);
  }
  return i;
}
function A(r) {
  if (!r)
    return;
  const t = {};
  for (const { address: e, ...n } of r) {
    if (!l(e, { strict: !1 }))
      throw new a({ address: e });
    if (t[e])
      throw new u({ address: e });
    t[e] = m(n);
  }
  return t;
}
export {
  m as serializeAccountStateOverride,
  s as serializeStateMapping,
  A as serializeStateOverride
};
