// @coinbase/cdp-hooks/dist/esm/index88.js
import { hexToBigInt as w } from "./index167.js";
import { getAction as g } from "./index237.js";
import { observe as k } from "./index239.js";
import { poll as h } from "./index240.js";
import { stringify as y } from "./index128.js";
import { getBlockNumber as v } from "./index62.js";
function H(r, { emitOnBegin: c = !1, emitMissed: a = !1, onBlockNumber: l, onError: b, poll: i, pollingInterval: m = r.pollingInterval }) {
  const N = typeof i < "u" ? i : !(r.transport.type === "webSocket" || r.transport.type === "ipc" || r.transport.type === "fallback" && (r.transport.transports[0].config.type === "webSocket" || r.transport.transports[0].config.type === "ipc"));
  let t;
  return N ? (() => {
    const p = y([
      "watchBlockNumber",
      r.uid,
      c,
      a,
      m
    ]);
    return k(p, { onBlockNumber: l, onError: b }, (s) => h(async () => {
      try {
        const o = await g(r, v, "getBlockNumber")({ cacheTime: 0 });
        if (t) {
          if (o === t)
            return;
          if (o - t > 1 && a)
            for (let e = t + 1n; e < o; e++)
              s.onBlockNumber(e, t), t = e;
        }
        (!t || o > t) && (s.onBlockNumber(o, t), t = o);
      } catch (o) {
        s.onError?.(o);
      }
    }, {
      emitOnBegin: c,
      interval: m
    }));
  })() : (() => {
    const p = y([
      "watchBlockNumber",
      r.uid,
      c,
      a
    ]);
    return k(p, { onBlockNumber: l, onError: b }, (s) => {
      let o = !0, e = () => o = !1;
      return (async () => {
        try {
          const f = (() => {
            if (r.transport.type === "fallback") {
              const n = r.transport.transports.find((u) => u.config.type === "webSocket" || u.config.type === "ipc");
              return n ? n.value : r.transport;
            }
            return r.transport;
          })(), { unsubscribe: B } = await f.subscribe({
            params: ["newHeads"],
            onData(n) {
              if (!o)
                return;
              const u = w(n.result?.number);
              s.onBlockNumber(u, t), t = u;
            },
            onError(n) {
              s.onError?.(n);
            }
          });
          e = B, o || e();
        } catch (f) {
          b?.(f);
        }
      })(), () => e();
    });
  })();
}
export {
  H as watchBlockNumber
};
