// @coinbase/cdp-hooks/dist/esm/index177.js
import { sha256 as v } from "./index263.js";
import { createCurve as G } from "./index264.js";
import { Field as P, mod as h, pow2 as t } from "./index265.js";
const B = {
  p: BigInt("0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffefffffc2f"),
  n: BigInt("0xfffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141"),
  h: BigInt(1),
  a: BigInt(0),
  b: BigInt(7),
  Gx: BigInt("0x79be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798"),
  Gy: BigInt("0x483ada7726a3c4655da4fbfc0e1108a8fd17b448a68554199c47d08ffb10d4b8")
};
BigInt(0);
const M = BigInt(1), x = BigInt(2), k = (n, f) => (n + f / x) / f;
function N(n) {
  const f = B.p, e = BigInt(3), a = BigInt(6), I = BigInt(11), b = BigInt(22), s = BigInt(23), i = BigInt(44), r = BigInt(88), c = n * n * n % f, o = c * c * n % f, d = t(o, e, f) * o % f, g = t(d, e, f) * o % f, m = t(g, x, f) * c % f, p = t(m, I, f) * m % f, _ = t(p, b, f) * p % f, u = t(_, i, f) * _ % f, q = t(u, r, f) * u % f, E = t(q, i, f) * _ % f, C = t(E, e, f) * o % f, F = t(C, s, f) * p % f, S = t(F, a, f) * c % f, w = t(S, x, f);
  if (!l.eql(l.sqr(w), n))
    throw new Error("Cannot find square root");
  return w;
}
const l = P(B.p, void 0, void 0, { sqrt: N }), V = G({
  ...B,
  Fp: l,
  lowS: !0,
  // Allow only low-S signatures by default in sign() and verify()
  endo: {
    // Endomorphism, see above
    beta: BigInt("0x7ae96a2b657c07106e64479eac3434e99cf0497512f58995c1396c28719501ee"),
    splitScalar: (n) => {
      const f = B.n, e = BigInt("0x3086d221a7d46bcde86c90e49284eb15"), a = -M * BigInt("0xe4437ed6010e88286f547fa90abfe4c3"), I = BigInt("0x114ca50f7a8e2f3f657c1108d9d44cfd8"), b = e, s = BigInt("0x100000000000000000000000000000000"), i = k(b * n, f), r = k(-a * n, f);
      let c = h(n - i * e - r * I, f), o = h(-i * a - r * b, f);
      const d = c > s, g = o > s;
      if (d && (c = f - c), g && (o = f - o), c > s || o > s)
        throw new Error("splitScalar: Endomorphism failed, k=" + n);
      return { k1neg: d, k1: c, k2neg: g, k2: o };
    }
  }
}, v);
export {
  V as secp256k1
};
