// @coinbase/cdp-hooks/dist/esm/index256.js
import { BaseError as a } from "./index107.js";
import { ExecutionRevertedError as s, FeeCapTooHighError as r, FeeCapTooLowError as i, NonceTooHighError as c, NonceTooLowError as u, NonceMaxValueError as d, InsufficientFundsError as f, IntrinsicGasTooHighError as w, IntrinsicGasTooLowError as E, TransactionTypeNotSupportedError as m, TipAboveFeeCapError as g, UnknownNodeError as F } from "./index126.js";
function G(e, n) {
  const o = (e.details || "").toLowerCase(), t = e instanceof a ? e.walk((x) => x?.code === s.code) : e;
  return t instanceof a ? new s({
    cause: e,
    message: t.details
  }) : s.nodeMessage.test(o) ? new s({
    cause: e,
    message: e.details
  }) : r.nodeMessage.test(o) ? new r({
    cause: e,
    maxFeePerGas: n?.maxFeePerGas
  }) : i.nodeMessage.test(o) ? new i({
    cause: e,
    maxFeePerGas: n?.maxFeePerGas
  }) : c.nodeMessage.test(o) ? new c({ cause: e, nonce: n?.nonce }) : u.nodeMessage.test(o) ? new u({ cause: e, nonce: n?.nonce }) : d.nodeMessage.test(o) ? new d({ cause: e, nonce: n?.nonce }) : f.nodeMessage.test(o) ? new f({ cause: e }) : w.nodeMessage.test(o) ? new w({ cause: e, gas: n?.gas }) : E.nodeMessage.test(o) ? new E({ cause: e, gas: n?.gas }) : m.nodeMessage.test(o) ? new m({ cause: e }) : g.nodeMessage.test(o) ? new g({
    cause: e,
    maxFeePerGas: n?.maxFeePerGas,
    maxPriorityFeePerGas: n?.maxPriorityFeePerGas
  }) : new F({
    cause: e
  });
}
export {
  G as getNodeError
};
