// @coinbase/cdp-hooks/dist/esm/index50.js
import { parseAccount as P } from "./index36.js";
import { numberToHex as y } from "./index134.js";
import { getCallError as F } from "./index248.js";
import { extract as L } from "./index249.js";
import { formatTransactionRequest as A } from "./index169.js";
import { assertRequest as G } from "./index182.js";
async function _(t, e) {
  const { account: o = t.account, blockNumber: r, blockTag: m = "latest", blobs: i, data: u, gas: f, gasPrice: b, maxFeePerBlobGas: p, maxFeePerGas: d, maxPriorityFeePerGas: h, to: l, value: x, ...g } = e, a = o ? P(o) : void 0;
  try {
    G(e);
    const k = (typeof r == "bigint" ? y(r) : void 0) || m, c = t.chain?.formatters?.transactionRequest?.format, q = (c || A)({
      // Pick out extra data that might exist on the chain's transaction request type.
      ...L(g, { format: c }),
      from: a?.address,
      blobs: i,
      data: u,
      gas: f,
      gasPrice: b,
      maxFeePerBlobGas: p,
      maxFeePerGas: d,
      maxPriorityFeePerGas: h,
      to: l,
      value: x
    }), n = await t.request({
      method: "eth_createAccessList",
      params: [q, k]
    });
    return {
      accessList: n.accessList,
      gasUsed: BigInt(n.gasUsed)
    };
  } catch (s) {
    throw F(s, {
      ...e,
      account: a,
      chain: t.chain
    });
  }
}
export {
  _ as createAccessList
};
