// @coinbase/cdp-hooks/dist/esm/index173.js
import { BytesSizeMismatchError as I } from "./index104.js";
import { InvalidAddressError as v } from "./index129.js";
import { InvalidDomainError as w, InvalidPrimaryTypeError as E, InvalidStructTypeError as z } from "./index131.js";
import { isAddress as D } from "./index156.js";
import { size as c } from "./index106.js";
import { numberToHex as T } from "./index134.js";
import { integerRegex as x, bytesRegex as P } from "./index158.js";
function C(e) {
  const { domain: o, message: y, primaryType: s, types: r } = e, a = (d, g) => {
    for (const h of d) {
      const { name: b, type: i } = h, t = g[b], m = i.match(x);
      if (m && (typeof t == "number" || typeof t == "bigint")) {
        const [u, n, l] = m;
        T(t, {
          signed: n === "int",
          size: Number.parseInt(l) / 8
        });
      }
      if (i === "address" && typeof t == "string" && !D(t))
        throw new v({ address: t });
      const f = i.match(P);
      if (f) {
        const [u, n] = f;
        if (n && c(t) !== Number.parseInt(n))
          throw new I({
            expectedSize: Number.parseInt(n),
            givenSize: c(t)
          });
      }
      const p = r[i];
      p && (S(i), a(p, t));
    }
  };
  if (r.EIP712Domain && o) {
    if (typeof o != "object")
      throw new w({ domain: o });
    a(r.EIP712Domain, o);
  }
  if (s !== "EIP712Domain")
    if (r[s])
      a(r[s], y);
    else
      throw new E({ primaryType: s, types: r });
}
function j({ domain: e }) {
  return [
    typeof e?.name == "string" && { name: "name", type: "string" },
    e?.version && { name: "version", type: "string" },
    (typeof e?.chainId == "number" || typeof e?.chainId == "bigint") && {
      name: "chainId",
      type: "uint256"
    },
    e?.verifyingContract && {
      name: "verifyingContract",
      type: "address"
    },
    e?.salt && { name: "salt", type: "bytes32" }
  ].filter(Boolean);
}
function S(e) {
  if (e === "address" || e === "bool" || e === "string" || e.startsWith("bytes") || e.startsWith("uint") || e.startsWith("int"))
    throw new z({ type: e });
}
export {
  j as getTypesForEIP712Domain,
  C as validateTypedData
};
