// @coinbase/cdp-hooks/dist/esm/index233.js
import { BaseError as t } from "./index221.js";
class n extends t {
  constructor({ param: e }) {
    super("Invalid ABI parameter.", {
      details: e
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "InvalidParameterError"
    });
  }
}
class s extends t {
  constructor({ param: e, name: r }) {
    super("Invalid ABI parameter.", {
      details: e,
      metaMessages: [
        `"${r}" is a protected Solidity keyword. More info: https://docs.soliditylang.org/en/latest/cheatsheet.html`
      ]
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "SolidityProtectedKeywordError"
    });
  }
}
class l extends t {
  constructor({ param: e, type: r, modifier: i }) {
    super("Invalid ABI parameter.", {
      details: e,
      metaMessages: [
        `Modifier "${i}" not allowed${r ? ` in "${r}" type` : ""}.`
      ]
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "InvalidModifierError"
    });
  }
}
class d extends t {
  constructor({ param: e, type: r, modifier: i }) {
    super("Invalid ABI parameter.", {
      details: e,
      metaMessages: [
        `Modifier "${i}" not allowed${r ? ` in "${r}" type` : ""}.`,
        `Data location can only be specified for array, struct, or mapping types, but "${i}" was given.`
      ]
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "InvalidFunctionModifierError"
    });
  }
}
class u extends t {
  constructor({ abiParameter: e }) {
    super("Invalid ABI parameter.", {
      details: JSON.stringify(e, null, 2),
      metaMessages: ["ABI parameter type is invalid."]
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "InvalidAbiTypeParameterError"
    });
  }
}
export {
  u as InvalidAbiTypeParameterError,
  d as InvalidFunctionModifierError,
  l as InvalidModifierError,
  n as InvalidParameterError,
  s as SolidityProtectedKeywordError
};
