// @coinbase/cdp-hooks/dist/esm/index294.js
import { fromString as d } from "./index297.js";
import { checksum as o } from "./index300.js";
import { BaseError as c } from "./index289.js";
import { keccak256 as f } from "./index292.js";
const l = /^0x[a-fA-F0-9]{40}$/;
function u(r, t = {}) {
  const { strict: s = !0 } = t;
  if (!l.test(r))
    throw new i({
      address: r,
      cause: new h()
    });
  if (s) {
    if (r.toLowerCase() === r)
      return;
    if (m(r) !== r)
      throw new i({
        address: r,
        cause: new p()
      });
  }
}
function m(r) {
  if (o.has(r))
    return o.get(r);
  u(r, { strict: !1 });
  const t = r.substring(2).toLowerCase(), s = f(d(t), { as: "Bytes" }), n = t.split("");
  for (let e = 0; e < 40; e += 2)
    s[e >> 1] >> 4 >= 8 && n[e] && (n[e] = n[e].toUpperCase()), (s[e >> 1] & 15) >= 8 && n[e + 1] && (n[e + 1] = n[e + 1].toUpperCase());
  const a = `0x${n.join("")}`;
  return o.set(r, a), a;
}
function k(r, t = {}) {
  const { strict: s = !0 } = t ?? {};
  try {
    return u(r, { strict: s }), !0;
  } catch {
    return !1;
  }
}
class i extends c {
  constructor({ address: t, cause: s }) {
    super(`Address "${t}" is invalid.`, {
      cause: s
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Address.InvalidAddressError"
    });
  }
}
class h extends c {
  constructor() {
    super("Address is not a 20 byte (40 hexadecimal character) value."), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Address.InvalidInputError"
    });
  }
}
class p extends c {
  constructor() {
    super("Address does not match its checksum counterpart."), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Address.InvalidChecksumError"
    });
  }
}
export {
  i as InvalidAddressError,
  p as InvalidChecksumError,
  h as InvalidInputError,
  u as assert,
  m as checksum,
  k as validate
};
