// @coinbase/cdp-hooks/dist/esm/index125.js
import { formatEther as E } from "./index115.js";
import { formatGwei as t } from "./index116.js";
import { BaseError as y } from "./index107.js";
import { prettyPrint as l } from "./index118.js";
class x extends y {
  constructor(e, { account: a, docsPath: n, chain: m, data: f, gas: u, gasPrice: r, maxFeePerGas: o, maxPriorityFeePerGas: s, nonce: d, to: p, value: i }) {
    const g = l({
      from: a?.address,
      to: p,
      value: typeof i < "u" && `${E(i)} ${m?.nativeCurrency?.symbol || "ETH"}`,
      data: f,
      gas: u,
      gasPrice: typeof r < "u" && `${t(r)} gwei`,
      maxFeePerGas: typeof o < "u" && `${t(o)} gwei`,
      maxPriorityFeePerGas: typeof s < "u" && `${t(s)} gwei`,
      nonce: d
    });
    super(e.shortMessage, {
      cause: e,
      docsPath: n,
      metaMessages: [
        ...e.metaMessages ? [...e.metaMessages, " "] : [],
        "Estimate Gas Arguments:",
        g
      ].filter(Boolean),
      name: "EstimateGasExecutionError"
    }), Object.defineProperty(this, "cause", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), this.cause = e;
  }
}
export {
  x as EstimateGasExecutionError
};
