// @coinbase/cdp-hooks/dist/esm/index123.js
import { BaseError as o } from "./index107.js";
class i extends o {
  constructor({ max: e, min: r, signed: s, size: a, value: t }) {
    super(`Number "${t}" is not in safe ${a ? `${a * 8}-bit ${s ? "signed" : "unsigned"} ` : ""}integer range ${e ? `(${r} to ${e})` : `(above ${r})`}`, { name: "IntegerOutOfRangeError" });
  }
}
class u extends o {
  constructor(e) {
    super(`Bytes value "${e}" is not a valid boolean. The bytes array must contain a single byte of either a 0 or 1 value.`, {
      name: "InvalidBytesBooleanError"
    });
  }
}
class c extends o {
  constructor(e) {
    super(`Hex value "${e}" is not a valid boolean. The hex value must be "0x0" (false) or "0x1" (true).`, { name: "InvalidHexBooleanError" });
  }
}
class v extends o {
  constructor({ givenSize: e, maxSize: r }) {
    super(`Size cannot exceed ${r} bytes. Given size: ${e} bytes.`, { name: "SizeOverflowError" });
  }
}
export {
  i as IntegerOutOfRangeError,
  u as InvalidBytesBooleanError,
  c as InvalidHexBooleanError,
  v as SizeOverflowError
};
