// @coinbase/cdp-hooks/dist/esm/index117.js
import { BaseError as a } from "./index107.js";
class f extends a {
  constructor({ address: r }) {
    super(`State for account "${r}" is set multiple times.`, {
      name: "AccountStateConflictError"
    });
  }
}
class s extends a {
  constructor() {
    super("state and stateDiff are set on the same account.", {
      name: "StateAssignmentConflictError"
    });
  }
}
function o(n) {
  return n.reduce((r, { slot: c, value: e }) => `${r}        ${c}: ${e}
`, "");
}
function u(n) {
  return n.reduce((r, { address: c, ...e }) => {
    let t = `${r}    ${c}:
`;
    return e.nonce && (t += `      nonce: ${e.nonce}
`), e.balance && (t += `      balance: ${e.balance}
`), e.code && (t += `      code: ${e.code}
`), e.state && (t += `      state:
`, t += o(e.state)), e.stateDiff && (t += `      stateDiff:
`, t += o(e.stateDiff)), t;
  }, `  State Override:
`).slice(0, -1);
}
export {
  f as AccountStateConflictError,
  s as StateAssignmentConflictError,
  o as prettyStateMapping,
  u as prettyStateOverride
};
