// @coinbase/cdp-hooks/dist/esm/index205.js
import { AbiDecodingZeroDataError as p } from "./index104.js";
import { BaseError as C } from "./index107.js";
import { RawContractError as w, ContractFunctionZeroDataError as g, ContractFunctionRevertedError as l, ContractFunctionExecutionError as D } from "./index110.js";
import { RpcRequestError as F } from "./index38.js";
import { InternalRpcError as O } from "./index121.js";
const _ = 3;
function Z(r, { abi: n, address: i, args: E, docsPath: d, functionName: o, sender: f }) {
  const e = r instanceof w ? r : r instanceof C ? r.walk((R) => "data" in R) || r.walk() : {}, { code: m, data: t, details: a, message: c, shortMessage: s } = e, u = r instanceof p ? new g({ functionName: o }) : [_, O.code].includes(m) && (t || a || c || s) ? new l({
    abi: n,
    data: typeof t == "object" ? t.data : t,
    functionName: o,
    message: e instanceof F ? a : s ?? c
  }) : r;
  return new D(u, {
    abi: n,
    args: E,
    contractAddress: i,
    docsPath: d,
    functionName: o,
    sender: f
  });
}
export {
  Z as getContractError
};
