// @coinbase/cdp-hooks/dist/esm/index124.js
import { BaseError as r } from "./index107.js";
class s extends r {
  constructor({ data: a }) {
    super("Unable to extract image from metadata. The metadata may be malformed or invalid.", {
      metaMessages: [
        "- Metadata must be a JSON object with at least an `image`, `image_url` or `image_data` property.",
        "",
        `Provided data: ${JSON.stringify(a)}`
      ],
      name: "EnsAvatarInvalidMetadataError"
    });
  }
}
class o extends r {
  constructor({ reason: a }) {
    super(`ENS NFT avatar URI is invalid. ${a}`, {
      name: "EnsAvatarInvalidNftUriError"
    });
  }
}
class n extends r {
  constructor({ uri: a }) {
    super(`Unable to resolve ENS avatar URI "${a}". The URI may be malformed, invalid, or does not respond with a valid image.`, { name: "EnsAvatarUriResolutionError" });
  }
}
class i extends r {
  constructor({ namespace: a }) {
    super(`ENS NFT avatar namespace "${a}" is not supported. Must be "erc721" or "erc1155".`, { name: "EnsAvatarUnsupportedNamespaceError" });
  }
}
export {
  s as EnsAvatarInvalidMetadataError,
  o as EnsAvatarInvalidNftUriError,
  i as EnsAvatarUnsupportedNamespaceError,
  n as EnsAvatarUriResolutionError
};
