// @coinbase/cdp-hooks/dist/esm/index38.js
import { stringify as s } from "./index128.js";
import { BaseError as a } from "./index107.js";
import { getUrl as o } from "./index119.js";
class m extends a {
  constructor({ body: t, cause: e, details: r, headers: d, status: u, url: l }) {
    super("HTTP request failed.", {
      cause: e,
      details: r,
      metaMessages: [
        u && `Status: ${u}`,
        `URL: ${o(l)}`,
        t && `Request body: ${s(t)}`
      ].filter(Boolean),
      name: "HttpRequestError"
    }), Object.defineProperty(this, "body", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "headers", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "status", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "url", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), this.body = t, this.headers = d, this.status = u, this.url = l;
  }
}
class f extends a {
  constructor({ body: t, error: e, url: r }) {
    super("RPC Request failed.", {
      cause: e,
      details: e.message,
      metaMessages: [`URL: ${o(r)}`, `Request body: ${s(t)}`],
      name: "RpcRequestError"
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "data", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), this.code = e.code, this.data = e.data;
  }
}
class p extends a {
  constructor({ body: t, url: e }) {
    super("The request took too long to respond.", {
      details: "The request timed out.",
      metaMessages: [`URL: ${o(e)}`, `Request body: ${s(t)}`],
      name: "TimeoutError"
    });
  }
}
export {
  m as HttpRequestError,
  f as RpcRequestError,
  p as TimeoutError
};
