// @coinbase/cdp-hooks/dist/esm/index169.js
import { bytesToHex as d, numberToHex as n } from "./index134.js";
const o = {
  legacy: "0x0",
  eip2930: "0x1",
  eip1559: "0x2",
  eip4844: "0x3",
  eip7702: "0x4"
};
function p(e) {
  const f = {};
  return typeof e.authorizationList < "u" && (f.authorizationList = a(e.authorizationList)), typeof e.accessList < "u" && (f.accessList = e.accessList), typeof e.blobVersionedHashes < "u" && (f.blobVersionedHashes = e.blobVersionedHashes), typeof e.blobs < "u" && (typeof e.blobs[0] != "string" ? f.blobs = e.blobs.map((i) => d(i)) : f.blobs = e.blobs), typeof e.data < "u" && (f.data = e.data), typeof e.from < "u" && (f.from = e.from), typeof e.gas < "u" && (f.gas = n(e.gas)), typeof e.gasPrice < "u" && (f.gasPrice = n(e.gasPrice)), typeof e.maxFeePerBlobGas < "u" && (f.maxFeePerBlobGas = n(e.maxFeePerBlobGas)), typeof e.maxFeePerGas < "u" && (f.maxFeePerGas = n(e.maxFeePerGas)), typeof e.maxPriorityFeePerGas < "u" && (f.maxPriorityFeePerGas = n(e.maxPriorityFeePerGas)), typeof e.nonce < "u" && (f.nonce = n(e.nonce)), typeof e.to < "u" && (f.to = e.to), typeof e.type < "u" && (f.type = o[e.type]), typeof e.value < "u" && (f.value = n(e.value)), f;
}
function a(e) {
  return e.map((f) => ({
    address: f.address,
    r: f.r ? n(BigInt(f.r)) : f.r,
    s: f.s ? n(BigInt(f.s)) : f.s,
    chainId: n(f.chainId),
    nonce: n(f.nonce),
    ...typeof f.yParity < "u" ? { yParity: n(f.yParity) } : {},
    ...typeof f.v < "u" && typeof f.yParity > "u" ? { v: n(f.v) } : {}
  }));
}
export {
  p as formatTransactionRequest,
  o as rpcTransactionType
};
