// @coinbase/cdp-hooks/dist/esm/index121.js
import { BaseError as C } from "./index107.js";
import { RpcRequestError as D } from "./index38.js";
const S = -1;
class r extends C {
  constructor(e, { code: t, docsPath: U, metaMessages: N, name: I, shortMessage: k }) {
    super(k, {
      cause: e,
      docsPath: U,
      metaMessages: N || e?.metaMessages,
      name: I || "RpcError"
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), this.name = I || e.name, this.code = e instanceof D ? e.code : t ?? S;
  }
}
class o extends r {
  constructor(e, t) {
    super(e, t), Object.defineProperty(this, "data", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), this.data = t.data;
  }
}
class s extends r {
  constructor(e) {
    super(e, {
      code: s.code,
      name: "ParseRpcError",
      shortMessage: "Invalid JSON was received by the server. An error occurred on the server while parsing the JSON text."
    });
  }
}
Object.defineProperty(s, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32700
});
class c extends r {
  constructor(e) {
    super(e, {
      code: c.code,
      name: "InvalidRequestRpcError",
      shortMessage: "JSON is not a valid request object."
    });
  }
}
Object.defineProperty(c, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32600
});
class a extends r {
  constructor(e, { method: t } = {}) {
    super(e, {
      code: a.code,
      name: "MethodNotFoundRpcError",
      shortMessage: `The method${t ? ` "${t}"` : ""} does not exist / is not available.`
    });
  }
}
Object.defineProperty(a, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32601
});
class u extends r {
  constructor(e) {
    super(e, {
      code: u.code,
      name: "InvalidParamsRpcError",
      shortMessage: [
        "Invalid parameters were provided to the RPC method.",
        "Double check you have provided the correct parameters."
      ].join(`
`)
    });
  }
}
Object.defineProperty(u, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32602
});
class n extends r {
  constructor(e) {
    super(e, {
      code: n.code,
      name: "InternalRpcError",
      shortMessage: "An internal error was received."
    });
  }
}
Object.defineProperty(n, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32603
});
class d extends r {
  constructor(e) {
    super(e, {
      code: d.code,
      name: "InvalidInputRpcError",
      shortMessage: [
        "Missing or invalid parameters.",
        "Double check you have provided the correct parameters."
      ].join(`
`)
    });
  }
}
Object.defineProperty(d, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32e3
});
class l extends r {
  constructor(e) {
    super(e, {
      code: l.code,
      name: "ResourceNotFoundRpcError",
      shortMessage: "Requested resource not found."
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "ResourceNotFoundRpcError"
    });
  }
}
Object.defineProperty(l, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32001
});
class i extends r {
  constructor(e) {
    super(e, {
      code: i.code,
      name: "ResourceUnavailableRpcError",
      shortMessage: "Requested resource not available."
    });
  }
}
Object.defineProperty(i, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32002
});
class b extends r {
  constructor(e) {
    super(e, {
      code: b.code,
      name: "TransactionRejectedRpcError",
      shortMessage: "Transaction creation failed."
    });
  }
}
Object.defineProperty(b, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32003
});
class p extends r {
  constructor(e, { method: t } = {}) {
    super(e, {
      code: p.code,
      name: "MethodNotSupportedRpcError",
      shortMessage: `Method${t ? ` "${t}"` : ""} is not supported.`
    });
  }
}
Object.defineProperty(p, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32004
});
class h extends r {
  constructor(e) {
    super(e, {
      code: h.code,
      name: "LimitExceededRpcError",
      shortMessage: "Request exceeds defined limit."
    });
  }
}
Object.defineProperty(h, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32005
});
class m extends r {
  constructor(e) {
    super(e, {
      code: m.code,
      name: "JsonRpcVersionUnsupportedError",
      shortMessage: "Version of JSON-RPC protocol is not supported."
    });
  }
}
Object.defineProperty(m, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32006
});
class f extends o {
  constructor(e) {
    super(e, {
      code: f.code,
      name: "UserRejectedRequestError",
      shortMessage: "User rejected the request."
    });
  }
}
Object.defineProperty(f, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 4001
});
class g extends o {
  constructor(e) {
    super(e, {
      code: g.code,
      name: "UnauthorizedProviderError",
      shortMessage: "The requested method and/or account has not been authorized by the user."
    });
  }
}
Object.defineProperty(g, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 4100
});
class v extends o {
  constructor(e, { method: t } = {}) {
    super(e, {
      code: v.code,
      name: "UnsupportedProviderMethodError",
      shortMessage: `The Provider does not support the requested method${t ? ` " ${t}"` : ""}.`
    });
  }
}
Object.defineProperty(v, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 4200
});
class w extends o {
  constructor(e) {
    super(e, {
      code: w.code,
      name: "ProviderDisconnectedError",
      shortMessage: "The Provider is disconnected from all chains."
    });
  }
}
Object.defineProperty(w, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 4900
});
class P extends o {
  constructor(e) {
    super(e, {
      code: P.code,
      name: "ChainDisconnectedError",
      shortMessage: "The Provider is not connected to the requested chain."
    });
  }
}
Object.defineProperty(P, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 4901
});
class y extends o {
  constructor(e) {
    super(e, {
      code: y.code,
      name: "SwitchChainError",
      shortMessage: "An error occurred when attempting to switch chain."
    });
  }
}
Object.defineProperty(y, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 4902
});
class j extends o {
  constructor(e) {
    super(e, {
      code: j.code,
      name: "UnsupportedNonOptionalCapabilityError",
      shortMessage: "This Wallet does not support a capability that was not marked as optional."
    });
  }
}
Object.defineProperty(j, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 5700
});
class E extends o {
  constructor(e) {
    super(e, {
      code: E.code,
      name: "UnsupportedChainIdError",
      shortMessage: "This Wallet does not support the requested chain ID."
    });
  }
}
Object.defineProperty(E, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 5710
});
class R extends o {
  constructor(e) {
    super(e, {
      code: R.code,
      name: "DuplicateIdError",
      shortMessage: "There is already a bundle submitted with this ID."
    });
  }
}
Object.defineProperty(R, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 5720
});
class x extends o {
  constructor(e) {
    super(e, {
      code: x.code,
      name: "UnknownBundleIdError",
      shortMessage: "This bundle id is unknown / has not been submitted"
    });
  }
}
Object.defineProperty(x, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 5730
});
class M extends o {
  constructor(e) {
    super(e, {
      code: M.code,
      name: "BundleTooLargeError",
      shortMessage: "The call bundle is too large for the Wallet to process."
    });
  }
}
Object.defineProperty(M, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 5740
});
class O extends o {
  constructor(e) {
    super(e, {
      code: O.code,
      name: "AtomicReadyWalletRejectedUpgradeError",
      shortMessage: "The Wallet can support atomicity after an upgrade, but the user rejected the upgrade."
    });
  }
}
Object.defineProperty(O, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 5750
});
class T extends o {
  constructor(e) {
    super(e, {
      code: T.code,
      name: "AtomicityNotSupportedError",
      shortMessage: "The wallet does not support atomic execution but the request requires it."
    });
  }
}
Object.defineProperty(T, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 5760
});
class J extends r {
  constructor(e) {
    super(e, {
      name: "UnknownRpcError",
      shortMessage: "An unknown RPC error occurred."
    });
  }
}
export {
  O as AtomicReadyWalletRejectedUpgradeError,
  T as AtomicityNotSupportedError,
  M as BundleTooLargeError,
  P as ChainDisconnectedError,
  R as DuplicateIdError,
  n as InternalRpcError,
  d as InvalidInputRpcError,
  u as InvalidParamsRpcError,
  c as InvalidRequestRpcError,
  m as JsonRpcVersionUnsupportedError,
  h as LimitExceededRpcError,
  a as MethodNotFoundRpcError,
  p as MethodNotSupportedRpcError,
  s as ParseRpcError,
  w as ProviderDisconnectedError,
  o as ProviderRpcError,
  l as ResourceNotFoundRpcError,
  i as ResourceUnavailableRpcError,
  r as RpcError,
  y as SwitchChainError,
  b as TransactionRejectedRpcError,
  g as UnauthorizedProviderError,
  x as UnknownBundleIdError,
  J as UnknownRpcError,
  E as UnsupportedChainIdError,
  j as UnsupportedNonOptionalCapabilityError,
  v as UnsupportedProviderMethodError,
  f as UserRejectedRequestError
};
