// @coinbase/cdp-hooks/dist/esm/index67.js
import { Eip712DomainNotFoundError as u } from "./index253.js";
import { getAction as d } from "./index237.js";
import { readContract as f } from "./index80.js";
async function E(a, o) {
  const { address: e, factory: r, factoryData: i } = o;
  try {
    const [n, t, s, c, m, p, y] = await d(a, f, "readContract")({
      abi: g,
      address: e,
      functionName: "eip712Domain",
      factory: r,
      factoryData: i
    });
    return {
      domain: {
        name: t,
        version: s,
        chainId: Number(c),
        verifyingContract: m,
        salt: p
      },
      extensions: y,
      fields: n
    };
  } catch (n) {
    const t = n;
    throw t.name === "ContractFunctionExecutionError" && t.cause.name === "ContractFunctionZeroDataError" ? new u({ address: e }) : t;
  }
}
const g = [
  {
    inputs: [],
    name: "eip712Domain",
    outputs: [
      { name: "fields", type: "bytes1" },
      { name: "name", type: "string" },
      { name: "version", type: "string" },
      { name: "chainId", type: "uint256" },
      { name: "verifyingContract", type: "address" },
      { name: "salt", type: "bytes32" },
      { name: "extensions", type: "uint256[]" }
    ],
    stateMutability: "view",
    type: "function"
  }
];
export {
  E as getEip712Domain
};
