// @coinbase/cdp-hooks/dist/esm/index118.js
import { BaseError as t } from "./index107.js";
function c(a) {
  const r = Object.entries(a).map(([n, e]) => e === void 0 || e === !1 ? null : [n, e]).filter(Boolean), s = r.reduce((n, [e]) => Math.max(n, e.length), 0);
  return r.map(([n, e]) => `  ${`${n}:`.padEnd(s + 1)}  ${e}`).join(`
`);
}
class l extends t {
  constructor() {
    super([
      "Cannot specify both a `gasPrice` and a `maxFeePerGas`/`maxPriorityFeePerGas`.",
      "Use `maxFeePerGas`/`maxPriorityFeePerGas` for EIP-1559 compatible networks, and `gasPrice` for others."
    ].join(`
`), { name: "FeeConflictError" });
  }
}
class u extends t {
  constructor({ v: r }) {
    super(`Invalid \`v\` value "${r}". Expected 27 or 28.`, {
      name: "InvalidLegacyVError"
    });
  }
}
class f extends t {
  constructor({ transaction: r }) {
    super("Cannot infer a transaction type from provided transaction.", {
      metaMessages: [
        "Provided Transaction:",
        "{",
        c(r),
        "}",
        "",
        "To infer the type, either provide:",
        "- a `type` to the Transaction, or",
        "- an EIP-1559 Transaction with `maxFeePerGas`, or",
        "- an EIP-2930 Transaction with `gasPrice` & `accessList`, or",
        "- an EIP-4844 Transaction with `blobs`, `blobVersionedHashes`, `sidecars`, or",
        "- an EIP-7702 Transaction with `authorizationList`, or",
        "- a Legacy Transaction with `gasPrice`"
      ],
      name: "InvalidSerializableTransactionError"
    });
  }
}
class p extends t {
  constructor({ storageKey: r }) {
    super(`Size for storage key "${r}" is invalid. Expected 32 bytes. Got ${Math.floor((r.length - 2) / 2)} bytes.`, { name: "InvalidStorageKeySizeError" });
  }
}
class h extends t {
  constructor({ blockHash: r, blockNumber: s, blockTag: n, hash: e, index: o }) {
    let i = "Transaction";
    n && o !== void 0 && (i = `Transaction at block time "${n}" at index "${o}"`), r && o !== void 0 && (i = `Transaction at block hash "${r}" at index "${o}"`), s && o !== void 0 && (i = `Transaction at block number "${s}" at index "${o}"`), e && (i = `Transaction with hash "${e}"`), super(`${i} could not be found.`, {
      name: "TransactionNotFoundError"
    });
  }
}
class m extends t {
  constructor({ hash: r }) {
    super(`Transaction receipt with hash "${r}" could not be found. The Transaction may not be processed on a block yet.`, {
      name: "TransactionReceiptNotFoundError"
    });
  }
}
class T extends t {
  constructor({ hash: r }) {
    super(`Timed out while waiting for transaction with hash "${r}" to be confirmed.`, { name: "WaitForTransactionReceiptTimeoutError" });
  }
}
export {
  l as FeeConflictError,
  u as InvalidLegacyVError,
  f as InvalidSerializableTransactionError,
  p as InvalidStorageKeySizeError,
  h as TransactionNotFoundError,
  m as TransactionReceiptNotFoundError,
  T as WaitForTransactionReceiptTimeoutError,
  c as prettyPrint
};
