// @coinbase/cdp-hooks/dist/esm/index46.js
import { universalResolverReverseAbi as f } from "./index97.js";
import { getChainContractAddress as u } from "./index198.js";
import { toHex as h } from "./index134.js";
import { isNullUniversalResolverError as p } from "./index243.js";
import { packetToBytes as C } from "./index244.js";
import { getAction as w } from "./index237.js";
import { readContract as A } from "./index80.js";
async function b(e, { address: t, blockNumber: s, blockTag: i, gatewayUrls: a, strict: c, universalResolverAddress: d }) {
  let o = d;
  if (!o) {
    if (!e.chain)
      throw new Error("client chain not configured. universalResolverAddress is required.");
    o = u({
      blockNumber: s,
      chain: e.chain,
      contract: "ensUniversalResolver"
    });
  }
  const v = `${t.toLowerCase().substring(2)}.addr.reverse`;
  try {
    const r = {
      address: o,
      abi: f,
      functionName: "reverse",
      args: [h(C(v))],
      blockNumber: s,
      blockTag: i
    }, n = w(e, A, "readContract"), [l, m] = a ? await n({
      ...r,
      args: [...r.args, a]
    }) : await n(r);
    return t.toLowerCase() !== m.toLowerCase() ? null : l;
  } catch (r) {
    if (c)
      throw r;
    if (p(r, "reverse"))
      return null;
    throw r;
  }
}
export {
  b as getEnsName
};
