// @coinbase/cdp-hooks/dist/esm/index188.js
import { batchGatewayAbi as o } from "./index97.js";
import { solidityError as l } from "./index111.js";
import { decodeFunctionData as f } from "./index154.js";
import { encodeErrorResult as n } from "./index160.js";
import { encodeFunctionResult as p } from "./index164.js";
const d = "x-batch-gateway:true";
async function g(t) {
  const { data: i, ccipRequest: c } = t, { args: [u] } = f({ abi: o, data: i }), e = [], s = [];
  return await Promise.all(u.map(async (r, a) => {
    try {
      s[a] = r.urls.includes(d) ? await g({ data: r.data, ccipRequest: c }) : await c(r), e[a] = !1;
    } catch (m) {
      e[a] = !0, s[a] = h(m);
    }
  })), p({
    abi: o,
    functionName: "query",
    result: [e, s]
  });
}
function h(t) {
  return t.name === "HttpRequestError" && t.status ? n({
    abi: o,
    errorName: "HttpError",
    args: [t.status, t.shortMessage]
  }) : n({
    abi: [l],
    errorName: "Error",
    args: ["shortMessage" in t ? t.shortMessage : t.message]
  });
}
export {
  g as localBatchGatewayRequest,
  d as localBatchGatewayUrl
};
