// @coinbase/cdp-hooks/dist/esm/index149.js
import { AbiEncodingLengthMismatchError as y, InvalidAbiEncodingTypeError as u, InvalidArrayError as g, AbiEncodingArrayLengthMismatchError as l, AbiEncodingBytesSizeMismatchError as b } from "./index104.js";
import { InvalidAddressError as w } from "./index129.js";
import { BaseError as A } from "./index107.js";
import { IntegerOutOfRangeError as x } from "./index123.js";
import { isAddress as z } from "./index156.js";
import { concat as f } from "./index138.js";
import { padHex as d } from "./index157.js";
import { size as a } from "./index106.js";
import { slice as E } from "./index146.js";
import { numberToHex as m, boolToHex as P, stringToHex as S } from "./index134.js";
import { integerRegex as I } from "./index158.js";
function K(e, t) {
  if (e.length !== t.length)
    throw new y({
      expectedLength: e.length,
      givenLength: t.length
    });
  const o = L({
    params: e,
    values: t
  }), n = h(o);
  return n.length === 0 ? "0x" : n;
}
function L({ params: e, values: t }) {
  const o = [];
  for (let n = 0; n < e.length; n++)
    o.push(p({ param: e[n], value: t[n] }));
  return o;
}
function p({ param: e, value: t }) {
  const o = W(e.type);
  if (o) {
    const [n, r] = o;
    return M(t, { length: n, param: { ...e, type: r } });
  }
  if (e.type === "tuple")
    return H(t, {
      param: e
    });
  if (e.type === "address")
    return B(t);
  if (e.type === "bool")
    return T(t);
  if (e.type.startsWith("uint") || e.type.startsWith("int")) {
    const n = e.type.startsWith("int"), [, , r = "256"] = I.exec(e.type) ?? [];
    return $(t, {
      signed: n,
      size: Number(r)
    });
  }
  if (e.type.startsWith("bytes"))
    return N(t, { param: e });
  if (e.type === "string")
    return C(t);
  throw new u(e.type, {
    docsPath: "/docs/contract/encodeAbiParameters"
  });
}
function h(e) {
  let t = 0;
  for (let i = 0; i < e.length; i++) {
    const { dynamic: c, encoded: s } = e[i];
    c ? t += 32 : t += a(s);
  }
  const o = [], n = [];
  let r = 0;
  for (let i = 0; i < e.length; i++) {
    const { dynamic: c, encoded: s } = e[i];
    c ? (o.push(m(t + r, { size: 32 })), n.push(s), r += a(s)) : o.push(s);
  }
  return f([...o, ...n]);
}
function B(e) {
  if (!z(e))
    throw new w({ address: e });
  return { dynamic: !1, encoded: d(e.toLowerCase()) };
}
function M(e, { length: t, param: o }) {
  const n = t === null;
  if (!Array.isArray(e))
    throw new g(e);
  if (!n && e.length !== t)
    throw new l({
      expectedLength: t,
      givenLength: e.length,
      type: `${o.type}[${t}]`
    });
  let r = !1;
  const i = [];
  for (let c = 0; c < e.length; c++) {
    const s = p({ param: o, value: e[c] });
    s.dynamic && (r = !0), i.push(s);
  }
  if (n || r) {
    const c = h(i);
    if (n) {
      const s = m(i.length, { size: 32 });
      return {
        dynamic: !0,
        encoded: i.length > 0 ? f([s, c]) : s
      };
    }
    if (r)
      return { dynamic: !0, encoded: c };
  }
  return {
    dynamic: !1,
    encoded: f(i.map(({ encoded: c }) => c))
  };
}
function N(e, { param: t }) {
  const [, o] = t.type.split("bytes"), n = a(e);
  if (!o) {
    let r = e;
    return n % 32 !== 0 && (r = d(r, {
      dir: "right",
      size: Math.ceil((e.length - 2) / 2 / 32) * 32
    })), {
      dynamic: !0,
      encoded: f([d(m(n, { size: 32 })), r])
    };
  }
  if (n !== Number.parseInt(o))
    throw new b({
      expectedSize: Number.parseInt(o),
      value: e
    });
  return { dynamic: !1, encoded: d(e, { dir: "right" }) };
}
function T(e) {
  if (typeof e != "boolean")
    throw new A(`Invalid boolean value: "${e}" (type: ${typeof e}). Expected: \`true\` or \`false\`.`);
  return { dynamic: !1, encoded: d(P(e)) };
}
function $(e, { signed: t, size: o = 256 }) {
  if (typeof o == "number") {
    const n = 2n ** (BigInt(o) - (t ? 1n : 0n)) - 1n, r = t ? -n - 1n : 0n;
    if (e > n || e < r)
      throw new x({
        max: n.toString(),
        min: r.toString(),
        signed: t,
        size: o / 8,
        value: e.toString()
      });
  }
  return {
    dynamic: !1,
    encoded: m(e, {
      size: 32,
      signed: t
    })
  };
}
function C(e) {
  const t = S(e), o = Math.ceil(a(t) / 32), n = [];
  for (let r = 0; r < o; r++)
    n.push(d(E(t, r * 32, (r + 1) * 32), {
      dir: "right"
    }));
  return {
    dynamic: !0,
    encoded: f([
      d(m(a(t), { size: 32 })),
      ...n
    ])
  };
}
function H(e, { param: t }) {
  let o = !1;
  const n = [];
  for (let r = 0; r < t.components.length; r++) {
    const i = t.components[r], c = Array.isArray(e) ? r : i.name, s = p({
      param: i,
      value: e[c]
    });
    n.push(s), s.dynamic && (o = !0);
  }
  return {
    dynamic: o,
    encoded: o ? h(n) : f(n.map(({ encoded: r }) => r))
  };
}
function W(e) {
  const t = e.match(/^(.*)\[(\d+)?\]$/);
  return t ? (
    // Return `null` if the array is dynamic.
    [t[2] ? Number(t[2]) : null, t[1]]
  ) : void 0;
}
export {
  K as encodeAbiParameters,
  W as getArrayComponents
};
