// @coinbase/cdp-hooks/dist/esm/index72.js
import { encodeEventTopics as h } from "./index161.js";
import { parseEventLogs as l } from "./index165.js";
import { numberToHex as u } from "./index134.js";
import { formatLog as y } from "./index142.js";
async function T(n, { address: a, blockHash: m, fromBlock: o, toBlock: i, event: r, events: f, args: g, strict: L } = {}) {
  const b = L ?? !1, e = f ?? (r ? [r] : void 0);
  let t = [];
  e && (t = [e.flatMap((c) => h({
    abi: [c],
    eventName: c.name,
    args: f ? void 0 : g
  }))], r && (t = t[0]));
  let s;
  m ? s = await n.request({
    method: "eth_getLogs",
    params: [{ address: a, topics: t, blockHash: m }]
  }) : s = await n.request({
    method: "eth_getLogs",
    params: [
      {
        address: a,
        topics: t,
        fromBlock: typeof o == "bigint" ? u(o) : o,
        toBlock: typeof i == "bigint" ? u(i) : i
      }
    ]
  });
  const p = s.map((d) => y(d));
  return e ? l({
    abi: e,
    args: g,
    logs: p,
    strict: b
  }) : p;
}
export {
  T as getLogs
};
