// @coinbase/cdp-hooks/dist/esm/index35.js
import { parseAccount as I } from "./index36.js";
import { uid as M } from "./index37.js";
function O(n) {
  const { batch: s, chain: e, ccipRead: u, key: d = "base", name: f = "Base Client", type: m = "base" } = n, o = n.experimental_blockTag ?? (typeof e?.experimental_preconfirmationTime == "number" ? "pending" : void 0), b = e?.blockTime ?? 12e3, p = Math.min(Math.max(Math.floor(b / 2), 500), 4e3), t = n.pollingInterval ?? p, g = n.cacheTime ?? t, x = n.account ? I(n.account) : void 0, { config: h, request: T, value: k } = n.transport({
    chain: e,
    pollingInterval: t
  }), _ = { ...h, ...k }, c = {
    account: x,
    batch: s,
    cacheTime: g,
    ccipRead: u,
    chain: e,
    key: d,
    name: f,
    pollingInterval: t,
    request: T,
    transport: _,
    type: m,
    uid: M(),
    ...o ? { experimental_blockTag: o } : {}
  };
  function i(l) {
    return (v) => {
      const a = v(l);
      for (const y in c)
        delete a[y];
      const r = { ...l, ...a };
      return Object.assign(r, { extend: i(r) });
    };
  }
  return Object.assign(c, { extend: i(c) });
}
export {
  O as createClient
};
