// @coinbase/cdp-hooks/dist/esm/index7.js
import { listSpendPermissions as E } from "@coinbase/cdp-core";
import "./index2.js";
import { useSendHookCallOnce as C } from "./index25.js";
import { useState as a, useCallback as g, useMemo as m, useEffect as P } from "react";
import { useCurrentUser as T, useEnforceAuthenticated as y } from "./index5.js";
const U = (e = {}) => {
  C("use_list_spend_permissions");
  const n = e.network ?? "base-sepolia", d = e.pageSize ?? 20, f = e.pageToken, { currentUser: c } = T(), [v, o] = a("idle"), [h, l] = a(void 0), [b, i] = a(void 0), [k, A] = a(0), p = y(
    g(
      async (t) => E(t),
      []
      // No dependencies - the listSpendPermissions function is stable
    )
  ), r = m(
    () => e.evmSmartAccount ?? c?.evmSmartAccountObjects?.[0]?.address,
    [e.evmSmartAccount, c?.evmSmartAccountObjects?.[0]?.address]
  ), S = m(
    () => e.enabled !== !1 && !!(r && n && c),
    [e.enabled, r, n, c?.userId]
  ), u = m(() => r ? {
    evmSmartAccount: r,
    network: n,
    pageSize: d,
    pageToken: f
  } : null, [r, n, d, f]);
  P(() => {
    if (!S || !r || !u) {
      o("idle"), l(void 0), i(void 0);
      return;
    }
    let t = !1;
    return (async () => {
      o("pending"), i(void 0);
      try {
        const s = await p(u);
        t || (l(s), o("success"));
      } catch (s) {
        t || (i(s instanceof Error ? s : new Error(String(s))), o("error"));
      }
    })(), () => {
      t = !0;
    };
  }, [S, r, u, p, k]);
  const w = g(() => {
    A((t) => t + 1);
  }, []);
  return {
    status: v,
    data: h,
    error: b,
    refetch: w
  };
};
export {
  U as useListSpendPermissions
};
