// @coinbase/cdp-hooks/dist/esm/index286.js
import { Hash as d, ahash as u, toBytes as H, clean as l, aexists as a, abytes as p } from "./index270.js";
class r extends d {
  constructor(t, i) {
    super(), this.finished = !1, this.destroyed = !1, u(t);
    const h = H(i);
    if (this.iHash = t.create(), typeof this.iHash.update != "function")
      throw new Error("Expected instance of class which extends utils.Hash");
    this.blockLen = this.iHash.blockLen, this.outputLen = this.iHash.outputLen;
    const o = this.blockLen, e = new Uint8Array(o);
    e.set(h.length > o ? t.create().update(h).digest() : h);
    for (let s = 0; s < e.length; s++)
      e[s] ^= 54;
    this.iHash.update(e), this.oHash = t.create();
    for (let s = 0; s < e.length; s++)
      e[s] ^= 106;
    this.oHash.update(e), l(e);
  }
  update(t) {
    return a(this), this.iHash.update(t), this;
  }
  digestInto(t) {
    a(this), p(t, this.outputLen), this.finished = !0, this.iHash.digestInto(t), this.oHash.update(t), this.oHash.digestInto(t), this.destroy();
  }
  digest() {
    const t = new Uint8Array(this.oHash.outputLen);
    return this.digestInto(t), t;
  }
  _cloneInto(t) {
    t || (t = Object.create(Object.getPrototypeOf(this), {}));
    const { oHash: i, iHash: h, finished: o, destroyed: e, blockLen: s, outputLen: c } = this;
    return t = t, t.finished = o, t.destroyed = e, t.blockLen = s, t.outputLen = c, t.oHash = i._cloneInto(t.oHash), t.iHash = h._cloneInto(t.iHash), t;
  }
  clone() {
    return this._cloneInto();
  }
  destroy() {
    this.destroyed = !0, this.oHash.destroy(), this.iHash.destroy();
  }
}
const y = (n, t, i) => new r(n, t).update(i).digest();
y.create = (n, t) => new r(n, t);
export {
  r as HMAC,
  y as hmac
};
