// @coinbase/cdp-hooks/dist/esm/index41.js
import { TimeoutError as m, HttpRequestError as i } from "./index38.js";
import { withTimeout as j } from "./index202.js";
import { stringify as c } from "./index128.js";
import { idCache as y } from "./index242.js";
function E(s, o = {}) {
  return {
    async request(p) {
      const { body: r, onRequest: g = o.onRequest, onResponse: h = o.onResponse, timeout: f = o.timeout ?? 1e4 } = p, u = {
        ...o.fetchOptions ?? {},
        ...p.fetchOptions ?? {}
      }, { headers: l, method: R, signal: q } = u;
      try {
        const t = await j(async ({ signal: n }) => {
          const a = {
            ...u,
            body: Array.isArray(r) ? c(r.map((d) => ({
              jsonrpc: "2.0",
              id: d.id ?? y.take(),
              ...d
            }))) : c({
              jsonrpc: "2.0",
              id: r.id ?? y.take(),
              ...r
            }),
            headers: {
              "Content-Type": "application/json",
              ...l
            },
            method: R || "POST",
            signal: q || (f > 0 ? n : null)
          }, T = new Request(s, a), w = await g?.(T, a) ?? { ...a, url: s };
          return await fetch(w.url ?? s, w);
        }, {
          errorInstance: new m({ body: r, url: s }),
          timeout: f,
          signal: !0
        });
        h && await h(t);
        let e;
        if (t.headers.get("Content-Type")?.startsWith("application/json"))
          e = await t.json();
        else {
          e = await t.text();
          try {
            e = JSON.parse(e || "{}");
          } catch (n) {
            if (t.ok)
              throw n;
            e = { error: e };
          }
        }
        if (!t.ok)
          throw new i({
            body: r,
            details: c(e.error) || t.statusText,
            headers: t.headers,
            status: t.status,
            url: s
          });
        return e;
      } catch (t) {
        throw t instanceof i || t instanceof m ? t : new i({
          body: r,
          cause: t,
          url: s
        });
      }
    }
  };
}
export {
  E as getHttpRpcClient
};
