// @coinbase/cdp-hooks/dist/esm/index126.js
import { formatGwei as a } from "./index116.js";
import { BaseError as t } from "./index107.js";
class s extends t {
  constructor({ cause: r, message: e } = {}) {
    const n = e?.replace("execution reverted: ", "")?.replace("execution reverted", "");
    super(`Execution reverted ${n ? `with reason: ${n}` : "for an unknown reason"}.`, {
      cause: r,
      name: "ExecutionRevertedError"
    });
  }
}
Object.defineProperty(s, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 3
});
Object.defineProperty(s, "nodeMessage", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: /execution reverted/
});
class c extends t {
  constructor({ cause: r, maxFeePerGas: e } = {}) {
    super(`The fee cap (\`maxFeePerGas\`${e ? ` = ${a(e)} gwei` : ""}) cannot be higher than the maximum allowed value (2^256-1).`, {
      cause: r,
      name: "FeeCapTooHighError"
    });
  }
}
Object.defineProperty(c, "nodeMessage", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: /max fee per gas higher than 2\^256-1|fee cap higher than 2\^256-1/
});
class i extends t {
  constructor({ cause: r, maxFeePerGas: e } = {}) {
    super(`The fee cap (\`maxFeePerGas\`${e ? ` = ${a(e)}` : ""} gwei) cannot be lower than the block base fee.`, {
      cause: r,
      name: "FeeCapTooLowError"
    });
  }
}
Object.defineProperty(i, "nodeMessage", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: /max fee per gas less than block base fee|fee cap less than block base fee|transaction is outdated/
});
class u extends t {
  constructor({ cause: r, nonce: e } = {}) {
    super(`Nonce provided for the transaction ${e ? `(${e}) ` : ""}is higher than the next one expected.`, { cause: r, name: "NonceTooHighError" });
  }
}
Object.defineProperty(u, "nodeMessage", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: /nonce too high/
});
class l extends t {
  constructor({ cause: r, nonce: e } = {}) {
    super([
      `Nonce provided for the transaction ${e ? `(${e}) ` : ""}is lower than the current nonce of the account.`,
      "Try increasing the nonce or find the latest nonce with `getTransactionCount`."
    ].join(`
`), { cause: r, name: "NonceTooLowError" });
  }
}
Object.defineProperty(l, "nodeMessage", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: /nonce too low|transaction already imported|already known/
});
class d extends t {
  constructor({ cause: r, nonce: e } = {}) {
    super(`Nonce provided for the transaction ${e ? `(${e}) ` : ""}exceeds the maximum allowed nonce.`, { cause: r, name: "NonceMaxValueError" });
  }
}
Object.defineProperty(d, "nodeMessage", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: /nonce has max value/
});
class h extends t {
  constructor({ cause: r } = {}) {
    super([
      "The total cost (gas * gas fee + value) of executing this transaction exceeds the balance of the account."
    ].join(`
`), {
      cause: r,
      metaMessages: [
        "This error could arise when the account does not have enough funds to:",
        " - pay for the total gas fee,",
        " - pay for the value to send.",
        " ",
        "The cost of the transaction is calculated as `gas * gas fee + value`, where:",
        " - `gas` is the amount of gas needed for transaction to execute,",
        " - `gas fee` is the gas fee,",
        " - `value` is the amount of ether to send to the recipient."
      ],
      name: "InsufficientFundsError"
    });
  }
}
Object.defineProperty(h, "nodeMessage", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: /insufficient funds|exceeds transaction sender account balance/
});
class f extends t {
  constructor({ cause: r, gas: e } = {}) {
    super(`The amount of gas ${e ? `(${e}) ` : ""}provided for the transaction exceeds the limit allowed for the block.`, {
      cause: r,
      name: "IntrinsicGasTooHighError"
    });
  }
}
Object.defineProperty(f, "nodeMessage", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: /intrinsic gas too high|gas limit reached/
});
class p extends t {
  constructor({ cause: r, gas: e } = {}) {
    super(`The amount of gas ${e ? `(${e}) ` : ""}provided for the transaction is too low.`, {
      cause: r,
      name: "IntrinsicGasTooLowError"
    });
  }
}
Object.defineProperty(p, "nodeMessage", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: /intrinsic gas too low/
});
class g extends t {
  constructor({ cause: r }) {
    super("The transaction type is not supported for this chain.", {
      cause: r,
      name: "TransactionTypeNotSupportedError"
    });
  }
}
Object.defineProperty(g, "nodeMessage", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: /transaction type not valid/
});
class b extends t {
  constructor({ cause: r, maxPriorityFeePerGas: e, maxFeePerGas: n } = {}) {
    super([
      `The provided tip (\`maxPriorityFeePerGas\`${e ? ` = ${a(e)} gwei` : ""}) cannot be higher than the fee cap (\`maxFeePerGas\`${n ? ` = ${a(n)} gwei` : ""}).`
    ].join(`
`), {
      cause: r,
      name: "TipAboveFeeCapError"
    });
  }
}
Object.defineProperty(b, "nodeMessage", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: /max priority fee per gas higher than max fee per gas|tip higher than fee cap/
});
class x extends t {
  constructor({ cause: r }) {
    super(`An error occurred while executing: ${r?.shortMessage}`, {
      cause: r,
      name: "UnknownNodeError"
    });
  }
}
export {
  s as ExecutionRevertedError,
  c as FeeCapTooHighError,
  i as FeeCapTooLowError,
  h as InsufficientFundsError,
  f as IntrinsicGasTooHighError,
  p as IntrinsicGasTooLowError,
  d as NonceMaxValueError,
  u as NonceTooHighError,
  l as NonceTooLowError,
  b as TipAboveFeeCapError,
  g as TransactionTypeNotSupportedError,
  x as UnknownNodeError
};
