// @coinbase/cdp-hooks/dist/esm/index56.js
import { BaseFeeScalarError as p, Eip1559FeesNotSupportedError as F } from "./index120.js";
import { getAction as l } from "./index237.js";
import { internal_estimateMaxPriorityFeePerGas as u } from "./index58.js";
import { getBlock as G } from "./index61.js";
import { getGasPrice as b } from "./index71.js";
async function E(e, s) {
  return g(e, s);
}
async function g(e, s) {
  const { block: i, chain: a = e.chain, request: t, type: f = "eip1559" } = s || {}, o = await (async () => typeof a?.fees?.baseFeeMultiplier == "function" ? a.fees.baseFeeMultiplier({
    block: i,
    client: e,
    request: t
  }) : a?.fees?.baseFeeMultiplier ?? 1.2)();
  if (o < 1)
    throw new p();
  const m = 10 ** (o.toString().split(".")[1]?.length ?? 0), n = (r) => r * BigInt(Math.ceil(o * m)) / BigInt(m), c = i || await l(e, G, "getBlock")({});
  if (typeof a?.fees?.estimateFeesPerGas == "function") {
    const r = await a.fees.estimateFeesPerGas({
      block: i,
      client: e,
      multiply: n,
      request: t,
      type: f
    });
    if (r !== null)
      return r;
  }
  if (f === "eip1559") {
    if (typeof c.baseFeePerGas != "bigint")
      throw new F();
    const r = typeof t?.maxPriorityFeePerGas == "bigint" ? t.maxPriorityFeePerGas : await u(e, {
      block: c,
      chain: a,
      request: t
    }), P = n(c.baseFeePerGas);
    return {
      maxFeePerGas: t?.maxFeePerGas ?? P + r,
      maxPriorityFeePerGas: r
    };
  }
  return {
    gasPrice: t?.gasPrice ?? n(await l(e, b, "getGasPrice")({}))
  };
}
export {
  E as estimateFeesPerGas,
  g as internal_estimateFeesPerGas
};
