// @coinbase/cdp-hooks/dist/esm/index83.js
import { parseAccount as p } from "./index36.js";
import { decodeFunctionResult as h } from "./index155.js";
import { encodeFunctionData as x } from "./index162.js";
import { getContractError as b } from "./index205.js";
import { getAction as g } from "./index237.js";
import { call as A } from "./index49.js";
async function R(s, m) {
  const { abi: t, address: r, args: o, dataSuffix: n, functionName: a, ...c } = m, e = c.account ? p(c.account) : s.account, d = x({ abi: t, args: o, functionName: a });
  try {
    const { data: i } = await g(s, A, "call")({
      batch: !1,
      data: `${d}${n ? n.replace("0x", "") : ""}`,
      to: r,
      ...c,
      account: e
    }), l = h({
      abi: t,
      args: o,
      functionName: a,
      data: i || "0x"
    }), f = t.filter((u) => "name" in u && u.name === m.functionName);
    return {
      result: l,
      request: {
        abi: f,
        address: r,
        args: o,
        dataSuffix: n,
        functionName: a,
        ...c,
        account: e
      }
    };
  } catch (i) {
    throw b(i, {
      abi: t,
      address: r,
      args: o,
      docsPath: "/docs/contract/simulateContract",
      functionName: a,
      sender: e?.address
    });
  }
}
export {
  R as simulateContract
};
