// @coinbase/cdp-hooks/dist/esm/index58.js
import { Eip1559FeesNotSupportedError as f } from "./index120.js";
import { hexToBigInt as u } from "./index167.js";
import { getAction as s } from "./index237.js";
import { getBlock as c } from "./index61.js";
import { getGasPrice as y } from "./index71.js";
async function h(e, o) {
  return p(e, o);
}
async function p(e, o) {
  const { block: i, chain: n = e.chain, request: m } = o || {};
  try {
    const r = n?.fees?.maxPriorityFeePerGas ?? n?.fees?.defaultPriorityFee;
    if (typeof r == "function") {
      const t = i || await s(e, c, "getBlock")({}), P = await r({
        block: t,
        client: e,
        request: m
      });
      if (P === null)
        throw new Error();
      return P;
    }
    if (typeof r < "u")
      return r;
    const a = await e.request({
      method: "eth_maxPriorityFeePerGas"
    });
    return u(a);
  } catch {
    const [r, a] = await Promise.all([
      i ? Promise.resolve(i) : s(e, c, "getBlock")({}),
      s(e, y, "getGasPrice")({})
    ]);
    if (typeof r.baseFeePerGas != "bigint")
      throw new f();
    const t = a - r.baseFeePerGas;
    return t < 0n ? 0n : t;
  }
}
export {
  h as estimateMaxPriorityFeePerGas,
  p as internal_estimateMaxPriorityFeePerGas
};
