// @coinbase/cdp-hooks/dist/esm/index24.js
import { validateSecureIframeEnvironment as d, createEvmKeyExportIframe as b, createSolanaKeyExportIframe as v } from "@coinbase/cdp-core";
import "./index2.js";
import { useSendHookCallOnce as i } from "./index25.js";
import { useState as m, useRef as S, useCallback as f, useEffect as h } from "react";
import { useCDP as K } from "./index4.js";
function p(e, u) {
  const { config: c } = K(), [E, l] = m(null), [I, s] = m(void 0), r = S(null), a = f(() => {
    r.current?.cleanup(), r.current = null, l(null), s(void 0);
  }, []), x = f((t) => {
    r.current?.updateTheme(t);
  }, []);
  return h(() => {
    const t = e.containerRef.current;
    if (!t) return;
    let n = !1;
    return (async () => {
      const o = await u({
        address: e.address,
        target: t,
        projectId: c.projectId,
        label: e.label,
        copiedLabel: e.copiedLabel,
        icon: e.icon,
        theme: e.theme,
        onStatusUpdate: (y, _) => {
          n || (l(y), s(_));
        }
      });
      n ? o.cleanup() : r.current = o;
    })(), () => {
      n = !0, a();
    };
  }, [
    e.containerRef,
    e.address,
    u,
    e.label,
    e.copiedLabel,
    e.icon,
    e.theme,
    c.projectId,
    a
  ]), {
    status: E,
    message: I,
    updateTheme: x,
    cleanup: a
  };
}
function T(e) {
  return i("use_evm_key_export_iframe"), d(), p(e, b);
}
function w(e) {
  return i("use_solana_key_export_iframe"), d(), p(e, v);
}
export {
  T as useEvmKeyExportIframe,
  w as useSolanaKeyExportIframe
};
