// @coinbase/cdp-hooks/dist/esm/index165.js
import { AbiEventSignatureNotFoundError as m, DecodeLogDataMismatch as y, DecodeLogTopicsMismatch as l } from "./index104.js";
import { isAddressEqual as d } from "./index166.js";
import { toBytes as A } from "./index133.js";
import { keccak256 as p } from "./index135.js";
import { toEventSelector as g } from "./index152.js";
import { decodeEventLog as b } from "./index151.js";
function D(u) {
  const { abi: i, args: f, logs: s, strict: a = !0 } = u, t = (() => {
    if (u.eventName)
      return Array.isArray(u.eventName) ? u.eventName : [u.eventName];
  })();
  return s.map((e) => {
    try {
      const r = i.find((n) => n.type === "event" && e.topics[0] === g(n));
      if (!r)
        return null;
      const o = b({
        ...e,
        abi: [r],
        strict: a
      });
      return t && !t.includes(o.eventName) || !v({
        args: o.args,
        inputs: r.inputs,
        matchArgs: f
      }) ? null : { ...o, ...e };
    } catch (r) {
      let o, n;
      if (r instanceof m)
        return null;
      if (r instanceof y || r instanceof l) {
        if (a)
          return null;
        o = r.abiItem.name, n = r.abiItem.inputs?.some((c) => !("name" in c && c.name));
      }
      return { ...e, args: n ? [] : {}, eventName: o };
    }
  }).filter(Boolean);
}
function v(u) {
  const { args: i, inputs: f, matchArgs: s } = u;
  if (!s)
    return !0;
  if (!i)
    return !1;
  function a(t, e, r) {
    try {
      return t.type === "address" ? d(e, r) : t.type === "string" || t.type === "bytes" ? p(A(e)) === r : e === r;
    } catch {
      return !1;
    }
  }
  return Array.isArray(i) && Array.isArray(s) ? s.every((t, e) => {
    if (t == null)
      return !0;
    const r = f[e];
    return r ? (Array.isArray(t) ? t : [t]).some((n) => a(r, n, i[e])) : !1;
  }) : typeof i == "object" && !Array.isArray(i) && typeof s == "object" && !Array.isArray(s) ? Object.entries(s).every(([t, e]) => {
    if (e == null)
      return !0;
    const r = f.find((n) => n.name === t);
    return r ? (Array.isArray(e) ? e : [e]).some((n) => a(r, n, i[t])) : !1;
  }) : !1;
}
export {
  D as parseEventLogs
};
