// @coinbase/cdp-react/dist/chunks/LinkAuthFlow.3HN888DV.js
import { jsx as i, jsxs as m, Fragment as p } from "react/jsx-runtime";
import { createContext as S, useContext as I, useMemo as v, forwardRef as A, useRef as L, useImperativeHandle as x, useEffect as N } from "react";
import { useAppConfig as C } from "../components/CDPReactProvider/index.js";
import { SignIn as _ } from "../components/SignIn/index.js";
import "@coinbase/cdp-hooks";
import { SwitchSlideTransition as T } from "../components/ui/SwitchSlideTransition/index.js";
import { VisuallyHidden as h } from "../components/ui/VisuallyHidden/index.js";
import { LinkAuthItems as E } from "../components/LinkAuth/LinkAuthItems.js";
import { LINK_AUTH_FLOW_VIEWS as R } from "../components/LinkAuth/types.js";
import { useSignInContext as y } from "../components/SignIn/SignInProvider.js";
import { SignInImage as b } from "../components/SignIn/SignInImage.js";
import { SignInForm as H } from "../components/SignIn/SignInForm.js";
import { SignInTitle as g } from "../components/SignIn/SignInTitle.js";
import { SignInDescription as w } from "../components/SignIn/SignInDescription.js";
import { SignInFooter as P } from "../components/SignIn/SignInFooter.js";
import '../assets/LinkAuthFlow.css';const j = "LinkAuthFlow-module__list___dkNmh", c = {
  "sign-in": "LinkAuthFlow-module__sign-in___F2bj7",
  "sign-in-title-desc-wrapper": "LinkAuthFlow-module__sign-in-title-desc-wrapper___6QeTc",
  "sign-in-image": "LinkAuthFlow-module__sign-in-image___XGuJI",
  "sign-in-form": "LinkAuthFlow-module__sign-in-form___REZlP",
  "sign-in-footer": "LinkAuthFlow-module__sign-in-footer___KYZNd",
  list: j
}, k = S(null), f = () => {
  const t = I(k);
  if (!t)
    throw new Error("useLinkAuthFlowInternal must be used within a LinkAuthFlowContext.Provider");
  return t;
}, $ = () => {
  const {
    signInRef: t,
    transitionRef: u,
    ...s
  } = f();
  return s;
}, K = ({
  direction: t = "left",
  children: u,
  onBack: s,
  onLink: e,
  onLinkSuccess: r,
  signInRef: a,
  transitionRef: n
}) => {
  const o = v(
    () => ({
      direction: t,
      back: s,
      link: e,
      linkSuccess: r,
      signInRef: a,
      transitionRef: n
    }),
    [s, e, r, n, a, t]
  );
  return /* @__PURE__ */ i(k.Provider, { value: o, children: u });
}, d = A((t, u) => {
  const { showCoinbaseFooter: s } = C(), { state: e, dispatch: r } = y(), { linkSuccess: a } = f(), n = L(null);
  return x(
    u,
    () => ({
      canGoBack: e.step === "verification",
      goBack: () => {
        r({ type: "RESET_STATE" });
      }
    }),
    [e.step, r]
  ), N(() => {
    let o;
    return e.step === "credentials" && (o = setTimeout(() => {
      n.current?.querySelector("input[type='email'], input[type='tel']")?.focus();
    }, 300)), e.step === "verification" && (o = setTimeout(() => {
      n.current?.querySelector("input[type='text']")?.focus();
    }, 300)), () => clearTimeout(o);
  }, [e.step]), /* @__PURE__ */ m(p, { children: [
    /* @__PURE__ */ i("div", { className: c["sign-in-image"], children: /* @__PURE__ */ i(b, {}) }),
    /* @__PURE__ */ i(H, { autoFocus: !1, onSuccess: a, children: ({ authMethod: o, step: l, Form: F }) => /* @__PURE__ */ m(p, { children: [
      /* @__PURE__ */ m("div", { className: c["sign-in-title-desc-wrapper"], children: [
        l === "verification" ? /* @__PURE__ */ i(h, { as: "div", children: /* @__PURE__ */ i(g, {}) }) : /* @__PURE__ */ i(g, {}),
        l === "credentials" ? /* @__PURE__ */ i(h, { as: "div", children: /* @__PURE__ */ i(w, { authMethod: o }) }) : /* @__PURE__ */ i(w, { authMethod: o })
      ] }),
      /* @__PURE__ */ i("div", { ref: n, className: c["sign-in-form"], children: F })
    ] }) }),
    s && /* @__PURE__ */ i(P, { className: c["sign-in-footer"] })
  ] });
});
d.displayName = "SignInContent";
const ii = ({ children: t }) => {
  const { direction: u, link: s, linkSuccess: e, signInRef: r, transitionRef: a } = f();
  return /* @__PURE__ */ i(
    T,
    {
      autoFocus: !1,
      animateHeight: !0,
      items: R,
      initialEntered: !0,
      timeout: 250,
      direction: u,
      transitionRef: a,
      children: ({ itemKey: n, ...o }) => {
        let l = null;
        return n === "list" && (l = /* @__PURE__ */ i(E, { className: c.list, onLink: s })), n === "email" && (l = /* @__PURE__ */ i("div", { children: /* @__PURE__ */ i(_, { className: c["sign-in"], authMethods: ["email"], onSuccess: e, children: /* @__PURE__ */ i(d, { ref: r }) }) })), n === "sms" && (l = /* @__PURE__ */ i("div", { children: /* @__PURE__ */ i(_, { className: c["sign-in"], authMethods: ["sms"], onSuccess: e, children: /* @__PURE__ */ i(d, { ref: r }) }) })), /* @__PURE__ */ i("div", { ...o, children: t ? t({ view: n, Content: l }) : l });
      }
    }
  );
};
export {
  K as L,
  ii as a,
  f as b,
  $ as u
};
