// @coinbase/cdp-react/dist/chunks/useSendComponentCallOnce.BHZMuo6E.js
import { useRef as M, useEffect as q } from "react";
function U(s) {
  return s && s.__esModule && Object.prototype.hasOwnProperty.call(s, "default") ? s.default : s;
}
var E = { exports: {} }, T = { exports: {} }, S;
function H() {
  return S || (S = 1, function() {
    var s = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/", l = {
      // Bit-wise rotation left
      rotl: function(c, u) {
        return c << u | c >>> 32 - u;
      },
      // Bit-wise rotation right
      rotr: function(c, u) {
        return c << 32 - u | c >>> u;
      },
      // Swap big-endian to little-endian and vice versa
      endian: function(c) {
        if (c.constructor == Number)
          return l.rotl(c, 8) & 16711935 | l.rotl(c, 24) & 4278255360;
        for (var u = 0; u < c.length; u++)
          c[u] = l.endian(c[u]);
        return c;
      },
      // Generate an array of any length of random bytes
      randomBytes: function(c) {
        for (var u = []; c > 0; c--)
          u.push(Math.floor(Math.random() * 256));
        return u;
      },
      // Convert a byte array to big-endian 32-bit words
      bytesToWords: function(c) {
        for (var u = [], a = 0, f = 0; a < c.length; a++, f += 8)
          u[f >>> 5] |= c[a] << 24 - f % 32;
        return u;
      },
      // Convert big-endian 32-bit words to a byte array
      wordsToBytes: function(c) {
        for (var u = [], a = 0; a < c.length * 32; a += 8)
          u.push(c[a >>> 5] >>> 24 - a % 32 & 255);
        return u;
      },
      // Convert a byte array to a hex string
      bytesToHex: function(c) {
        for (var u = [], a = 0; a < c.length; a++)
          u.push((c[a] >>> 4).toString(16)), u.push((c[a] & 15).toString(16));
        return u.join("");
      },
      // Convert a hex string to a byte array
      hexToBytes: function(c) {
        for (var u = [], a = 0; a < c.length; a += 2)
          u.push(parseInt(c.substr(a, 2), 16));
        return u;
      },
      // Convert a byte array to a base-64 string
      bytesToBase64: function(c) {
        for (var u = [], a = 0; a < c.length; a += 3)
          for (var f = c[a] << 16 | c[a + 1] << 8 | c[a + 2], d = 0; d < 4; d++)
            a * 8 + d * 6 <= c.length * 8 ? u.push(s.charAt(f >>> 6 * (3 - d) & 63)) : u.push("=");
        return u.join("");
      },
      // Convert a base-64 string to a byte array
      base64ToBytes: function(c) {
        c = c.replace(/[^A-Z0-9+\/]/ig, "");
        for (var u = [], a = 0, f = 0; a < c.length; f = ++a % 4)
          f != 0 && u.push((s.indexOf(c.charAt(a - 1)) & Math.pow(2, -2 * f + 8) - 1) << f * 2 | s.indexOf(c.charAt(a)) >>> 6 - f * 2);
        return u;
      }
    };
    T.exports = l;
  }()), T.exports;
}
var F, m;
function x() {
  if (m) return F;
  m = 1;
  var s = {
    // UTF-8 encoding
    utf8: {
      // Convert a string to a byte array
      stringToBytes: function(l) {
        return s.bin.stringToBytes(unescape(encodeURIComponent(l)));
      },
      // Convert a byte array to a string
      bytesToString: function(l) {
        return decodeURIComponent(escape(s.bin.bytesToString(l)));
      }
    },
    // Binary encoding
    bin: {
      // Convert a string to a byte array
      stringToBytes: function(l) {
        for (var c = [], u = 0; u < l.length; u++)
          c.push(l.charCodeAt(u) & 255);
        return c;
      },
      // Convert a byte array to a string
      bytesToString: function(l) {
        for (var c = [], u = 0; u < l.length; u++)
          c.push(String.fromCharCode(l[u]));
        return c.join("");
      }
    }
  };
  return F = s, F;
}
/*!
 * Determine if an object is a Buffer
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
var B, O;
function N() {
  if (O) return B;
  O = 1, B = function(c) {
    return c != null && (s(c) || l(c) || !!c._isBuffer);
  };
  function s(c) {
    return !!c.constructor && typeof c.constructor.isBuffer == "function" && c.constructor.isBuffer(c);
  }
  function l(c) {
    return typeof c.readFloatLE == "function" && typeof c.slice == "function" && s(c.slice(0, 0));
  }
  return B;
}
var A;
function V() {
  return A || (A = 1, function() {
    var s = H(), l = x().utf8, c = N(), u = x().bin, a = function(f, d) {
      f.constructor == String ? d && d.encoding === "binary" ? f = u.stringToBytes(f) : f = l.stringToBytes(f) : c(f) ? f = Array.prototype.slice.call(f, 0) : !Array.isArray(f) && f.constructor !== Uint8Array && (f = f.toString());
      for (var o = s.bytesToWords(f), y = f.length * 8, r = 1732584193, t = -271733879, e = -1732584194, n = 271733878, i = 0; i < o.length; i++)
        o[i] = (o[i] << 8 | o[i] >>> 24) & 16711935 | (o[i] << 24 | o[i] >>> 8) & 4278255360;
      o[y >>> 5] |= 128 << y % 32, o[(y + 64 >>> 9 << 4) + 14] = y;
      for (var p = a._ff, h = a._gg, v = a._hh, _ = a._ii, i = 0; i < o.length; i += 16) {
        var P = r, D = t, I = e, L = n;
        r = p(r, t, e, n, o[i + 0], 7, -680876936), n = p(n, r, t, e, o[i + 1], 12, -389564586), e = p(e, n, r, t, o[i + 2], 17, 606105819), t = p(t, e, n, r, o[i + 3], 22, -1044525330), r = p(r, t, e, n, o[i + 4], 7, -176418897), n = p(n, r, t, e, o[i + 5], 12, 1200080426), e = p(e, n, r, t, o[i + 6], 17, -1473231341), t = p(t, e, n, r, o[i + 7], 22, -45705983), r = p(r, t, e, n, o[i + 8], 7, 1770035416), n = p(n, r, t, e, o[i + 9], 12, -1958414417), e = p(e, n, r, t, o[i + 10], 17, -42063), t = p(t, e, n, r, o[i + 11], 22, -1990404162), r = p(r, t, e, n, o[i + 12], 7, 1804603682), n = p(n, r, t, e, o[i + 13], 12, -40341101), e = p(e, n, r, t, o[i + 14], 17, -1502002290), t = p(t, e, n, r, o[i + 15], 22, 1236535329), r = h(r, t, e, n, o[i + 1], 5, -165796510), n = h(n, r, t, e, o[i + 6], 9, -1069501632), e = h(e, n, r, t, o[i + 11], 14, 643717713), t = h(t, e, n, r, o[i + 0], 20, -373897302), r = h(r, t, e, n, o[i + 5], 5, -701558691), n = h(n, r, t, e, o[i + 10], 9, 38016083), e = h(e, n, r, t, o[i + 15], 14, -660478335), t = h(t, e, n, r, o[i + 4], 20, -405537848), r = h(r, t, e, n, o[i + 9], 5, 568446438), n = h(n, r, t, e, o[i + 14], 9, -1019803690), e = h(e, n, r, t, o[i + 3], 14, -187363961), t = h(t, e, n, r, o[i + 8], 20, 1163531501), r = h(r, t, e, n, o[i + 13], 5, -1444681467), n = h(n, r, t, e, o[i + 2], 9, -51403784), e = h(e, n, r, t, o[i + 7], 14, 1735328473), t = h(t, e, n, r, o[i + 12], 20, -1926607734), r = v(r, t, e, n, o[i + 5], 4, -378558), n = v(n, r, t, e, o[i + 8], 11, -2022574463), e = v(e, n, r, t, o[i + 11], 16, 1839030562), t = v(t, e, n, r, o[i + 14], 23, -35309556), r = v(r, t, e, n, o[i + 1], 4, -1530992060), n = v(n, r, t, e, o[i + 4], 11, 1272893353), e = v(e, n, r, t, o[i + 7], 16, -155497632), t = v(t, e, n, r, o[i + 10], 23, -1094730640), r = v(r, t, e, n, o[i + 13], 4, 681279174), n = v(n, r, t, e, o[i + 0], 11, -358537222), e = v(e, n, r, t, o[i + 3], 16, -722521979), t = v(t, e, n, r, o[i + 6], 23, 76029189), r = v(r, t, e, n, o[i + 9], 4, -640364487), n = v(n, r, t, e, o[i + 12], 11, -421815835), e = v(e, n, r, t, o[i + 15], 16, 530742520), t = v(t, e, n, r, o[i + 2], 23, -995338651), r = _(r, t, e, n, o[i + 0], 6, -198630844), n = _(n, r, t, e, o[i + 7], 10, 1126891415), e = _(e, n, r, t, o[i + 14], 15, -1416354905), t = _(t, e, n, r, o[i + 5], 21, -57434055), r = _(r, t, e, n, o[i + 12], 6, 1700485571), n = _(n, r, t, e, o[i + 3], 10, -1894986606), e = _(e, n, r, t, o[i + 10], 15, -1051523), t = _(t, e, n, r, o[i + 1], 21, -2054922799), r = _(r, t, e, n, o[i + 8], 6, 1873313359), n = _(n, r, t, e, o[i + 15], 10, -30611744), e = _(e, n, r, t, o[i + 6], 15, -1560198380), t = _(t, e, n, r, o[i + 13], 21, 1309151649), r = _(r, t, e, n, o[i + 4], 6, -145523070), n = _(n, r, t, e, o[i + 11], 10, -1120210379), e = _(e, n, r, t, o[i + 2], 15, 718787259), t = _(t, e, n, r, o[i + 9], 21, -343485551), r = r + P >>> 0, t = t + D >>> 0, e = e + I >>> 0, n = n + L >>> 0;
      }
      return s.endian([r, t, e, n]);
    };
    a._ff = function(f, d, o, y, r, t, e) {
      var n = f + (d & o | ~d & y) + (r >>> 0) + e;
      return (n << t | n >>> 32 - t) + d;
    }, a._gg = function(f, d, o, y, r, t, e) {
      var n = f + (d & y | o & ~y) + (r >>> 0) + e;
      return (n << t | n >>> 32 - t) + d;
    }, a._hh = function(f, d, o, y, r, t, e) {
      var n = f + (d ^ o ^ y) + (r >>> 0) + e;
      return (n << t | n >>> 32 - t) + d;
    }, a._ii = function(f, d, o, y, r, t, e) {
      var n = f + (o ^ (d | ~y)) + (r >>> 0) + e;
      return (n << t | n >>> 32 - t) + d;
    }, a._blocksize = 16, a._digestsize = 16, E.exports = function(f, d) {
      if (f == null)
        throw new Error("Illegal argument " + f);
      var o = s.wordsToBytes(a(f, d));
      return d && d.asBytes ? o : d && d.asString ? u.bytesToString(o) : s.bytesToHex(o);
    };
  }()), E.exports;
}
var G = V();
const J = /* @__PURE__ */ U(G);
function K() {
  try {
    return typeof navigator < "u" && navigator.product === "ReactNative";
  } catch {
    return !1;
  }
}
function $() {
  try {
    if (typeof window > "u")
      return typeof process < "u" && process.env && "__NEXT_PROCESSED_ENV" in process.env;
    const s = window;
    return !!(s.__NEXT_DATA__ || s.next || document.getElementById("__next"));
  } catch {
    return !1;
  }
}
function z() {
  try {
    if (typeof window > "u")
      return !1;
    const s = window;
    return !!(s.__VUE__ || s.__VUE_DEVTOOLS_GLOBAL_HOOK__ || document.querySelector("[data-v-]"));
  } catch {
    return !1;
  }
}
function W() {
  try {
    if (typeof window > "u")
      return !1;
    const s = window;
    return s.__REACT_DEVTOOLS_GLOBAL_HOOK__ || s.React ? !0 : !!document.querySelector(
      "[data-reactroot], [data-reactid], ._reactRoot, ._reactRootContainer"
    );
  } catch {
    return !1;
  }
}
function X() {
  return W() ? "react" : K() ? "react-native" : $() ? "nextjs" : z() ? "vue" : "unknown";
}
let R = null;
function Z() {
  return R === null && (R = X()), R;
}
const b = "1f8d3e558f493e38a037dbfadb8ba344", Q = "d473b1dcc969105adab157226cd9cf63", C = Symbol.for("@coinbase/cdp-analytics");
function Y() {
  const s = globalThis;
  return s[C] || (s[C] = {
    identifier: "",
    enabled: !0,
    isProduction: !1,
    isUsingMocks: !1,
    sendEvent: w,
    sendActionCallEvent: (l) => w({ ...l, type: "action_call" }),
    sendHookCallEvent: (l) => w({ ...l, type: "hook_call" }),
    sendComponentCallEvent: (l) => w({ ...l, type: "component_call" }),
    versionRegistry: {},
    registerPackageVersion: (l, c) => {
      g.versionRegistry[l] = c;
    }
  }), s[C];
}
const g = Y();
async function w(s) {
  if (!g.enabled || !g.identifier)
    return;
  const l = Date.now(), u = [{
    user_id: g.identifier,
    event_type: s.type,
    platform: k() ? "web" : "native",
    timestamp: l,
    event_properties: {
      project_name: "cdp-embedded-wallet",
      framework: Z(),
      isLocalHost: j(),
      isUsingMocks: g.isUsingMocks,
      package_versions: g.versionRegistry,
      ...s
    }
  }], a = JSON.stringify(u), f = l.toString(), d = J(a + f), o = {
    client: g.isProduction ? Q : b,
    e: a,
    checksum: d
  };
  await fetch("https://cca-lite.coinbase.com/amp", {
    method: "POST",
    mode: "no-cors",
    headers: {
      "Content-Type": "application/json"
    },
    body: JSON.stringify(o)
  });
}
function k() {
  return typeof window < "u" && typeof document < "u";
}
function j() {
  return k() ? window.location.hostname === "localhost" || window.location.hostname === "127.0.0.1" : !1;
}
function nt(s) {
  const l = M(!1);
  q(() => {
    l.current || (g.sendComponentCallEvent({ name: s }), l.current = !0);
  }, [s]);
}
export {
  g as A,
  nt as u
};
