// @coinbase/cdp-api-client/dist/esm/index5.js
import { cdpApiClient as s } from "./index7.js";
const d = (e, t, n) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/auth/init`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: t
  },
  n
), i = (e, t, n) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/auth/verify/email`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: t
  },
  n
), p = (e, t, n) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/auth/verify/sms`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: t
  },
  n
), c = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/auth/verify/oauth/${t}`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), l = (e, t, n) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/auth/refresh`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: t
  },
  n
), u = (e, t, n) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/auth/logout`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: t
  },
  n
), m = (e, t) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/auth/custom/authenticate`,
    method: "POST"
  },
  t
), h = (e, t) => s(
  { url: `/v2/embedded-wallet-api/projects/${e}/auth/mfa`, method: "GET" },
  t
), $ = (e, t, n) => s(
  { url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}`, method: "GET" },
  n
), T = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/wallet-secrets`,
    method: "PUT",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), v = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), j = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/sign`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), S = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/sign/transaction`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), b = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/send/transaction`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), y = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/sign/message`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), E = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/sign/typed-data`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), w = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/export`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), P = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/smart-accounts`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), O = (e, t, n, a, r) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/smart-accounts/${n}/user-operations/${a}`,
    method: "GET"
  },
  r
), C = (e, t, n, a, r) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/smart-accounts/${n}/send`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: a
  },
  r
), A = (e, t, n, a, r) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/smart-accounts/${n}/spend-permissions`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: a
  },
  r
), f = (e, t, n, a, r) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/smart-accounts/${n}/spend-permissions/list`,
    method: "GET",
    params: a
  },
  r
), U = (e, t, n, a, r) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/smart-accounts/${n}/spend-permissions/revoke`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: a
  },
  r
), g = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/solana`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), W = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/solana/export`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), M = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/solana/sign`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), x = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/solana/sign/message`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), G = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/solana/sign/transaction`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), k = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/solana/send/transaction`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), H = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/mfa/enroll/${n}/init`,
    method: "POST"
  },
  a
), V = (e, t, n, a, r) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/mfa/enroll/${n}/submit`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: a
  },
  r
), D = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/mfa/verify/${n}/init`,
    method: "POST"
  },
  a
), q = (e, t, n, a, r) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/mfa/verify/${n}/submit`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: a
  },
  r
), z = (e, t, n) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/mfa/prompt`,
    method: "PUT"
  },
  n
);
export {
  m as authenticateUserWithCustomAuth,
  v as createEndUserEvmAccount,
  P as createEndUserEvmSmartAccount,
  g as createEndUserSolanaAccount,
  A as createSpendPermissionWithEndUserAccount,
  w as exportEndUserEvmAccount,
  W as exportEndUserSolanaAccount,
  $ as getAuthenticatedEndUser,
  h as getMfaConfig,
  O as getUserOperationWithEndUserAccount,
  d as initiateAuthentication,
  H as initiateMfaEnrollment,
  D as initiateMfaVerification,
  f as listSpendPermissionsWithEndUserAccount,
  u as logOutEndUser,
  z as recordMfaEnrollmentPrompted,
  l as refreshAccessToken,
  T as registerTemporaryWalletSecret,
  U as revokeSpendPermissionWithEndUserAccount,
  b as sendEvmTransactionWithEndUserAccount,
  k as sendSolanaTransactionWithEndUserAccount,
  C as sendUserOperationWithEndUserAccount,
  j as signEvmHashWithEndUserAccount,
  y as signEvmMessageWithEndUserAccount,
  S as signEvmTransactionWithEndUserAccount,
  E as signEvmTypedDataWithEndUserAccount,
  M as signSolanaHashWithEndUserAccount,
  x as signSolanaMessageWithEndUserAccount,
  G as signSolanaTransactionWithEndUserAccount,
  V as submitMfaEnrollment,
  q as submitMfaVerification,
  i as verifyEmailAuthentication,
  c as verifyOAuthCode,
  p as verifySmsAuthentication
};
