// @coinbase/cdp-react/dist/components/Fund/utils/buildPaymentMethods.js
import { jsx as r } from "react/jsx-runtime";
import { IconAppleLogo as u } from "../../../icons/IconAppleLogo.js";
import { IconCoinbaseMark as A } from "../../../icons/IconCoinbaseMark.js";
import { IconCreditCard as C } from "../../../icons/IconCreditCard.js";
const s = 2, c = 500, b = (n) => [
  // Check API-provided methods
  n.some((e) => e.id === "ACH_BANK_ACCOUNT") && "ACH",
  n.some((e) => e.id === "CARD") && "debit",
  n.some((e) => e.id === "FIAT_WALLET") && "cash",
  // Always include these methods
  "crypto balance"
].filter(Boolean).join(", "), M = ({ limits: n }) => {
  const o = n.flatMap((t) => t.min ? [Number(t.min)] : []), e = n.flatMap((t) => t.max ? [Number(t.max)] : []);
  return {
    id: n[0].id,
    // use the first limit id as the coinbase payment method id to avoid passing in an invalid option
    name: "Coinbase",
    description: b(n),
    icon: /* @__PURE__ */ r(A, {}),
    minAmount: o.length > 0 ? Math.min(...o) : s,
    maxAmount: e.length > 0 ? Math.max(...e) : c
  };
}, l = (n) => [
  {
    id: "APPLE_PAY_GUEST",
    name: "Apple Pay",
    icon: /* @__PURE__ */ r(u, {})
  },
  {
    id: "CARD_GUEST",
    name: "Debit card",
    icon: /* @__PURE__ */ r(C, {})
  }
].map((e) => {
  const t = n.limits.find(
    (i) => i.id === e.id.replace("_GUEST", "")
  );
  return {
    ...e,
    description: `Up to $500/week. No sign up required.${e.id === "APPLE_PAY_GUEST" ? " Debit only." : ""}`,
    minAmount: Number(t?.min) || s,
    maxAmount: Number(t?.max) || c
  };
}), f = (n, o, e) => {
  const t = o.toUpperCase(), i = e.toUpperCase(), a = n.paymentCurrencies.find(
    (p) => p.id.toUpperCase() === t
  );
  if (!a)
    return [];
  const d = M(a);
  let m = [];
  return i === "US" && t === "USD" && (m = l(a)), [d, ...m];
};
export {
  b as buildCoinbasePaymentMethodDescription,
  f as buildPaymentMethods
};
