// @coinbase/cdp-react/dist/components/ui/ButtonBase/index.js
import { jsx as l } from "react/jsx-runtime";
import { Primitive as n } from "@radix-ui/react-primitive";
import { c as u } from "../../../chunks/lite.1fxw3LjI.js";
import { forwardRef as B } from "react";
import "../../../theme/theme.js";
import '../../../assets/ButtonBase.css';const r = "ButtonBase-module__btn___9S8PJ", s = {
  btn: r,
  "full-width": "ButtonBase-module__full-width___vLJsn",
  "size-lg": "ButtonBase-module__size-lg___569jT",
  "size-md": "ButtonBase-module__size-md___sAsBJ",
  "size-sm": "ButtonBase-module__size-sm___xHCtU",
  "size-xs": "ButtonBase-module__size-xs___LiiO7"
}, i = {
  lg: s["size-lg"],
  md: s["size-md"],
  sm: s["size-sm"],
  xs: s["size-xs"],
  none: ""
}, z = B(
  ({
    children: _,
    className: m = "",
    fullWidth: t = !1,
    isPending: a,
    size: e = "md",
    ...o
  }, d) => /* @__PURE__ */ l(
    n.button,
    {
      className: o.asChild ? "" : u(
        s.btn,
        i[e] ?? i.md,
        t && s["full-width"],
        m
      ),
      ref: d,
      "data-pending": a,
      "data-size": e,
      "data-full-width": t === !0 ? !0 : void 0,
      ...o,
      children: _
    }
  )
);
z.displayName = "ButtonBase";
export {
  z as ButtonBase
};
