// @coinbase/cdp-react/dist/components/CopyAddress/index.js
import { jsxs as c, jsx as o } from "react/jsx-runtime";
import { c as u } from "../../chunks/lite.1fxw3LjI.js";
import { useRef as m, useState as y, useCallback as C } from "react";
import { Label as b } from "../forms/Label/index.js";
import { IconCheck as f } from "../../icons/IconCheck.js";
import { IconCopy as h } from "../../icons/IconCopy.js";
import '../../assets/CopyAddress.css';const A = "CopyAddress-module__label___O2o5m", e = {
  "address-copy": "CopyAddress-module__address-copy___KpKeI",
  label: A,
  "address-input-container": "CopyAddress-module__address-input-container___qq0TH",
  "address-truncated": "CopyAddress-module__address-truncated___kr8St",
  "copy-button": "CopyAddress-module__copy-button___64o2y",
  "success-icon": "CopyAddress-module__success-icon___qXipU",
  "copy-icon": "CopyAddress-module__copy-icon___lUbYG"
}, T = ({
  className: l = "",
  address: s,
  label: r,
  onCopy: t,
  onCopySuccess: d,
  ...i
}) => {
  const p = m(null), [_, a] = y(!1), n = C(() => {
    if (s) {
      t?.();
      try {
        if (navigator.clipboard)
          navigator.clipboard.writeText(s);
        else
          throw new Error("Clipboard API not supported");
      } catch {
        p.current?.select(), document.execCommand("copy");
      }
      a(!0), setTimeout(() => {
        a(!1);
      }, 2e3), d?.();
    }
  }, [s, t, d]);
  return /* @__PURE__ */ c("div", { className: u(e["address-copy"], l), ...i, children: [
    r && /* @__PURE__ */ o(b, { as: "span", className: e.label, children: r }),
    /* @__PURE__ */ c("div", { className: e["address-input-container"], onClick: n, children: [
      /* @__PURE__ */ c("span", { className: e["address-truncated"], children: [
        s.slice(0, 6),
        "...",
        s.slice(-4)
      ] }),
      /* @__PURE__ */ o(
        "button",
        {
          type: "button",
          "aria-label": "copy address",
          onClick: n,
          className: e["copy-button"],
          children: _ ? /* @__PURE__ */ o(f, { className: e["success-icon"] }) : /* @__PURE__ */ o(h, { className: e["copy-icon"] })
        }
      )
    ] })
  ] });
};
export {
  T as CopyAddress
};
